{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/Andoniadou-Lab/consensus_pituitary_atlas/blob/main/download_pseudoalign/Create_pituitary_atlas_rna.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "i7Iu8vKDAyry"
      },
      "source": [
        "# Install packages"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "yxxoe8-45IoL",
        "outputId": "b26cac50-1b54-459b-e576-f883b5026004"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\u001b[?25l     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m0.0/49.6 kB\u001b[0m \u001b[31m?\u001b[0m eta \u001b[36m-:--:--\u001b[0m\r\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m49.6/49.6 kB\u001b[0m \u001b[31m1.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m36.5/36.5 MB\u001b[0m \u001b[31m68.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m169.3/169.3 kB\u001b[0m \u001b[31m9.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m3.3/3.3 MB\u001b[0m \u001b[31m104.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m50.9/50.9 MB\u001b[0m \u001b[31m3.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m2.1/2.1 MB\u001b[0m \u001b[31m73.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m58.2/58.2 kB\u001b[0m \u001b[31m3.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m24.0/24.0 MB\u001b[0m \u001b[31m88.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m276.4/276.4 kB\u001b[0m \u001b[31m16.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m40.8/40.8 kB\u001b[0m \u001b[31m2.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m8.8/8.8 MB\u001b[0m \u001b[31m110.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m80.0/80.0 kB\u001b[0m \u001b[31m4.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Building wheel for loompy (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Reading package lists...\n",
            "Building dependency tree...\n",
            "Reading state information...\n",
            "jq is already the newest version (1.6-2.1ubuntu3.1).\n",
            "0 upgraded, 0 newly installed, 0 to remove and 38 not upgraded.\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  blends-common libkdf5-2 libncbi-vdb2 libncbi-wvdb2 med-config menu\n",
            "Suggested packages:\n",
            "  blends-doc menu-l10n gksu | kde-runtime | ktsuss\n",
            "The following NEW packages will be installed:\n",
            "  blends-common libkdf5-2 libncbi-vdb2 libncbi-wvdb2 med-config menu\n",
            "  sra-toolkit\n",
            "0 upgraded, 7 newly installed, 0 to remove and 38 not upgraded.\n",
            "Need to get 8,290 kB of archives.\n",
            "After this operation, 23.0 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 menu amd64 2.1.47ubuntu4 [354 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu jammy/universe amd64 blends-common all 0.7.4ubuntu1 [15.9 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libkdf5-2 amd64 2.11.2+dfsg-4build2 [14.7 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libncbi-vdb2 amd64 2.11.2+dfsg-4build2 [1,364 kB]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libncbi-wvdb2 amd64 2.11.2+dfsg-4build2 [1,252 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu jammy/universe amd64 sra-toolkit amd64 2.11.3+dfsg-1ubuntu1 [5,277 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu jammy/universe amd64 med-config all 3.7.1 [11.6 kB]\n",
            "Fetched 8,290 kB in 3s (3,191 kB/s)\n",
            "Preconfiguring packages ...\n",
            "Selecting previously unselected package menu.\n",
            "(Reading database ... 126675 files and directories currently installed.)\n",
            "Preparing to unpack .../0-menu_2.1.47ubuntu4_amd64.deb ...\n",
            "Unpacking menu (2.1.47ubuntu4) ...\n",
            "Selecting previously unselected package blends-common.\n",
            "Preparing to unpack .../1-blends-common_0.7.4ubuntu1_all.deb ...\n",
            "Unpacking blends-common (0.7.4ubuntu1) ...\n",
            "Selecting previously unselected package libkdf5-2.\n",
            "Preparing to unpack .../2-libkdf5-2_2.11.2+dfsg-4build2_amd64.deb ...\n",
            "Unpacking libkdf5-2 (2.11.2+dfsg-4build2) ...\n",
            "Selecting previously unselected package libncbi-vdb2.\n",
            "Preparing to unpack .../3-libncbi-vdb2_2.11.2+dfsg-4build2_amd64.deb ...\n",
            "Unpacking libncbi-vdb2 (2.11.2+dfsg-4build2) ...\n",
            "Selecting previously unselected package libncbi-wvdb2.\n",
            "Preparing to unpack .../4-libncbi-wvdb2_2.11.2+dfsg-4build2_amd64.deb ...\n",
            "Unpacking libncbi-wvdb2 (2.11.2+dfsg-4build2) ...\n",
            "Selecting previously unselected package sra-toolkit.\n",
            "Preparing to unpack .../5-sra-toolkit_2.11.3+dfsg-1ubuntu1_amd64.deb ...\n",
            "Unpacking sra-toolkit (2.11.3+dfsg-1ubuntu1) ...\n",
            "Selecting previously unselected package med-config.\n",
            "Preparing to unpack .../6-med-config_3.7.1_all.deb ...\n",
            "Unpacking med-config (3.7.1) ...\n",
            "Setting up libncbi-vdb2 (2.11.2+dfsg-4build2) ...\n",
            "Setting up libkdf5-2 (2.11.2+dfsg-4build2) ...\n",
            "Setting up libncbi-wvdb2 (2.11.2+dfsg-4build2) ...\n",
            "Setting up menu (2.1.47ubuntu4) ...\n",
            "Setting up blends-common (0.7.4ubuntu1) ...\n",
            "Setting up sra-toolkit (2.11.3+dfsg-1ubuntu1) ...\n",
            "Setting up med-config (3.7.1) ...\n",
            "Adding group `med' (GID 107) ...\n",
            "Done.\n",
            "Processing triggers for man-db (2.10.2-1) ...\n",
            "Processing triggers for libc-bin (2.35-0ubuntu3.8) ...\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc_proxy.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_loader.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtcm.so.1 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_5.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtcm_debug.so.1 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libumf.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbb.so.12 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_level_zero_v2.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_opencl.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_level_zero.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_0.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libhwloc.so.15 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc.so.2 is not a symbolic link\n",
            "\n",
            "Processing triggers for menu (2.1.47ubuntu4) ...\n",
            "Collecting epitome_tools\n",
            "  Downloading epitome_tools-0.0.17.tar.gz (36.2 MB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m36.2/36.2 MB\u001b[0m \u001b[31m61.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Requirement already satisfied: xgboost in /usr/local/lib/python3.12/dist-packages (from epitome_tools) (3.0.5)\n",
            "Requirement already satisfied: scipy in /usr/local/lib/python3.12/dist-packages (from epitome_tools) (1.16.2)\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.12/dist-packages (from epitome_tools) (2.0.2)\n",
            "Requirement already satisfied: nvidia-nccl-cu12 in /usr/local/lib/python3.12/dist-packages (from xgboost->epitome_tools) (2.27.3)\n",
            "Building wheels for collected packages: epitome_tools\n",
            "  Building wheel for epitome_tools (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for epitome_tools: filename=epitome_tools-0.0.17-py3-none-any.whl size=36571017 sha256=5d174def22806a0a248b61cc3974322b8f68cd56dd924a6dac362f4daa0ac92d\n",
            "  Stored in directory: /root/.cache/pip/wheels/63/6c/d9/91ee7ea0bf87db74f371a01a638704eab28edecab589d5ae9c\n",
            "Successfully built epitome_tools\n",
            "Installing collected packages: epitome_tools\n",
            "Successfully installed epitome_tools-0.0.17\n"
          ]
        }
      ],
      "source": [
        "!pip install kb-python -q\n",
        "!pip install ffq -q\n",
        "!apt-get install jq -q\n",
        "!apt-get install -y sra-toolkit\n",
        "!pip install epitome_tools"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "alWL-ethDIQx",
        "outputId": "9507f358-ee43-44a0-a7a5-1a964cc2c02b"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Requirement already satisfied: epitome_tools in /usr/local/lib/python3.12/dist-packages (0.0.17)\n",
            "Requirement already satisfied: xgboost in /usr/local/lib/python3.12/dist-packages (from epitome_tools) (3.0.5)\n",
            "Requirement already satisfied: scipy in /usr/local/lib/python3.12/dist-packages (from epitome_tools) (1.16.2)\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.12/dist-packages (from epitome_tools) (2.0.2)\n",
            "Requirement already satisfied: nvidia-nccl-cu12 in /usr/local/lib/python3.12/dist-packages (from xgboost->epitome_tools) (2.27.3)\n"
          ]
        }
      ],
      "source": [
        "!pip install epitome_tools --upgrade"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "XCUkwecJLgcx"
      },
      "source": [
        "# Installing bamtofastq from 10X - for cases where only the bam file is accessible"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "F1wjAVxKscgc",
        "outputId": "9d816855-5b4e-4208-83df-36ff8f4a8610"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "--2025-10-17 11:40:03--  https://github.com/10XGenomics/bamtofastq/releases/download/v1.4.1/bamtofastq_linux\n",
            "Resolving github.com (github.com)... 20.27.177.113\n",
            "Connecting to github.com (github.com)|20.27.177.113|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://release-assets.githubusercontent.com/github-production-release-asset/112696910/68945361-1bfe-475a-9b64-d1a39a48e48a?sp=r&sv=2018-11-09&sr=b&spr=https&se=2025-10-17T12%3A22%3A51Z&rscd=attachment%3B+filename%3Dbamtofastq_linux&rsct=application%2Foctet-stream&skoid=96c2d410-5711-43a1-aedd-ab1947aa7ab0&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skt=2025-10-17T11%3A22%3A28Z&ske=2025-10-17T12%3A22%3A51Z&sks=b&skv=2018-11-09&sig=XhNnazRXJD8bUUwEgNGxBdJE0eC0H4hY0oEp%2F1PXPQw%3D&jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmVsZWFzZS1hc3NldHMuZ2l0aHVidXNlcmNvbnRlbnQuY29tIiwia2V5Ijoia2V5MSIsImV4cCI6MTc2MDcwMzAwMywibmJmIjoxNzYwNzAxMjAzLCJwYXRoIjoicmVsZWFzZWFzc2V0cHJvZHVjdGlvbi5ibG9iLmNvcmUud2luZG93cy5uZXQifQ.ygI7gYxN4eshfRj1eNV6rG_wGinxFMxn4mX91GdnaHY&response-content-disposition=attachment%3B%20filename%3Dbamtofastq_linux&response-content-type=application%2Foctet-stream [following]\n",
            "--2025-10-17 11:40:04--  https://release-assets.githubusercontent.com/github-production-release-asset/112696910/68945361-1bfe-475a-9b64-d1a39a48e48a?sp=r&sv=2018-11-09&sr=b&spr=https&se=2025-10-17T12%3A22%3A51Z&rscd=attachment%3B+filename%3Dbamtofastq_linux&rsct=application%2Foctet-stream&skoid=96c2d410-5711-43a1-aedd-ab1947aa7ab0&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skt=2025-10-17T11%3A22%3A28Z&ske=2025-10-17T12%3A22%3A51Z&sks=b&skv=2018-11-09&sig=XhNnazRXJD8bUUwEgNGxBdJE0eC0H4hY0oEp%2F1PXPQw%3D&jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmVsZWFzZS1hc3NldHMuZ2l0aHVidXNlcmNvbnRlbnQuY29tIiwia2V5Ijoia2V5MSIsImV4cCI6MTc2MDcwMzAwMywibmJmIjoxNzYwNzAxMjAzLCJwYXRoIjoicmVsZWFzZWFzc2V0cHJvZHVjdGlvbi5ibG9iLmNvcmUud2luZG93cy5uZXQifQ.ygI7gYxN4eshfRj1eNV6rG_wGinxFMxn4mX91GdnaHY&response-content-disposition=attachment%3B%20filename%3Dbamtofastq_linux&response-content-type=application%2Foctet-stream\n",
            "Resolving release-assets.githubusercontent.com (release-assets.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...\n",
            "Connecting to release-assets.githubusercontent.com (release-assets.githubusercontent.com)|185.199.108.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 18540256 (18M) [application/octet-stream]\n",
            "Saving to: ‘bamtofastq’\n",
            "\n",
            "bamtofastq          100%[===================>]  17.68M  13.7MB/s    in 1.3s    \n",
            "\n",
            "2025-10-17 11:40:05 (13.7 MB/s) - ‘bamtofastq’ saved [18540256/18540256]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://github.com/10XGenomics/bamtofastq/releases/download/v1.4.1/bamtofastq_linux -O bamtofastq\n",
        "!chmod +x bamtofastq\n",
        "!mv bamtofastq /usr/local/bin/"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "l_KC14dalOvU",
        "outputId": "a85db0f3-c7cd-4069-bea8-cd8b6b5a394a"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "-- The C compiler identification is GNU 11.4.0\n",
            "-- The CXX compiler identification is GNU 11.4.0\n",
            "-- Detecting C compiler ABI info\n",
            "-- Detecting C compiler ABI info - done\n",
            "-- Check for working C compiler: /usr/bin/cc - skipped\n",
            "-- Detecting C compile features\n",
            "-- Detecting C compile features - done\n",
            "-- Detecting CXX compiler ABI info\n",
            "-- Detecting CXX compiler ABI info - done\n",
            "-- Check for working CXX compiler: /usr/bin/c++ - skipped\n",
            "-- Detecting CXX compile features\n",
            "-- Detecting CXX compile features - done\n",
            "-- Performing Test COMPILER_SUPPORTS_CXX17\n",
            "-- Performing Test COMPILER_SUPPORTS_CXX17 - Success\n",
            "-- Performing Test CMAKE_HAVE_LIBC_PTHREAD\n",
            "-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success\n",
            "-- Found Threads: TRUE\n",
            "-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found version \"1.2.11\")\n",
            "-- Configuring done (1.3s)\n",
            "-- Generating done (0.0s)\n",
            "-- Build files have been written to: /content/kallisto/build\n",
            "[  3%] Creating directories for 'bifrost'\n",
            "[  7%] No download step for 'bifrost'\n",
            "[ 11%] No update step for 'bifrost'\n",
            "[ 15%] No patch step for 'bifrost'\n",
            "[ 19%] Performing configure step for 'bifrost'\n",
            "-- The C compiler identification is GNU 11.4.0\n",
            "-- The CXX compiler identification is GNU 11.4.0\n",
            "-- Detecting C compiler ABI info\n",
            "-- Detecting C compiler ABI info - done\n",
            "-- Check for working C compiler: /usr/bin/cc - skipped\n",
            "-- Detecting C compile features\n",
            "-- Detecting C compile features - done\n",
            "-- Detecting CXX compiler ABI info\n",
            "-- Detecting CXX compiler ABI info - done\n",
            "-- Check for working CXX compiler: /usr/bin/c++ - skipped\n",
            "-- Detecting CXX compile features\n",
            "-- Detecting CXX compile features - done\n",
            "-- Performing Test CMAKE_HAVE_LIBC_PTHREAD\n",
            "-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success\n",
            "-- Found Threads: TRUE\n",
            "-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found version \"1.2.11\")\n",
            "-- Configuring done (0.4s)\n",
            "-- Generating done (0.0s)\n",
            "-- Build files have been written to: /content/kallisto/ext/bifrost/build\n",
            "[ 23%] Performing build step for 'bifrost'\n",
            "[  3%] Building CXX object src/CMakeFiles/bifrost_static.dir/Bifrost.cpp.o\n",
            "[  6%] Building CXX object src/CMakeFiles/bifrost_static.dir/BitContainer.cpp.o\n",
            "[  9%] Building CXX object src/CMakeFiles/bifrost_static.dir/BlockedBloomFilter.cpp.o\n",
            "[ 12%] Building CXX object src/CMakeFiles/bifrost_static.dir/ColorSet.cpp.o\n",
            "[ 15%] Building CXX object src/CMakeFiles/bifrost_static.dir/CompressedCoverage.cpp.o\n",
            "[ 18%] Building CXX object src/CMakeFiles/bifrost_static.dir/CompressedSequence.cpp.o\n",
            "[ 21%] Building CXX object src/CMakeFiles/bifrost_static.dir/FASTX_Parser.cpp.o\n",
            "[ 25%] Building CXX object src/CMakeFiles/bifrost_static.dir/GFA_Parser.cpp.o\n",
            "[ 28%] Building CXX object src/CMakeFiles/bifrost_static.dir/Kmer.cpp.o\n",
            "[ 31%] Building CXX object src/CMakeFiles/bifrost_static.dir/KmerIterator.cpp.o\n",
            "[ 34%] Building CXX object src/CMakeFiles/bifrost_static.dir/MinimizerIndex.cpp.o\n",
            "[ 37%] Building CXX object src/CMakeFiles/bifrost_static.dir/TinyBitmap.cpp.o\n",
            "[ 40%] Building CXX object src/CMakeFiles/bifrost_static.dir/UnitigMap.cpp.o\n",
            "[ 43%] Building C object src/CMakeFiles/bifrost_static.dir/roaring.c.o\n",
            "[ 46%] Linking CXX static library libbifrost.a\n",
            "[ 46%] Built target bifrost_static\n",
            "[ 50%] Building CXX object src/CMakeFiles/bifrost_dynamic.dir/Bifrost.cpp.o\n",
            "[ 53%] Building CXX object src/CMakeFiles/bifrost_dynamic.dir/BitContainer.cpp.o\n",
            "[ 56%] Building CXX object src/CMakeFiles/bifrost_dynamic.dir/BlockedBloomFilter.cpp.o\n",
            "[ 59%] Building CXX object src/CMakeFiles/bifrost_dynamic.dir/ColorSet.cpp.o\n",
            "[ 62%] Building CXX object src/CMakeFiles/bifrost_dynamic.dir/CompressedCoverage.cpp.o\n",
            "[ 65%] Building CXX object src/CMakeFiles/bifrost_dynamic.dir/CompressedSequence.cpp.o\n",
            "[ 68%] Building CXX object src/CMakeFiles/bifrost_dynamic.dir/FASTX_Parser.cpp.o\n",
            "[ 71%] Building CXX object src/CMakeFiles/bifrost_dynamic.dir/GFA_Parser.cpp.o\n",
            "[ 75%] Building CXX object src/CMakeFiles/bifrost_dynamic.dir/Kmer.cpp.o\n",
            "[ 78%] Building CXX object src/CMakeFiles/bifrost_dynamic.dir/KmerIterator.cpp.o\n",
            "[ 81%] Building CXX object src/CMakeFiles/bifrost_dynamic.dir/MinimizerIndex.cpp.o\n",
            "[ 84%] Building CXX object src/CMakeFiles/bifrost_dynamic.dir/TinyBitmap.cpp.o\n",
            "[ 87%] Building CXX object src/CMakeFiles/bifrost_dynamic.dir/UnitigMap.cpp.o\n",
            "[ 90%] Building C object src/CMakeFiles/bifrost_dynamic.dir/roaring.c.o\n",
            "[ 93%] Linking CXX shared library libbifrost.so\n",
            "[ 93%] Built target bifrost_dynamic\n",
            "[ 96%] Building CXX object src/CMakeFiles/Bifrost.dir/Bifrost.cpp.o\n",
            "[100%] Linking CXX executable Bifrost\n",
            "[100%] Built target Bifrost\n",
            "[ 26%] No install step for 'bifrost'\n",
            "[ 30%] Completed 'bifrost'\n",
            "[ 30%] Built target bifrost\n",
            "[ 34%] Building CXX object src/CMakeFiles/kallisto_core.dir/BUSData.cpp.o\n",
            "[ 38%] Building CXX object src/CMakeFiles/kallisto_core.dir/BUSTools.cpp.o\n",
            "[ 42%] Building CXX object src/CMakeFiles/kallisto_core.dir/Bootstrap.cpp.o\n",
            "[ 46%] Building CXX object src/CMakeFiles/kallisto_core.dir/GeneModel.cpp.o\n",
            "[ 50%] Building CXX object src/CMakeFiles/kallisto_core.dir/H5Writer.cpp.o\n",
            "[ 53%] Building CXX object src/CMakeFiles/kallisto_core.dir/KmerIndex.cpp.o\n",
            "[ 57%] Building CXX object src/CMakeFiles/kallisto_core.dir/MinCollector.cpp.o\n",
            "[ 61%] Building CXX object src/CMakeFiles/kallisto_core.dir/PlaintextWriter.cpp.o\n",
            "[ 65%] Building CXX object src/CMakeFiles/kallisto_core.dir/ProcessReads.cpp.o\n",
            "[ 69%] Building CXX object src/CMakeFiles/kallisto_core.dir/PseudoBam.cpp.o\n",
            "[ 73%] Building CXX object src/CMakeFiles/kallisto_core.dir/common.cpp.o\n",
            "[ 76%] Building CXX object src/CMakeFiles/kallisto_core.dir/h5utils.cpp.o\n",
            "[ 80%] Building CXX object src/CMakeFiles/kallisto_core.dir/hash.cpp.o\n",
            "[ 84%] Building CXX object src/CMakeFiles/kallisto_core.dir/main.cpp.o\n",
            "[ 88%] Building CXX object src/CMakeFiles/kallisto_core.dir/weights.cpp.o\n",
            "[ 92%] Linking CXX static library libkallisto_core.a\n",
            "[ 92%] Built target kallisto_core\n",
            "[ 96%] Building CXX object src/CMakeFiles/kallisto.dir/main.cpp.o\n",
            "[100%] Linking CXX executable kallisto\n",
            "[100%] Built target kallisto\n",
            "[ 30%] Built target bifrost\n",
            "[ 92%] Built target kallisto_core\n",
            "[100%] Built target kallisto\n",
            "Install the project...\n",
            "-- Install configuration: \"Release\"\n",
            "-- Installing: /usr/local/bin/kallisto\n"
          ]
        },
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "Cloning into 'kallisto'...\n",
            "CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):\n",
            "  Compatibility with CMake < 3.10 will be removed from a future version of\n",
            "  CMake.\n",
            "\n",
            "  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax\n",
            "  to tell CMake that the project requires at least <min> but has been updated\n",
            "  to work with policies introduced by <max> or earlier.\n",
            "\n",
            "\n",
            "shared build\n",
            "CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):\n",
            "  Compatibility with CMake < 3.10 will be removed from a future version of\n",
            "  CMake.\n",
            "\n",
            "  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax\n",
            "  to tell CMake that the project requires at least <min> but has been updated\n",
            "  to work with policies introduced by <max> or earlier.\n",
            "\n",
            "\n",
            "Compilation architecture: native\n",
            "Build type: Release\n",
            "Maximum k-mer size: 31\n",
            "Maximum g-mer size: 31\n"
          ]
        }
      ],
      "source": [
        "%%bash\n",
        "git clone https://github.com/pachterlab/kallisto && \\\n",
        "cd kallisto && \\\n",
        "mkdir build && \\\n",
        "cd build && \\\n",
        "cmake .. && \\\n",
        "make && \\\n",
        "make install"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "J35MRCeVLmJu"
      },
      "source": [
        "# Mounting Drive onto colab"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "GMPOU0JAUaaH",
        "outputId": "c90dac0a-b366-4762-cd2d-cd83fd4bfcc5"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Mounted at /content/drive\n"
          ]
        }
      ],
      "source": [
        "#mount drive\n",
        "from google.colab import drive\n",
        "drive.mount('/content/drive')"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "lIfhIDaNLpKg"
      },
      "source": [
        "# Load the dataset table"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 463
        },
        "id": "E7pez07vCi8f",
        "outputId": "c1dd9631-d6c8-4eed-b005-1640feec5668"
      },
      "outputs": [
        {
          "data": {
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe"
            },
            "text/html": [
              "\n",
              "  <div id=\"df-79c51fe4-88c7-4559-b3f8-b01874afcaaf\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>GEO</th>\n",
              "      <th>SRA_ID</th>\n",
              "      <th>Name</th>\n",
              "      <th>Conditions</th>\n",
              "      <th>Normal</th>\n",
              "      <th>Background</th>\n",
              "      <th>Sorted</th>\n",
              "      <th>Author</th>\n",
              "      <th>DOI</th>\n",
              "      <th>Age</th>\n",
              "      <th>...</th>\n",
              "      <th>filtering_junk</th>\n",
              "      <th>median_cellassign_prob</th>\n",
              "      <th>most_recent_workflow</th>\n",
              "      <th>n_cells</th>\n",
              "      <th>processed_atac</th>\n",
              "      <th>passed_qc_tcc</th>\n",
              "      <th>tcc_pseudobulked</th>\n",
              "      <th>Core</th>\n",
              "      <th>Notes 1</th>\n",
              "      <th>Notes 2</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>159</th>\n",
              "      <td>GSM7663100</td>\n",
              "      <td>SRX21168594</td>\n",
              "      <td>Pituitary, aged, sample1, scRNA-seq</td>\n",
              "      <td>normal</td>\n",
              "      <td>1</td>\n",
              "      <td>C57BL/6</td>\n",
              "      <td>0</td>\n",
              "      <td>Wei et al. (2025)</td>\n",
              "      <td>NaN</td>\n",
              "      <td>24 months</td>\n",
              "      <td>...</td>\n",
              "      <td>1.0</td>\n",
              "      <td>1.0</td>\n",
              "      <td>v_0.01</td>\n",
              "      <td>7710.0</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>1.0</td>\n",
              "      <td>missing sex</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>160</th>\n",
              "      <td>GSM7663101</td>\n",
              "      <td>SRX21168595</td>\n",
              "      <td>Pituitary, young, sample2, scRNA-seq</td>\n",
              "      <td>normal</td>\n",
              "      <td>1</td>\n",
              "      <td>C57BL/6</td>\n",
              "      <td>0</td>\n",
              "      <td>Wei et al. (2025)</td>\n",
              "      <td>NaN</td>\n",
              "      <td>6 months</td>\n",
              "      <td>...</td>\n",
              "      <td>1.0</td>\n",
              "      <td>1.0</td>\n",
              "      <td>v_0.01</td>\n",
              "      <td>7736.0</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>1.0</td>\n",
              "      <td>missing sex</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>161</th>\n",
              "      <td>GSM7663102</td>\n",
              "      <td>SRX21168596</td>\n",
              "      <td>Pituitary, aged, sample2, scRNA-seq</td>\n",
              "      <td>normal</td>\n",
              "      <td>1</td>\n",
              "      <td>C57BL/6</td>\n",
              "      <td>0</td>\n",
              "      <td>Wei et al. (2025)</td>\n",
              "      <td>NaN</td>\n",
              "      <td>24 months</td>\n",
              "      <td>...</td>\n",
              "      <td>1.0</td>\n",
              "      <td>1.0</td>\n",
              "      <td>v_0.01</td>\n",
              "      <td>6507.0</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>1.0</td>\n",
              "      <td>missing sex</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>162</th>\n",
              "      <td>MMP3-CONT1</td>\n",
              "      <td>MMP3-CONT1</td>\n",
              "      <td>Pituitary_ctrl</td>\n",
              "      <td>normal</td>\n",
              "      <td>1</td>\n",
              "      <td>C57BL/6</td>\n",
              "      <td>0</td>\n",
              "      <td>Kover et al. (2025)</td>\n",
              "      <td>NaN</td>\n",
              "      <td>4 days</td>\n",
              "      <td>...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>v_0.01</td>\n",
              "      <td>2451.0</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>163</th>\n",
              "      <td>sox2egfp_14m</td>\n",
              "      <td>sox2egfp_14m</td>\n",
              "      <td>sox2_egfp</td>\n",
              "      <td>sorted</td>\n",
              "      <td>1</td>\n",
              "      <td>NaN</td>\n",
              "      <td>1</td>\n",
              "      <td>Andoniadou lab</td>\n",
              "      <td>NaN</td>\n",
              "      <td>420</td>\n",
              "      <td>...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>5 rows × 41 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-79c51fe4-88c7-4559-b3f8-b01874afcaaf')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-79c51fe4-88c7-4559-b3f8-b01874afcaaf button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-79c51fe4-88c7-4559-b3f8-b01874afcaaf');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "    <div id=\"df-34028ccc-7c7a-4fef-a36b-de9362a87e8a\">\n",
              "      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-34028ccc-7c7a-4fef-a36b-de9362a87e8a')\"\n",
              "                title=\"Suggest charts\"\n",
              "                style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "      </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "      <script>\n",
              "        async function quickchart(key) {\n",
              "          const quickchartButtonEl =\n",
              "            document.querySelector('#' + key + ' button');\n",
              "          quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "          quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "          try {\n",
              "            const charts = await google.colab.kernel.invokeFunction(\n",
              "                'suggestCharts', [key], {});\n",
              "          } catch (error) {\n",
              "            console.error('Error during call to suggestCharts:', error);\n",
              "          }\n",
              "          quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "          quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "        }\n",
              "        (() => {\n",
              "          let quickchartButtonEl =\n",
              "            document.querySelector('#df-34028ccc-7c7a-4fef-a36b-de9362a87e8a button');\n",
              "          quickchartButtonEl.style.display =\n",
              "            google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "        })();\n",
              "      </script>\n",
              "    </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "text/plain": [
              "              GEO        SRA_ID                                  Name  \\\n",
              "159    GSM7663100   SRX21168594   Pituitary, aged, sample1, scRNA-seq   \n",
              "160    GSM7663101   SRX21168595  Pituitary, young, sample2, scRNA-seq   \n",
              "161    GSM7663102   SRX21168596   Pituitary, aged, sample2, scRNA-seq   \n",
              "162    MMP3-CONT1    MMP3-CONT1                        Pituitary_ctrl   \n",
              "163  sox2egfp_14m  sox2egfp_14m                             sox2_egfp   \n",
              "\n",
              "    Conditions  Normal Background  Sorted               Author  DOI  \\\n",
              "159     normal       1    C57BL/6       0    Wei et al. (2025)  NaN   \n",
              "160     normal       1    C57BL/6       0    Wei et al. (2025)  NaN   \n",
              "161     normal       1    C57BL/6       0    Wei et al. (2025)  NaN   \n",
              "162     normal       1    C57BL/6       0  Kover et al. (2025)  NaN   \n",
              "163     sorted       1        NaN       1       Andoniadou lab  NaN   \n",
              "\n",
              "           Age  ...  filtering_junk median_cellassign_prob  \\\n",
              "159  24 months  ...             1.0                    1.0   \n",
              "160   6 months  ...             1.0                    1.0   \n",
              "161  24 months  ...             1.0                    1.0   \n",
              "162     4 days  ...             NaN                    NaN   \n",
              "163        420  ...             NaN                    NaN   \n",
              "\n",
              "     most_recent_workflow n_cells  processed_atac passed_qc_tcc  \\\n",
              "159                v_0.01  7710.0             NaN           NaN   \n",
              "160                v_0.01  7736.0             NaN           NaN   \n",
              "161                v_0.01  6507.0             NaN           NaN   \n",
              "162                v_0.01  2451.0             NaN           NaN   \n",
              "163                   NaN     NaN             NaN           NaN   \n",
              "\n",
              "    tcc_pseudobulked Core      Notes 1 Notes 2  \n",
              "159              NaN  1.0  missing sex     NaN  \n",
              "160              NaN  1.0  missing sex     NaN  \n",
              "161              NaN  1.0  missing sex     NaN  \n",
              "162              NaN  NaN          NaN     NaN  \n",
              "163              NaN  NaN          NaN     NaN  \n",
              "\n",
              "[5 rows x 41 columns]"
            ]
          },
          "execution_count": 1,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "import pandas as pd\n",
        "df = pd.read_excel('/content/drive/MyDrive/pituitary_atlas/source_table/pituitary_atlas.xlsx')\n",
        "#only keep sn or sc for now\n",
        "df = df[df['Modality'].isin(['sc','sn','multi_rna'])]\n",
        "#keep where 10X version is not PARSE_WT_MegaV2\n",
        "df = df[df['10X version'] != 'PARSE_WT_MegaV2']\n",
        "df = df.reset_index(drop=True)\n",
        "df.tail()"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "LKe_tUQ9Lr2m"
      },
      "source": [
        "\n",
        "# Download all SRA files"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 315
        },
        "id": "wtOqbiXOCBa-",
        "outputId": "04770c95-f11f-4284-d678-e0fb1401a0b7"
      },
      "outputs": [
        {
          "output_type": "error",
          "ename": "ModuleNotFoundError",
          "evalue": "No module named 'epitome_tools'",
          "traceback": [
            "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
            "\u001b[0;31mModuleNotFoundError\u001b[0m                       Traceback (most recent call last)",
            "\u001b[0;32m/tmp/ipython-input-306440136.py\u001b[0m in \u001b[0;36m<cell line: 0>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0;32mimport\u001b[0m \u001b[0mepitome_tools\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0matlas\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0matlas\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
            "\u001b[0;31mModuleNotFoundError\u001b[0m: No module named 'epitome_tools'",
            "",
            "\u001b[0;31m---------------------------------------------------------------------------\u001b[0;32m\nNOTE: If your import is failing due to a missing package, you can\nmanually install dependencies using either !pip or !apt.\n\nTo view examples of installing some common dependencies, click the\n\"Open Examples\" button below.\n\u001b[0;31m---------------------------------------------------------------------------\u001b[0m\n"
          ],
          "errorDetails": {
            "actions": [
              {
                "action": "open_url",
                "actionText": "Open Examples",
                "url": "/notebooks/snippets/importing_libraries.ipynb"
              }
            ]
          }
        }
      ],
      "source": [
        "import epitome_tools.atlas as atlas"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "4l4366FZDxcU",
        "outputId": "19bde102-4dfb-4de2-f2e8-bbb5485c5b4f"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "Folder created\n",
            "wget -P /content/drive/MyDrive/pituitary_atlas/raw/SRX19698101 /path/to/target/folder $(ffq --ncbi SRX19698101 | jq -r '.[] | .url' | tr '\n",
            "' ' ')\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n",
            "SRA_ID already downloaded\n"
          ]
        }
      ],
      "source": [
        "atlas.sra_downloader(raw_path='/content/drive/MyDrive/pituitary_atlas/raw/',df=df)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Sqb2FgfiDimx",
        "outputId": "acf7fccf-3664-46cc-a4fc-a3949585d2eb"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "14month_reads_0_1_AAAMCTGM5\n",
            "bamtofastq_S1_L001_R1_001.fastq.gz\n",
            "bamtofastq_S1_L001_R2_001.fastq.gz\n",
            "bamtofastq_S1_L002_R1_002.fastq.gz\n",
            ".ipynb_checkpoints\n",
            "bamtofastq_S1_L003_R1_003.fastq.gz\n",
            "bamtofastq_S1_L002_R2_002.fastq.gz\n",
            "bamtofastq_S1_L003_R2_003.fastq.gz\n",
            "MMP3-CONT-R1_S6_L003_I1_001.fastq.gz\n",
            "MMP3-CONT-R1_S6_L003_I2_001.fastq.gz\n",
            "MMP3-CONT-R1_S6_L003_R1_001.fastq.gz\n",
            "MMP3-CONT-R1_S6_L003_R2_001.fastq.gz\n",
            "MMP3-CONT-R1_S6_L004_I1_001.fastq.gz\n",
            "MMP3-CONT-R1_S6_L004_I2_001.fastq.gz\n",
            "MMP3-CONT-R1_S6_L004_R1_001.fastq.gz\n",
            "MMP3-CONT-R1_S6_L004_R2_001.fastq.gz\n",
            "MMP3-CONT-atac1_S6_L001_I1_001.fastq.gz\n",
            ".ipynb_checkpoints\n",
            "MMP3-CONT-atac1_S6_L001_R1_001.fastq.gz\n",
            "MMP3-CONT-atac1_S6_L001_R2_001.fastq.gz\n",
            "MMP3-CONT-atac1_S6_L001_R3_001.fastq.gz\n",
            "MMP3-CONT-atac1_S6_L002_I1_001.fastq.gz\n",
            "MMP3-CONT-atac1_S6_L002_R1_001.fastq.gz\n",
            "MMP3-CONT-atac1_S6_L002_R2_001.fastq.gz\n",
            "MMP3-CONT-atac1_S6_L002_R3_001.fastq.gz\n",
            "SRR25434582_1.fastq.gz\n",
            "SRR25434582_2.fastq.gz\n",
            "SRR25434583_1.fastq.gz\n",
            "SRR25434583_2.fastq.gz\n",
            "SRR25434584_1.fastq.gz\n",
            "SRR25434584_2.fastq.gz\n",
            "SRR25434585_2.fastq.gz\n",
            "SRR25434585_1.fastq.gz\n",
            "SRR25434586_1.fastq.gz\n",
            "SRR25434586_2.fastq.gz\n",
            "SRR25434587_1.fastq.gz\n",
            "SRR25434587_2.fastq.gz\n",
            "SRR25434588_2.fastq.gz\n",
            "SRR25434588_1.fastq.gz\n",
            "SRR25434589_2.fastq.gz\n",
            "SRR25434589_1.fastq.gz\n",
            "SRR25434590_1.fastq.gz\n",
            "SRR25434590_2.fastq.gz\n",
            "SRR25434591_2.fastq.gz\n",
            "SRR25434591_1.fastq.gz\n",
            "SRR25434592_1.fastq.gz\n",
            "SRR25434592_2.fastq.gz\n",
            "SRR25434593_1.fastq.gz\n",
            "SRR25434593_2.fastq.gz\n",
            "SRR25434594_1.fastq.gz\n",
            "SRR25434594_2.fastq.gz\n",
            "SRR25434595_1.fastq.gz\n",
            "SRR25434595_2.fastq.gz\n",
            "SRR25434596_1.fastq.gz\n",
            "SRR25434596_2.fastq.gz\n",
            "SRR25434597_1.fastq.gz\n",
            "SRR25434597_2.fastq.gz\n",
            "SRR33975503\n",
            "SRR33975503_1.fastq.gz\n",
            "SRR33975503_2.fastq.gz\n",
            "SRR33975504\n",
            "SRR33975504_1.fastq.gz\n",
            "SRR33975504_2.fastq.gz\n",
            "SRR33975505\n",
            "SRR33975505_1.fastq.gz\n",
            "SRR33975505_2.fastq.gz\n",
            "SRR33975506\n",
            "SRR33975506_1.fastq.gz\n",
            "SRR33975506_2.fastq.gz\n",
            "SRR30559062.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30559062.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30559062.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR30559062.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR30559058.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30559058.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30559058.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR30559058.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR30559056.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30559056.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30559056.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR30559056.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR30559059.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30559059.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30559059.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR30559059.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR30559055.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30559055.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30559055.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR30559055.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR30559057.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30559057.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30559057.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR30559057.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR30559060.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30559060.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30559060.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR30559060.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR30559064.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30559064.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30559064.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR30559064.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR30559061.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30559061.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30559061.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR30559061.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR30559063.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30559063.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30559063.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR30559063.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR30559065.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30559065.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30559065.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR30559065.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR30559066.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30559066.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30559066.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR30559066.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR30559068.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30559068.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30559068.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR30559068.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR30559067.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30559067.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30559067.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR30559067.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR30559070.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30559070.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30559070.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR30559070.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR30559069.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30559069.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30559069.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR30559069.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR31330960.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR31330960.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30275225.1_1.fastq.gz\n",
            "SRR30275225.1_2.fastq.gz\n",
            "SRR30275225.1_3.fastq.gz\n",
            "SRR30275225.1_4.fastq.gz\n",
            "SRR30275223.1_1.fastq.gz\n",
            "SRR30275223.1_2.fastq.gz\n",
            "SRR30275223.1_3.fastq.gz\n",
            "SRR30275223.1_4.fastq.gz\n",
            "SRR30275221.1_1.fastq.gz\n",
            "SRR30275221.1_2.fastq.gz\n",
            "SRR30275221.1_3.fastq.gz\n",
            "SRR30275221.1_4.fastq.gz\n",
            "SRR30275224.1_1.fastq.gz\n",
            "SRR30275224.1_2.fastq.gz\n",
            "SRR30275224.1_3.fastq.gz\n",
            "SRR30275224.1_4.fastq.gz\n",
            "SRR30275222.1_1.fastq.gz\n",
            "SRR30275222.1_2.fastq.gz\n",
            "SRR30275222.1_3.fastq.gz\n",
            "SRR30275222.1_4.fastq.gz\n",
            "SRR30275226.1_1.fastq.gz\n",
            "SRR30275226.1_2.fastq.gz\n",
            "SRR30275226.1_3.fastq.gz\n",
            "SRR30275226.1_4.fastq.gz\n",
            "SRR30275229.1_1.fastq.gz\n",
            "SRR30275229.1_2.fastq.gz\n",
            "SRR30275229.1_3.fastq.gz\n",
            "SRR30275229.1_4.fastq.gz\n",
            "SRR30275230.1_1.fastq.gz\n",
            "SRR30275230.1_2.fastq.gz\n",
            "SRR30275230.1_3.fastq.gz\n",
            "SRR30275230.1_4.fastq.gz\n",
            "SRR30275231.1_1.fastq.gz\n",
            "SRR30275231.1_2.fastq.gz\n",
            "SRR30275231.1_3.fastq.gz\n",
            "SRR30275231.1_4.fastq.gz\n",
            "SRR30275232.1_1.fastq.gz\n",
            "SRR30275232.1_2.fastq.gz\n",
            "SRR30275232.1_3.fastq.gz\n",
            "SRR30275232.1_4.fastq.gz\n",
            "SRR30275228.1_1.fastq.gz\n",
            "SRR30275228.1_2.fastq.gz\n",
            "SRR30275228.1_3.fastq.gz\n",
            "SRR30275228.1_4.fastq.gz\n",
            "SRR30275227.1_1.fastq.gz\n",
            "SRR30275227.1_2.fastq.gz\n",
            "SRR30275227.1_3.fastq.gz\n",
            "SRR30275227.1_4.fastq.gz\n",
            "SRR30275235.1_1.fastq.gz\n",
            "SRR30275235.1_2.fastq.gz\n",
            "SRR30275235.1_3.fastq.gz\n",
            "SRR30275235.1_4.fastq.gz\n",
            "SRR30275233.1_1.fastq.gz\n",
            "SRR30275233.1_2.fastq.gz\n",
            "SRR30275233.1_3.fastq.gz\n",
            "SRR30275233.1_4.fastq.gz\n",
            "SRR30275237.1_1.fastq.gz\n",
            "SRR30275237.1_2.fastq.gz\n",
            "SRR30275237.1_3.fastq.gz\n",
            "SRR30275237.1_4.fastq.gz\n",
            "SRR30275234.1_1.fastq.gz\n",
            "SRR30275234.1_2.fastq.gz\n",
            "SRR30275234.1_3.fastq.gz\n",
            "SRR30275234.1_4.fastq.gz\n",
            "SRR30275238.1_1.fastq.gz\n",
            "SRR30275238.1_2.fastq.gz\n",
            "SRR30275238.1_3.fastq.gz\n",
            "SRR30275238.1_4.fastq.gz\n",
            "SRR30275236.1_1.fastq.gz\n",
            "SRR30275236.1_2.fastq.gz\n",
            "SRR30275236.1_3.fastq.gz\n",
            "SRR30275236.1_4.fastq.gz\n",
            "SRR30275239.1_1.fastq.gz\n",
            "SRR30275239.1_2.fastq.gz\n",
            "SRR30275239.1_3.fastq.gz\n",
            "SRR30275239.1_4.fastq.gz\n",
            "SRR30275240.1_1.fastq.gz\n",
            "SRR30275240.1_2.fastq.gz\n",
            "SRR30275240.1_3.fastq.gz\n",
            "SRR30275240.1_4.fastq.gz\n",
            "SRR30275241.1_1.fastq.gz\n",
            "SRR30275241.1_2.fastq.gz\n",
            "SRR30275241.1_3.fastq.gz\n",
            "SRR30275241.1_4.fastq.gz\n",
            "SRR30275244.1_1.fastq.gz\n",
            "SRR30275244.1_2.fastq.gz\n",
            "SRR30275244.1_3.fastq.gz\n",
            "SRR30275244.1_4.fastq.gz\n",
            "SRR30275243.1_1.fastq.gz\n",
            "SRR30275243.1_2.fastq.gz\n",
            "SRR30275243.1_3.fastq.gz\n",
            "SRR30275243.1_4.fastq.gz\n",
            "SRR30275242.1_1.fastq.gz\n",
            "SRR30275242.1_2.fastq.gz\n",
            "SRR30275242.1_3.fastq.gz\n",
            "SRR30275242.1_4.fastq.gz\n",
            "SRR30275246.1_1.fastq.gz\n",
            "SRR30275246.1_2.fastq.gz\n",
            "SRR30275246.1_3.fastq.gz\n",
            "SRR30275246.1_4.fastq.gz\n",
            "SRR30275245.1_1.fastq.gz\n",
            "SRR30275245.1_2.fastq.gz\n",
            "SRR30275245.1_3.fastq.gz\n",
            "SRR30275245.1_4.fastq.gz\n",
            "SRR30275247.1_1.fastq.gz\n",
            "SRR30275247.1_2.fastq.gz\n",
            "SRR30275247.1_3.fastq.gz\n",
            "SRR30275247.1_4.fastq.gz\n",
            "SRR30275249.1_1.fastq.gz\n",
            "SRR30275249.1_2.fastq.gz\n",
            "SRR30275249.1_3.fastq.gz\n",
            "SRR30275249.1_4.fastq.gz\n",
            "SRR30275250.1_1.fastq.gz\n",
            "SRR30275250.1_2.fastq.gz\n",
            "SRR30275250.1_3.fastq.gz\n",
            "SRR30275250.1_4.fastq.gz\n",
            "SRR30275248.1_1.fastq.gz\n",
            "SRR30275248.1_2.fastq.gz\n",
            "SRR30275248.1_3.fastq.gz\n",
            "SRR30275248.1_4.fastq.gz\n",
            "SRR30275254.1_1.fastq.gz\n",
            "SRR30275254.1_2.fastq.gz\n",
            "SRR30275254.1_3.fastq.gz\n",
            "SRR30275254.1_4.fastq.gz\n",
            "SRR30275251.1_1.fastq.gz\n",
            "SRR30275251.1_2.fastq.gz\n",
            "SRR30275251.1_3.fastq.gz\n",
            "SRR30275251.1_4.fastq.gz\n",
            "SRR30275252.1_1.fastq.gz\n",
            "SRR30275252.1_2.fastq.gz\n",
            "SRR30275252.1_3.fastq.gz\n",
            "SRR30275252.1_4.fastq.gz\n",
            "SRR30275255.1_1.fastq.gz\n",
            "SRR30275255.1_2.fastq.gz\n",
            "SRR30275255.1_3.fastq.gz\n",
            "SRR30275255.1_4.fastq.gz\n",
            "SRR30275256.1_1.fastq.gz\n",
            "SRR30275256.1_2.fastq.gz\n",
            "SRR30275256.1_3.fastq.gz\n",
            "SRR30275256.1_4.fastq.gz\n",
            "SRR30275253.1_1.fastq.gz\n",
            "SRR30275253.1_2.fastq.gz\n",
            "SRR30275253.1_3.fastq.gz\n",
            "SRR30275253.1_4.fastq.gz\n",
            "SRR30275257.1_1.fastq.gz\n",
            "SRR30275257.1_2.fastq.gz\n",
            "SRR30275257.1_3.fastq.gz\n",
            "SRR30275257.1_4.fastq.gz\n",
            "SRR30275259.1_1.fastq.gz\n",
            "SRR30275259.1_2.fastq.gz\n",
            "SRR30275259.1_3.fastq.gz\n",
            "SRR30275259.1_4.fastq.gz\n",
            "SRR30275258.1_1.fastq.gz\n",
            "SRR30275258.1_2.fastq.gz\n",
            "SRR30275258.1_3.fastq.gz\n",
            "SRR30275258.1_4.fastq.gz\n",
            "SRR30275260.1_1.fastq.gz\n",
            "SRR30275260.1_2.fastq.gz\n",
            "SRR30275260.1_3.fastq.gz\n",
            "SRR30275260.1_4.fastq.gz\n",
            "SRR30275261.1_1.fastq.gz\n",
            "SRR30275261.1_2.fastq.gz\n",
            "SRR30275261.1_3.fastq.gz\n",
            "SRR30275261.1_4.fastq.gz\n",
            "SRR30275262.1_1.fastq.gz\n",
            "SRR30275262.1_2.fastq.gz\n",
            "SRR30275262.1_3.fastq.gz\n",
            "SRR30275262.1_4.fastq.gz\n",
            ".ipynb_checkpoints\n",
            "SRR30275263.1_1.fastq.gz\n",
            "SRR30275263.1_2.fastq.gz\n",
            "SRR30275263.1_3.fastq.gz\n",
            "SRR30275263.1_4.fastq.gz\n",
            "SRR30275265.1_1.fastq.gz\n",
            "SRR30275265.1_2.fastq.gz\n",
            "SRR30275265.1_3.fastq.gz\n",
            "SRR30275265.1_4.fastq.gz\n",
            "SRR30275266.1_1.fastq.gz\n",
            "SRR30275266.1_2.fastq.gz\n",
            "SRR30275266.1_3.fastq.gz\n",
            "SRR30275266.1_4.fastq.gz\n",
            "SRR30275267.1_1.fastq.gz\n",
            "SRR30275267.1_2.fastq.gz\n",
            "SRR30275267.1_3.fastq.gz\n",
            "SRR30275267.1_4.fastq.gz\n",
            "SRR30275264.1_1.fastq.gz\n",
            "SRR30275264.1_2.fastq.gz\n",
            "SRR30275264.1_3.fastq.gz\n",
            "SRR30275264.1_4.fastq.gz\n",
            "SRR30275274.1_1.fastq.gz\n",
            "SRR30275274.1_2.fastq.gz\n",
            "SRR30275274.1_3.fastq.gz\n",
            "SRR30275274.1_4.fastq.gz\n",
            "SRR30275268.1_1.fastq.gz\n",
            "SRR30275268.1_2.fastq.gz\n",
            "SRR30275268.1_3.fastq.gz\n",
            "SRR30275268.1_4.fastq.gz\n",
            "SRR30275270.1_1.fastq.gz\n",
            "SRR30275270.1_2.fastq.gz\n",
            "SRR30275270.1_3.fastq.gz\n",
            "SRR30275270.1_4.fastq.gz\n",
            "SRR30275272.1_1.fastq.gz\n",
            "SRR30275272.1_2.fastq.gz\n",
            "SRR30275272.1_3.fastq.gz\n",
            "SRR30275272.1_4.fastq.gz\n",
            "SRR30275271.1_1.fastq.gz\n",
            "SRR30275271.1_2.fastq.gz\n",
            "SRR30275271.1_3.fastq.gz\n",
            "SRR30275271.1_4.fastq.gz\n",
            "SRR30275269.1_1.fastq.gz\n",
            "SRR30275269.1_2.fastq.gz\n",
            "SRR30275269.1_3.fastq.gz\n",
            "SRR30275269.1_4.fastq.gz\n",
            "SRR30275273.1_1.fastq.gz\n",
            "SRR30275273.1_2.fastq.gz\n",
            "SRR30275273.1_3.fastq.gz\n",
            "SRR30275273.1_4.fastq.gz\n",
            "SRR30275275.1_1.fastq.gz\n",
            "SRR30275275.1_2.fastq.gz\n",
            "SRR30275275.1_3.fastq.gz\n",
            "SRR30275275.1_4.fastq.gz\n",
            "SRR30275278.1_1.fastq.gz\n",
            "SRR30275278.1_2.fastq.gz\n",
            "SRR30275278.1_3.fastq.gz\n",
            "SRR30275278.1_4.fastq.gz\n",
            "SRR30275277.1_1.fastq.gz\n",
            "SRR30275277.1_2.fastq.gz\n",
            "SRR30275277.1_3.fastq.gz\n",
            "SRR30275277.1_4.fastq.gz\n",
            "SRR30275276.1_1.fastq.gz\n",
            "SRR30275276.1_2.fastq.gz\n",
            "SRR30275276.1_3.fastq.gz\n",
            "SRR30275276.1_4.fastq.gz\n",
            "SRR30275279.1_1.fastq.gz\n",
            "SRR30275279.1_2.fastq.gz\n",
            "SRR30275279.1_3.fastq.gz\n",
            "SRR30275279.1_4.fastq.gz\n",
            "SRR30275280.1_1.fastq.gz\n",
            "SRR30275280.1_2.fastq.gz\n",
            "SRR30275280.1_3.fastq.gz\n",
            "SRR30275280.1_4.fastq.gz\n",
            "SRR30275281.1_1.fastq.gz\n",
            "SRR30275281.1_2.fastq.gz\n",
            "SRR30275281.1_3.fastq.gz\n",
            "SRR30275281.1_4.fastq.gz\n",
            "SRR30275282.1_1.fastq.gz\n",
            "SRR30275282.1_2.fastq.gz\n",
            "SRR30275282.1_3.fastq.gz\n",
            "SRR30275282.1_4.fastq.gz\n",
            "SRR30275283.1_1.fastq.gz\n",
            "SRR30275283.1_2.fastq.gz\n",
            "SRR30275283.1_3.fastq.gz\n",
            "SRR30275283.1_4.fastq.gz\n",
            "SRR30275285.1_1.fastq.gz\n",
            "SRR30275285.1_2.fastq.gz\n",
            "SRR30275285.1_3.fastq.gz\n",
            "SRR30275285.1_4.fastq.gz\n",
            "SRR30275284.1_1.fastq.gz\n",
            "SRR30275284.1_2.fastq.gz\n",
            "SRR30275284.1_3.fastq.gz\n",
            "SRR30275284.1_4.fastq.gz\n",
            "SRR30275286.1_1.fastq.gz\n",
            "SRR30275286.1_2.fastq.gz\n",
            "SRR30275286.1_3.fastq.gz\n",
            "SRR30275286.1_4.fastq.gz\n",
            "SRR22756092.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR22756092.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR22756091.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR22756091.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR22756094.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR22756094.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR22756093.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR22756093.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR22756096.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR22756096.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR22756098.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR22756098.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR22756095.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR22756095.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR22756097.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR22756097.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR22756102.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR22756102.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR22756099.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR22756099.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR22756100.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR22756100.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR22756101.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR22756101.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR22756104.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR22756104.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR22756103.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR22756103.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR22756105.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR22756105.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR22756106.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR22756106.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190704.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190704.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190704.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190704.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26190705.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190705.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190705.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190705.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26190706.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190706.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190706.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190706.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26190707.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190707.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190707.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190707.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26190709.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190709.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190709.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190709.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26190708.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190708.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190708.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190708.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26190710.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190710.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190710.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190710.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26190711.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190711.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190711.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190711.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26190712.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190712.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190712.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190712.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26190713.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190713.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190713.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190713.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26190714.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190714.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190714.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190714.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26190715.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190715.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190715.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190715.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26190716.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190716.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190716.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190716.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26190717.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190717.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190717.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190717.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26190732.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190732.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190732.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190732.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26190733.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190733.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190733.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190733.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26190734.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190734.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190734.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190734.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26190735.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190735.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190735.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190735.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26190737.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190737.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190737.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190737.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26190736.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190736.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190736.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190736.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26190739.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190739.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190739.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190739.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26190738.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190738.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190738.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190738.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26190741.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190741.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190741.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190741.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26190740.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190740.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190740.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190740.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26190742.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190742.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190742.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190742.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26190743.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190743.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190743.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190743.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26190744.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190744.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190744.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190744.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26190745.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190745.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190745.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190745.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26190747.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190747.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190747.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190747.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26190746.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190746.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190746.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190746.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26190748.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190748.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190748.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190748.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26190749.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26190749.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26190749.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26190749.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR23886271.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR23886271.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR23886270.lite.1\n",
            "SRR23886269.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR23886269.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30414123.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30414123.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30414123.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR30414121.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30414121.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30414121.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR30414122.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30414122.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30414122.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR30414124.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30414124.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30414124.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR30414125.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30414125.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30414125.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR30414126.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30414126.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30414126.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR30414128.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30414128.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30414128.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR30414127.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30414127.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30414127.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR30414129.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30414129.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30414129.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR30414130.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30414130.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30414130.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR30414131.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30414131.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30414131.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR30414132.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30414132.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30414132.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR25916626.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR25916626.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR25916626.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR25916626.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR25916627.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR25916627.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR25916627.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR25916627.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR26515799.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26515799.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26515800.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26515800.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26515800.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26515801.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26515801.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26515801.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26515802.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26515802.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26515802.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26515803.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26515803.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26515803.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR28551835.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR28551835.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR28551836.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR28551836.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR28551837.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR28551837.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR28551838.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR28551838.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR28551839.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR28551839.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR28551840.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR28551840.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR28551841.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR28551841.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR28551842.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR28551842.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30058744.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30058744.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30058744.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR30058745.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30058745.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30058745.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR30058746.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30058746.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30058746.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR30058747.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR30058747.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR30058747.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR10700818.sralite.1_1.fastq.gz\n",
            "fastq files found\n",
            "fastqs\n",
            "SRR10700816.sralite.1_1.fastq.gz\n",
            "fastq files found\n",
            "fastqs\n",
            "SRR24919233.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR24919233.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR24919233.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR24919233.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR24919232.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR24919232.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR24919232.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR24919232.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR24919234_atac.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR24919234_atac.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR24919234_atac.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR24919234_atac.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR24919235_atac.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR24919235_atac.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR24919235_atac.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR24919235_atac.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR24919235.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR24919235.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR24919235.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR24919235.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR24919234.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR24919234.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR24919234.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR24919234.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR13203971.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "fastqs\n",
            "SRR11952693.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11952693.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11952693.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR11952692.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11952692.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11952692.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR11952691.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11952691.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11952691.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR12074654.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR12074654.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR12074654.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR12074655.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR12074655.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR12074655.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR12074656.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR12074656.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR12074656.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR12074657.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR12074657.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR12074657.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR12074651.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR12074651.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR12074651.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR12074652.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR12074652.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR12074652.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR12074653.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR12074653.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR12074653.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR12074650.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR12074650.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR12074650.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR19529123.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR19529123.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR19529123.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR19529124.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR19529124.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR19529124.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR19529125.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR19529125.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR19529125.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR19529126.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR19529126.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR19529126.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR22247484.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR22247484.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR22247484.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR22247485.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR22247485.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR22247485.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR22247486.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR22247486.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR22247486.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR22247486.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR22247487.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR22247487.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR22247487.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR22247487.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR22247488.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR22247488.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR22247488.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR22247489.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR22247489.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR22247489.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR17102613.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR17102613.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR17102613.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR17102611.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR17102611.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR17102611.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR17102610.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR17102610.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR17102610.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR17102612.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR17102612.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR17102612.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR17102616.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR17102616.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR17102616.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR17102619.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR17102619.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR17102619.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR17102618.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR17102618.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR17102618.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR17102617.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR17102617.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR17102617.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR17102626.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR17102626.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR17102626.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR17102625.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR17102625.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR17102625.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR17102615.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR17102615.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR17102615.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR17102614.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR17102614.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR17102614.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR17102622.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR17102622.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR17102622.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR17102624.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR17102624.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR17102624.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR17102621.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR17102621.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR17102621.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR17102623.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR17102623.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR17102623.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR17102620.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR17102620.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR17102620.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR17102633.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR17102633.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR17102633.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR17102631.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR17102631.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR17102631.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR17102632.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR17102632.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR17102632.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            ".ipynb_checkpoints\n",
            "SRR17102627.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR17102627.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR17102627.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR17102628.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR17102628.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR17102628.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR17102629.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR17102629.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR17102629.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR17102630.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR17102630.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR17102630.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR26276677.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276677.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26276678.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276678.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26276680.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276680.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26276679.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276679.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26276682.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276682.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26276681.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276681.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26276683.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276683.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26276684.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276684.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26276686.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276686.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26276685.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276685.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26276693.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276693.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26276698.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276698.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26276687.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276687.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26276688.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276688.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26276703.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276703.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26276708.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276708.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26276691.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276691.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26276689.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276689.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26276690.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276690.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26276692.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276692.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26276694.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276694.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26276696.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276696.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26276695.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276695.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26276697.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276697.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26276699.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276699.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26276700.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276700.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26276701.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276701.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26276702.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276702.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26276704.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276704.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26276706.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276706.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26276705.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276705.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR26276707.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR26276707.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR25436962.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR25436962.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR25436962.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR25436962.lite.1_4.fastq.gz\n",
            "fastq files found\n",
            "SRR25436964.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR25436964.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR25436963.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR25436963.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR25436966.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR25436966.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR25436965.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR25436965.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR25436968.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR25436968.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR25436967.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR25436967.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR25436969.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR25436969.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR25436970.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR25436970.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "ERR8026762.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR8026762.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "ERR8026762.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "ERR8026763.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR8026763.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "ERR8026763.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "ERR8026761.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR8026761.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "ERR8026761.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "ERR8026760.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR8026760.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "ERR8026760.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "ERR8026758.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR8026758.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "ERR8026758.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "ERR8026759.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR8026759.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "ERR8026759.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "ERR8026757.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR8026757.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "ERR8026757.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "ERR8026756.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR8026756.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "ERR8026756.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "ERR5173697.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173690.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173694.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173696.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173695.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173691.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173692.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173693.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173687.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173682.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173686.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173684.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173685.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173683.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173689.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173688.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173678.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173676.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173674.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173679.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173680.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173681.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173677.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173675.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173672.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173668.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173673.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173670.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173671.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173667.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173669.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173666.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173663.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173662.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173664.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173665.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173661.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173659.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173660.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173658.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173652.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173654.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173657.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173655.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173656.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173653.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            ".ipynb_checkpoints\n",
            "ERR5173650.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173646.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173649.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173648.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173647.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173651.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            ".ipynb_checkpoints\n",
            "ERR5173640.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173643.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173641.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173645.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173642.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "ERR5173644.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            ".ipynb_checkpoints\n",
            "SRR7898909.bam\n",
            "SRR25511315.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR25511315.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            ".ipynb_checkpoints\n",
            "SRR25511316.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR25511316.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            ".ipynb_checkpoints\n",
            "SRR25511317.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR25511317.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR25511318.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR25511318.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR13081818.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR13081818.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR13081817.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR13081817.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11268113.sralite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11268113.sralite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11268112.sralite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11268112.sralite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11268111.sralite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11268111.sralite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11268110.sralite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11268110.sralite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11268109.sralite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11268109.sralite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11268108.sralite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11268108.sralite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11268107.sralite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11268107.sralite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11268106.sralite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11268106.sralite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11268105.sralite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11268105.sralite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11268104.sralite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11268104.sralite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR9951029.sralite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR9951029.sralite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR9951029.sralite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR9951028.sralite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR9951028.sralite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR9951028.sralite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR9951026.sralite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR9951026.sralite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR9951026.sralite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR9951027.sralite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR9951027.sralite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR9951027.sralite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR9951023.sralite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR9951023.sralite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR9951023.sralite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR9951024.sralite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR9951024.sralite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR9951024.sralite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR9951022.sralite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR9951022.sralite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR9951022.sralite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR9951025.sralite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR9951025.sralite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR9951025.sralite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR8487461.sralite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR8487461.sralite.1_2.fastq.gz\n",
            "fastq files found\n",
            ".ipynb_checkpoints\n",
            "SRR7898910.sralite.1_1.fastq.gz\n",
            "fastq files found\n",
            "fastqs\n",
            ".ipynb_checkpoints\n",
            "fastqs\n",
            "SRR7898909.sralite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11945389.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11945389.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11945389.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR11945390.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11945390.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11945390.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR11945387.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11945387.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11945387.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR11945388.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11945388.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11945388.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR11945385.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11945385.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11945385.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR11945386.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11945386.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11945386.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR11945384.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11945384.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11945384.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR11945383.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11945383.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11945383.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR11945382.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11945382.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11945382.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR11945381.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11945381.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11945381.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR11945380.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11945380.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11945380.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR11945379.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11945379.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11945379.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR11945360.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11945360.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11945360.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR11945361.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11945361.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11945361.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR11945358.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11945358.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11945358.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR11945359.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11945359.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11945359.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR11945356.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11945356.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11945356.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR11945357.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11945357.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11945357.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR11945355.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11945355.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11945355.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR11945354.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11945354.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11945354.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR11945353.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11945353.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11945353.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR11945366.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11945366.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11945366.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR11945365.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11945365.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11945365.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR11945364.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11945364.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11945364.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            ".ipynb_checkpoints\n",
            "SRR11945363.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR11945363.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11945363.lite.1_2.fastq.gz\n",
            "fastq files found\n",
            "SRR11945362.lite.1_3.fastq.gz\n",
            "fastq files found\n",
            "SRR11945362.lite.1_1.fastq.gz\n",
            "fastq files found\n",
            "SRR11945362.lite.1_2.fastq.gz\n",
            "fastq files found\n"
          ]
        }
      ],
      "source": [
        "atlas.fastq_downloader(raw_path='/content/drive/MyDrive/pituitary_atlas/raw/',df=df)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "AubfT6-aGahg"
      },
      "outputs": [],
      "source": [
        "run_kb_count_nac(\"/content/drive/MyDrive/pituitary_atlas/processed/nac/\",\"/content/drive/MyDrive/pituitary_atlas/raw/\", df, species=\"mouse\")"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "F0moJbCgDfIX"
      },
      "source": [
        "#Also downloading Multiome onlist for the multiome datasets"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "BJGijdDb-vAq",
        "outputId": "c5fa1ab1-d025-4f79-996b-014bab174ea6"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "--2024-10-16 08:26:15--  https://teichlab.github.io/scg_lib_structs/data/10X-Genomics/gex_737K-arc-v1.txt.gz\n",
            "Resolving teichlab.github.io (teichlab.github.io)... 185.199.111.153, 185.199.110.153, 185.199.108.153, ...\n",
            "Connecting to teichlab.github.io (teichlab.github.io)|185.199.111.153|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 2142553 (2.0M) [application/gzip]\n",
            "Saving to: ‘10xMultiome/gex_737K-arc-v1.txt.gz’\n",
            "\n",
            "\rgex_737K-arc-v1.txt   0%[                    ]       0  --.-KB/s               \rgex_737K-arc-v1.txt 100%[===================>]   2.04M  --.-KB/s    in 0.07s   \n",
            "\n",
            "2024-10-16 08:26:15 (29.4 MB/s) - ‘10xMultiome/gex_737K-arc-v1.txt.gz’ saved [2142553/2142553]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "#donwload multiome onlist\n",
        "!wget -P 10xMultiome/ https://teichlab.github.io/scg_lib_structs/data/10X-Genomics/gex_737K-arc-v1.txt.gz\n",
        "!gunzip 10xMultiome/gex_737K-arc-v1.txt.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "KDOvd0gniNZH"
      },
      "source": [
        "# Run alignment"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "WQJWwEi3iawr",
        "outputId": "c59870ce-2f9d-4945-be76-cba97bfaa944"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "\n",
            "        SRA_ID: SRX8489818\n",
            "        experiment_type: sc\n",
            "        tech: 10XV2\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX8489819\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX8489820\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX8489821\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX8489822\n",
            "        experiment_type: sc\n",
            "        tech: 10XV2\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX8489823\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX8489824\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX8489834\n",
            "        experiment_type: sn\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX8489835\n",
            "        experiment_type: sn\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX8489836\n",
            "        experiment_type: sn\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX8489837\n",
            "        experiment_type: sn\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX8489838\n",
            "        experiment_type: sn\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX8489839\n",
            "        experiment_type: sn\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX8489840\n",
            "        experiment_type: sn\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX8489841\n",
            "        experiment_type: sn\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX8489842\n",
            "        experiment_type: sn\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX4736337\n",
            "        experiment_type: sc\n",
            "        tech: 10XV2\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX4736338\n",
            "        experiment_type: sc\n",
            "        tech: 10XV2\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX5292053\n",
            "        experiment_type: sc\n",
            "        tech: 10XV2\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX6699502\n",
            "        experiment_type: sc\n",
            "        tech: 10XV2\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX6699503\n",
            "        experiment_type: sc\n",
            "        tech: 10XV2\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX7874737\n",
            "        experiment_type: sc\n",
            "        tech: 10XV2\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX7874738\n",
            "        experiment_type: sc\n",
            "        tech: 10XV2\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX7874739\n",
            "        experiment_type: sc\n",
            "        tech: DROPSEQ\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX7874740\n",
            "        experiment_type: sc\n",
            "        tech: DROPSEQ\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX7874741\n",
            "        experiment_type: sc\n",
            "        tech: DROPSEQ\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX7874742\n",
            "        experiment_type: sc\n",
            "        tech: DROPSEQ\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX7874743\n",
            "        experiment_type: sc\n",
            "        tech: DROPSEQ\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX7874744\n",
            "        experiment_type: sc\n",
            "        tech: DROPSEQ\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX7874745\n",
            "        experiment_type: sc\n",
            "        tech: DROPSEQ\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX7874746\n",
            "        experiment_type: sc\n",
            "        tech: DROPSEQ\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX9528564\n",
            "        experiment_type: sc\n",
            "        tech: 10XV2\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX9528565\n",
            "        experiment_type: sc\n",
            "        tech: 10XV2\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21241798\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21241799\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21241800\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21241801\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: ERX4978537\n",
            "        experiment_type: sc\n",
            "        tech: 10XV2\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: ERX4978538\n",
            "        experiment_type: sc\n",
            "        tech: 10XV2\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: ERX4978539\n",
            "        experiment_type: sc\n",
            "        tech: 10XV2\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: ERX4978540\n",
            "        experiment_type: sc\n",
            "        tech: 10XV2\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: ERX4978541\n",
            "        experiment_type: sc\n",
            "        tech: 10XV2\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: ERX4978542\n",
            "        experiment_type: sc\n",
            "        tech: 10XV2\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: ERX4978543\n",
            "        experiment_type: sc\n",
            "        tech: 10XV2\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: ERX4978544\n",
            "        experiment_type: sc\n",
            "        tech: 10XV2\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: ERX7593481\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: ERX7593482\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: ERX7593483\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: ERX7593484\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21170823\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21170824\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21170825\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21986105\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21986106\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21986107\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21986108\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21986109\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21986110\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21986111\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21986112\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21986113\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21986114\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21986115\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21986116\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX13290057\n",
            "        experiment_type: sn\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX13290058\n",
            "        experiment_type: sn\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX13290059\n",
            "        experiment_type: sn\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX13290060\n",
            "        experiment_type: sn\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX13290061\n",
            "        experiment_type: sn\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX13290062\n",
            "        experiment_type: sn\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX18224104\n",
            "        experiment_type: sc\n",
            "        tech: 10XV2\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX18224105\n",
            "        experiment_type: sc\n",
            "        tech: 10XV2\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX18224106\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX18224107\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX18224108\n",
            "        experiment_type: sc\n",
            "        tech: 10XV2\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX18224109\n",
            "        experiment_type: sc\n",
            "        tech: 10XV2\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX15581370\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX15581371\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX8602211\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX8602212\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX8497092\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX8497093\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX8497094\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX9637224\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX20679984\n",
            "        experiment_type: multi_rna\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX7377164\n",
            "        experiment_type: sc\n",
            "        tech: 10XV2\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX7377166\n",
            "        experiment_type: sc\n",
            "        tech: 10XV2\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX25532498\n",
            "        experiment_type: sc\n",
            "        tech: 10XV2\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX25532499\n",
            "        experiment_type: sc\n",
            "        tech: 10XV2\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX24151432\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX24151433\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX24151434\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX24151435\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX24151436\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX24151437\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX24151438\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX24151439\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX22219345\n",
            "        experiment_type: sc\n",
            "        tech: 10XV2\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX22219346\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21636476\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21636477\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX25839820\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX25839821\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX19698102\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX19698101\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX19698100\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21902348\n",
            "        experiment_type: multi_rna\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21902349\n",
            "        experiment_type: multi_rna\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21902350\n",
            "        experiment_type: multi_rna\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21902351\n",
            "        experiment_type: multi_rna\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21902352\n",
            "        experiment_type: multi_rna\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21902353\n",
            "        experiment_type: multi_rna\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21902354\n",
            "        experiment_type: multi_rna\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21902355\n",
            "        experiment_type: multi_rna\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21902356\n",
            "        experiment_type: multi_rna\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21902364\n",
            "        experiment_type: multi_rna\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21902365\n",
            "        experiment_type: multi_rna\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21902366\n",
            "        experiment_type: multi_rna\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21902367\n",
            "        experiment_type: multi_rna\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21902368\n",
            "        experiment_type: multi_rna\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21902369\n",
            "        experiment_type: multi_rna\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21902370\n",
            "        experiment_type: multi_rna\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX18716672\n",
            "        experiment_type: sc\n",
            "        tech: 10XV2\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX18716673\n",
            "        experiment_type: sc\n",
            "        tech: 10XV2\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX18716674\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX18716675\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: ERX13202612\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: ERX13202613\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: ERX13202614\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: ERX13202615\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: ERX12265374\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: ERX12265375\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: ERX12265376\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: ERX12265377\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX25736219\n",
            "        experiment_type: sn\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX25736220\n",
            "        experiment_type: sn\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX25736221\n",
            "        experiment_type: sn\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX25736222\n",
            "        experiment_type: sn\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX25736223\n",
            "        experiment_type: sn\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX25736224\n",
            "        experiment_type: sn\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX25736225\n",
            "        experiment_type: sn\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX25736226\n",
            "        experiment_type: sn\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX25736227\n",
            "        experiment_type: sn\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX25736228\n",
            "        experiment_type: sn\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX25736229\n",
            "        experiment_type: sn\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX26708357\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX25982294\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX25982295\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX25982296\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX25982297\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX25982298\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX25982301\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX25982300\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX25982299\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX29174027\n",
            "        experiment_type: sn\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX29174028\n",
            "        experiment_type: sn\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX29174029\n",
            "        experiment_type: sn\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX29174030\n",
            "        experiment_type: sn\n",
            "        tech: 10XV3\n",
            "        \n",
            "SRA_ID already aligned for nac\n",
            "\n",
            "        SRA_ID: SRX21168593\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "['/content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434594', '/content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434594_2.fastq.gz', '/content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434595', '/content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434595_2.fastq.gz', '/content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434596', '/content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434596_2.fastq.gz', '/content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434597', '/content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434597_2.fastq.gz']\n",
            "Running kb count\n",
            "kb count -i index_nac.idx -g t2g_nac.txt -c1 cdna.txt -c2 nascent.txt -x 10XV3 --workflow=nac --sum=total --overwrite --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168593/ /content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434596_2.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434596_1.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434595_2.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434595_1.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434594_2.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434594_1.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434597_2.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434597_1.fastq.gz\n",
            "\n",
            "        SRA_ID: SRX21168594\n",
            "        experiment_type: sc\n",
            "        tech: 10XV3\n",
            "        \n",
            "['/content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434590', '/content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434590_2.fastq.gz', '/content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434591_2.fastq.gz', '/content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434591', '/content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434592', '/content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434592_2.fastq.gz', '/content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434593', '/content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434593_2.fastq.gz']\n",
            "Running kb count\n",
            "kb count -i index_nac.idx -g t2g_nac.txt -c1 cdna.txt -c2 nascent.txt -x 10XV3 --workflow=nac --sum=total --overwrite --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168594/ /content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434590_2.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434590_1.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434593_2.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434593_1.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434592_2.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434592_1.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434591_2.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434591_1.fastq.gz\n"
          ]
        }
      ],
      "source": [
        "atlas.run_kb_count_nac(nac_path='/content/drive/MyDrive/pituitary_atlas/processed/nac/',raw_path='/content/drive/MyDrive/pituitary_atlas/raw/',df=df)"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Alignment where files needed to be manually specified"
      ],
      "metadata": {
        "id": "DLLjcOIdPErk"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Pp9SjwXIeBWi",
        "outputId": "9b882644-0756-4589-8f9d-2e4af386b56b"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[2025-08-15 06:46:17,366]    INFO [count_nac] Using index index_nac.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168593/ from\n",
            "[2025-08-15 06:46:17,366]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434596_1.fastq.gz\n",
            "[2025-08-15 06:46:17,366]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434596_2.fastq.gz\n",
            "[2025-08-15 06:46:17,366]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434595_1.fastq.gz\n",
            "[2025-08-15 06:46:17,366]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434595_2.fastq.gz\n",
            "[2025-08-15 06:46:17,366]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434594_1.fastq.gz\n",
            "[2025-08-15 06:46:17,366]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434594_2.fastq.gz\n",
            "[2025-08-15 06:46:17,366]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434597_1.fastq.gz\n",
            "[2025-08-15 06:46:17,367]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434597_2.fastq.gz\n",
            "[2025-08-15 08:32:37,015]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168593/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168593/tmp/output.s.bus\n",
            "[2025-08-15 08:38:52,876]    INFO [count_nac] On-list not provided\n",
            "[2025-08-15 08:38:52,877]    INFO [count_nac] Copying pre-packaged 10XV3 on-list to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168593/\n",
            "[2025-08-15 08:38:53,797]    INFO [count_nac] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168593/tmp/output.s.bus\n",
            "[2025-08-15 08:40:38,357]    INFO [count_nac] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168593/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168593/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168593/10x_version3_whitelist.txt\n",
            "[2025-08-15 08:42:49,463]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168593/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168593/output.unfiltered.bus\n",
            "[2025-08-15 08:48:26,403]    INFO [count_nac] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168593/counts_unfiltered/cells_x_genes from BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168593/output.unfiltered.bus\n",
            "[2025-08-15 08:52:12,692]    INFO [count_nac] Writing gene names to file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168593/counts_unfiltered/cells_x_genes.genes.names.txt\n",
            "[2025-08-15 08:52:12,951] WARNING [count_nac] 1599 gene IDs do not have corresponding valid gene names. These genes will use their gene IDs instead.\n",
            "[2025-08-15 08:52:12,989]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168593/counts_unfiltered/cells_x_genes.cell.mtx\n",
            "[2025-08-15 08:53:10,832]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168593/counts_unfiltered/cells_x_genes.cell.mtx\n",
            "[2025-08-15 08:54:12,443]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168593/counts_unfiltered/cells_x_genes.nucleus.mtx\n",
            "[2025-08-15 08:55:06,794]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168593/counts_unfiltered/cells_x_genes.nucleus.mtx\n",
            "[2025-08-15 08:56:11,340]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168593/counts_unfiltered/cells_x_genes.total.mtx\n",
            "[2025-08-15 08:57:02,158]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168593/counts_unfiltered/cells_x_genes.total.mtx\n",
            "[2025-08-15 08:58:20,196]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168593/counts_unfiltered/cells_x_genes.mature.mtx\n",
            "[2025-08-15 08:58:27,229]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168593/counts_unfiltered/cells_x_genes.nascent.mtx\n",
            "[2025-08-15 08:58:32,366]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168593/counts_unfiltered/cells_x_genes.ambiguous.mtx\n",
            "[2025-08-15 08:58:38,495]    INFO [count_nac] Combining matrices\n",
            "[2025-08-15 08:58:43,350]    INFO [count_nac] Writing matrices to h5ad /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168593/counts_unfiltered/adata.h5ad\n"
          ]
        }
      ],
      "source": [
        "!kb count -i index_nac.idx -g t2g_nac.txt -c1 cdna.txt -c2 nascent.txt -x 10XV3 --workflow=nac --sum=total --overwrite --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168593/ /content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434596_1.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434596_2.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434595_1.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434595_2.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434594_1.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434594_2.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434597_1.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168593/SRR25434597_2.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "aMseOrdAia2i",
        "outputId": "c9e20037-b121-4cbe-bce5-30787c934859"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[2025-08-15 08:59:11,256]    INFO [count_nac] Using index index_nac.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168594/ from\n",
            "[2025-08-15 08:59:11,256]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434590_1.fastq.gz\n",
            "[2025-08-15 08:59:11,256]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434590_2.fastq.gz\n",
            "[2025-08-15 08:59:11,256]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434593_1.fastq.gz\n",
            "[2025-08-15 08:59:11,256]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434593_2.fastq.gz\n",
            "[2025-08-15 08:59:11,256]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434592_1.fastq.gz\n",
            "[2025-08-15 08:59:11,256]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434592_2.fastq.gz\n",
            "[2025-08-15 08:59:11,256]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434591_1.fastq.gz\n",
            "[2025-08-15 08:59:11,257]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434591_2.fastq.gz\n",
            "[2025-08-15 11:11:18,462]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168594/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168594/tmp/output.s.bus\n",
            "[2025-08-15 11:16:39,880]    INFO [count_nac] On-list not provided\n",
            "[2025-08-15 11:16:39,881]    INFO [count_nac] Copying pre-packaged 10XV3 on-list to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168594/\n",
            "[2025-08-15 11:16:40,782]    INFO [count_nac] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168594/tmp/output.s.bus\n",
            "[2025-08-15 11:18:27,574]    INFO [count_nac] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168594/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168594/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168594/10x_version3_whitelist.txt\n",
            "[2025-08-15 11:20:32,188]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168594/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168594/output.unfiltered.bus\n",
            "[2025-08-15 11:25:51,799]    INFO [count_nac] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168594/counts_unfiltered/cells_x_genes from BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168594/output.unfiltered.bus\n",
            "[2025-08-15 11:29:44,758]    INFO [count_nac] Writing gene names to file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168594/counts_unfiltered/cells_x_genes.genes.names.txt\n",
            "[2025-08-15 11:29:45,037] WARNING [count_nac] 1599 gene IDs do not have corresponding valid gene names. These genes will use their gene IDs instead.\n",
            "[2025-08-15 11:29:45,080]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168594/counts_unfiltered/cells_x_genes.cell.mtx\n",
            "[2025-08-15 11:30:36,324]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168594/counts_unfiltered/cells_x_genes.cell.mtx\n",
            "[2025-08-15 11:31:37,058]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168594/counts_unfiltered/cells_x_genes.nucleus.mtx\n",
            "[2025-08-15 11:32:18,166]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168594/counts_unfiltered/cells_x_genes.nucleus.mtx\n",
            "[2025-08-15 11:33:22,806]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168594/counts_unfiltered/cells_x_genes.total.mtx\n",
            "[2025-08-15 11:34:27,148]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168594/counts_unfiltered/cells_x_genes.total.mtx\n",
            "[2025-08-15 11:35:46,951]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168594/counts_unfiltered/cells_x_genes.mature.mtx\n",
            "[2025-08-15 11:35:53,744]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168594/counts_unfiltered/cells_x_genes.nascent.mtx\n",
            "[2025-08-15 11:36:08,150]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168594/counts_unfiltered/cells_x_genes.ambiguous.mtx\n",
            "[2025-08-15 11:36:16,146]    INFO [count_nac] Combining matrices\n",
            "[2025-08-15 11:36:21,606]    INFO [count_nac] Writing matrices to h5ad /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168594/counts_unfiltered/adata.h5ad\n"
          ]
        }
      ],
      "source": [
        "!kb count -i index_nac.idx -g t2g_nac.txt -c1 cdna.txt -c2 nascent.txt -x 10XV3 --workflow=nac --sum=total --overwrite --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168594/ /content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434590_1.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434590_2.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434593_1.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434593_2.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434592_1.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434592_2.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434591_1.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168594/SRR25434591_2.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "background_save": true,
          "base_uri": "https://localhost:8080/"
        },
        "id": "tV6hsOCIEQWC",
        "outputId": "d21f1b0c-4f32-4f5b-8d54-cb00e71da4ae"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[2025-08-15 12:00:07,849]    INFO [count_nac] Using index index_nac.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168595/ from\n",
            "[2025-08-15 12:00:07,850]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168595/SRR25434586_1.fastq.gz\n",
            "[2025-08-15 12:00:07,850]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168595/SRR25434586_2.fastq.gz\n",
            "[2025-08-15 12:00:07,850]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168595/SRR25434587_1.fastq.gz\n",
            "[2025-08-15 12:00:07,850]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168595/SRR25434587_2.fastq.gz\n",
            "[2025-08-15 12:00:07,850]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168595/SRR25434588_1.fastq.gz\n",
            "[2025-08-15 12:00:07,850]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168595/SRR25434588_2.fastq.gz\n",
            "[2025-08-15 12:00:07,850]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168595/SRR25434589_1.fastq.gz\n",
            "[2025-08-15 12:00:07,850]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168595/SRR25434589_2.fastq.gz\n",
            "[2025-08-15 13:55:09,328]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168595/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168595/tmp/output.s.bus\n",
            "[2025-08-15 13:59:09,815]    INFO [count_nac] On-list not provided\n",
            "[2025-08-15 13:59:09,816]    INFO [count_nac] Copying pre-packaged 10XV3 on-list to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168595/\n",
            "[2025-08-15 13:59:10,674]    INFO [count_nac] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168595/tmp/output.s.bus\n",
            "[2025-08-15 14:01:32,219]    INFO [count_nac] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168595/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168595/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168595/10x_version3_whitelist.txt\n",
            "[2025-08-15 14:03:43,264]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168595/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168595/output.unfiltered.bus\n",
            "[2025-08-15 14:07:43,965]    INFO [count_nac] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168595/counts_unfiltered/cells_x_genes from BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168595/output.unfiltered.bus\n",
            "[2025-08-15 14:11:19,626]    INFO [count_nac] Writing gene names to file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168595/counts_unfiltered/cells_x_genes.genes.names.txt\n",
            "[2025-08-15 14:11:19,880] WARNING [count_nac] 1599 gene IDs do not have corresponding valid gene names. These genes will use their gene IDs instead.\n",
            "[2025-08-15 14:11:19,918]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168595/counts_unfiltered/cells_x_genes.cell.mtx\n",
            "[2025-08-15 14:12:05,221]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168595/counts_unfiltered/cells_x_genes.cell.mtx\n",
            "[2025-08-15 14:12:55,569]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168595/counts_unfiltered/cells_x_genes.nucleus.mtx\n",
            "[2025-08-15 14:13:32,246]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168595/counts_unfiltered/cells_x_genes.nucleus.mtx\n",
            "[2025-08-15 14:14:48,854]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168595/counts_unfiltered/cells_x_genes.total.mtx\n",
            "[2025-08-15 14:15:33,608]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168595/counts_unfiltered/cells_x_genes.total.mtx\n",
            "[2025-08-15 14:16:42,467]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168595/counts_unfiltered/cells_x_genes.mature.mtx\n",
            "[2025-08-15 14:16:48,417]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168595/counts_unfiltered/cells_x_genes.nascent.mtx\n",
            "[2025-08-15 14:16:53,085]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168595/counts_unfiltered/cells_x_genes.ambiguous.mtx\n",
            "[2025-08-15 14:16:59,324]    INFO [count_nac] Combining matrices\n",
            "[2025-08-15 14:17:04,183]    INFO [count_nac] Writing matrices to h5ad /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168595/counts_unfiltered/adata.h5ad\n"
          ]
        }
      ],
      "source": [
        "!kb count -i index_nac.idx -g t2g_nac.txt -c1 cdna.txt -c2 nascent.txt -x 10XV3 --workflow=nac --sum=total --overwrite --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168595/ /content/drive/MyDrive/pituitary_atlas/raw/SRX21168595/SRR25434586_1.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168595/SRR25434586_2.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168595/SRR25434587_1.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168595/SRR25434587_2.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168595/SRR25434588_1.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168595/SRR25434588_2.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168595/SRR25434589_1.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168595/SRR25434589_2.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "background_save": true
        },
        "id": "OSVmOpkpEQgN",
        "outputId": "b0c8f3f6-aacd-4551-bb95-63a37599783a"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[2025-08-15 16:59:15,396]    INFO [count_nac] Using index index_nac.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168596/ from\n",
            "[2025-08-15 16:59:15,396]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168596/SRR25434582_1.fastq.gz\n",
            "[2025-08-15 16:59:15,397]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168596/SRR25434582_2.fastq.gz\n",
            "[2025-08-15 16:59:15,397]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168596/SRR25434583_1.fastq.gz\n",
            "[2025-08-15 16:59:15,397]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168596/SRR25434583_2.fastq.gz\n",
            "[2025-08-15 16:59:15,397]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168596/SRR25434584_1.fastq.gz\n",
            "[2025-08-15 16:59:15,397]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168596/SRR25434584_2.fastq.gz\n",
            "[2025-08-15 16:59:15,397]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168596/SRR25434585_1.fastq.gz\n",
            "[2025-08-15 16:59:15,397]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX21168596/SRR25434585_2.fastq.gz\n",
            "[2025-08-15 19:03:27,631]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168596/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168596/tmp/output.s.bus\n",
            "[2025-08-15 19:08:46,726]    INFO [count_nac] On-list not provided\n",
            "[2025-08-15 19:08:46,727]    INFO [count_nac] Copying pre-packaged 10XV3 on-list to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168596/\n",
            "[2025-08-15 19:08:49,180]    INFO [count_nac] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168596/tmp/output.s.bus\n",
            "[2025-08-15 19:09:56,867]    INFO [count_nac] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168596/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168596/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168596/10x_version3_whitelist.txt\n",
            "[2025-08-15 19:12:45,092]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168596/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168596/output.unfiltered.bus\n",
            "[2025-08-15 19:17:46,661]    INFO [count_nac] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168596/counts_unfiltered/cells_x_genes from BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168596/output.unfiltered.bus\n",
            "[2025-08-15 19:21:22,158]    INFO [count_nac] Writing gene names to file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168596/counts_unfiltered/cells_x_genes.genes.names.txt\n",
            "[2025-08-15 19:21:22,731] WARNING [count_nac] 1599 gene IDs do not have corresponding valid gene names. These genes will use their gene IDs instead.\n",
            "[2025-08-15 19:21:23,453]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168596/counts_unfiltered/cells_x_genes.cell.mtx\n",
            "[2025-08-15 19:22:39,948]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168596/counts_unfiltered/cells_x_genes.cell.mtx\n",
            "[2025-08-15 19:23:39,643]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168596/counts_unfiltered/cells_x_genes.nucleus.mtx\n",
            "[2025-08-15 19:24:23,271]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168596/counts_unfiltered/cells_x_genes.nucleus.mtx\n",
            "[2025-08-15 19:25:26,271]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168596/counts_unfiltered/cells_x_genes.total.mtx\n",
            "[2025-08-15 19:26:17,267]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168596/counts_unfiltered/cells_x_genes.total.mtx\n",
            "[2025-08-15 19:27:30,953]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168596/counts_unfiltered/cells_x_genes.mature.mtx\n",
            "[2025-08-15 19:27:35,780]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168596/counts_unfiltered/cells_x_genes.nascent.mtx\n",
            "[2025-08-15 19:27:39,435]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168596/counts_unfiltered/cells_x_genes.ambiguous.mtx\n",
            "[2025-08-15 19:27:43,632]    INFO [count_nac] Combining matrices\n",
            "[2025-08-15 19:27:47,878]    INFO [count_nac] Writing matrices to h5ad /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168596/counts_unfiltered/adata.h5ad\n"
          ]
        }
      ],
      "source": [
        "!kb count -i index_nac.idx -g t2g_nac.txt -c1 cdna.txt -c2 nascent.txt -x 10XV3 --workflow=nac --sum=total --overwrite --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21168596/ /content/drive/MyDrive/pituitary_atlas/raw/SRX21168596/SRR25434582_1.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168596/SRR25434582_2.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168596/SRR25434583_1.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168596/SRR25434583_2.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168596/SRR25434584_1.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168596/SRR25434584_2.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168596/SRR25434585_1.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX21168596/SRR25434585_2.fastq.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "hJIoYvQxtRa2"
      },
      "source": [
        "# Cheung et al (2018) uploaded strange fastq files, and therefore we turned to the .bam files and aimed to convert those back to fastq files using the 10X bamtofastq tool."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "8wYtMHnItSkZ",
        "outputId": "ed7223fc-edd2-4178-d533-b100df336239"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "--2024-07-08 09:41:48--  https://sra-pub-src-1.s3.amazonaws.com/SRR7898909/pool1_possorted_genome_bam.bam.1\n",
            "Resolving sra-pub-src-1.s3.amazonaws.com (sra-pub-src-1.s3.amazonaws.com)... 52.216.220.169, 3.5.30.143, 52.216.186.235, ...\n",
            "Connecting to sra-pub-src-1.s3.amazonaws.com (sra-pub-src-1.s3.amazonaws.com)|52.216.220.169|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 29532119570 (28G) [application/x-troff-man]\n",
            "Saving to: ‘/content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/pool1_possorted_genome_bam.bam.1’\n",
            "\n",
            "/content/drive/MyDr 100%[===================>]  27.50G  54.4MB/s    in 9m 56s  \n",
            "\n",
            "2024-07-08 09:51:45 (47.2 MB/s) - ‘/content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/pool1_possorted_genome_bam.bam.1’ saved [29532119570/29532119570]\n",
            "\n",
            "Downloaded file to /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/pool1_possorted_genome_bam.bam.1\n",
            "--2024-07-08 09:51:47--  https://sra-pub-src-1.s3.amazonaws.com/SRR7898910/pool2_possorted_genome_bam.bam.1\n",
            "Resolving sra-pub-src-1.s3.amazonaws.com (sra-pub-src-1.s3.amazonaws.com)... 52.216.43.161, 52.217.169.129, 52.216.184.27, ...\n",
            "Connecting to sra-pub-src-1.s3.amazonaws.com (sra-pub-src-1.s3.amazonaws.com)|52.216.43.161|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 35721801699 (33G) [application/x-troff-man]\n",
            "Saving to: ‘/content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/pool2_possorted_genome_bam.bam.1’\n",
            "\n",
            "/content/drive/MyDr 100%[===================>]  33.27G  50.0MB/s    in 12m 2s  \n",
            "\n",
            "2024-07-08 10:03:49 (47.2 MB/s) - ‘/content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/pool2_possorted_genome_bam.bam.1’ saved [35721801699/35721801699]\n",
            "\n",
            "Downloaded file to /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/pool2_possorted_genome_bam.bam.1\n"
          ]
        }
      ],
      "source": [
        "import os\n",
        "\n",
        "# Ensure the destination directory exists\n",
        "output_dir = \"/content/drive/MyDrive/pituitary_atlas/raw/SRX4736337\"\n",
        "os.makedirs(output_dir, exist_ok=True)\n",
        "# URL of the file to be downloaded\n",
        "url = \"https://sra-pub-src-1.s3.amazonaws.com/SRR7898909/pool1_possorted_genome_bam.bam.1\"\n",
        "# Path where the file will be saved\n",
        "output_file_path = os.path.join(output_dir, \"pool1_possorted_genome_bam.bam.1\")\n",
        "# Use wget to download the file\n",
        "!wget -O {output_file_path} {url}\n",
        "print(f\"Downloaded file to {output_file_path}\")\n",
        "\n",
        "url2 = \"https://sra-pub-src-1.s3.amazonaws.com/SRR7898910/pool2_possorted_genome_bam.bam.1\"\n",
        "output_dir = \"/content/drive/MyDrive/pituitary_atlas/raw/SRX4736338\"\n",
        "output_file_path2 = os.path.join(output_dir, \"pool2_possorted_genome_bam.bam.1\")\n",
        "!wget -O {output_file_path2} {url2}\n",
        "print(f\"Downloaded file to {output_file_path2}\")"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ZkTKL40MtyHL",
        "outputId": "b4cce9e9-5f04-4472-a717-2a573767d4f0"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "bamtofastq v1.4.1\n",
            "Writing finished.  Observed 273079474 read pairs. Wrote 273079474 read pairs\n"
          ]
        }
      ],
      "source": [
        "!bamtofastq /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/pool1_possorted_genome_bam.bam.1 /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "el2sFAie_YuO",
        "outputId": "7419a5b6-b911-4353-9e43-51cdc22fdc77"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[2024-07-08 11:07:41,292]    INFO [count_nac] Using index index_nac.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736337/ from\n",
            "[2024-07-08 11:07:41,292]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R1_001.fastq.gz\n",
            "[2024-07-08 11:07:41,292]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R2_001.fastq.gz\n",
            "[2024-07-08 11:07:41,292]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R1_002.fastq.gz\n",
            "[2024-07-08 11:07:41,292]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R2_002.fastq.gz\n",
            "[2024-07-08 11:07:41,292]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R1_003.fastq.gz\n",
            "[2024-07-08 11:07:41,292]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R2_003.fastq.gz\n",
            "[2024-07-08 11:07:41,292]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R1_001.fastq.gz\n",
            "[2024-07-08 11:07:41,292]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R2_001.fastq.gz\n",
            "[2024-07-08 11:07:41,292]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R1_002.fastq.gz\n",
            "[2024-07-08 11:07:41,292]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R2_002.fastq.gz\n",
            "[2024-07-08 11:07:41,292]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R1_003.fastq.gz\n",
            "[2024-07-08 11:07:41,292]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R2_003.fastq.gz\n",
            "[2024-07-08 11:52:19,760]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736337/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736337/tmp/output.s.bus\n",
            "[2024-07-08 11:55:02,402]    INFO [count_nac] On-list not provided\n",
            "[2024-07-08 11:55:02,402]    INFO [count_nac] Copying pre-packaged 10XV2 on-list to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736337/\n",
            "[2024-07-08 11:55:02,562]    INFO [count_nac] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736337/tmp/output.s.bus\n",
            "[2024-07-08 11:55:24,309]    INFO [count_nac] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736337/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736337/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736337/10x_version2_whitelist.txt\n",
            "[2024-07-08 11:56:38,768]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736337/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736337/output.unfiltered.bus\n",
            "[2024-07-08 11:57:10,942]    INFO [count_nac] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736337/counts_unfiltered/cells_x_genes from BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736337/output.unfiltered.bus\n",
            "[2024-07-08 12:00:32,199]    INFO [count_nac] Writing gene names to file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736337/counts_unfiltered/cells_x_genes.genes.names.txt\n",
            "[2024-07-08 12:00:32,558] WARNING [count_nac] 1599 gene IDs do not have corresponding valid gene names. These genes will use their gene IDs instead.\n",
            "[2024-07-08 12:00:32,610]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736337/counts_unfiltered/cells_x_genes.cell.mtx\n",
            "[2024-07-08 12:01:09,054]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736337/counts_unfiltered/cells_x_genes.cell.mtx\n",
            "[2024-07-08 12:01:59,918]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736337/counts_unfiltered/cells_x_genes.nucleus.mtx\n",
            "[2024-07-08 12:02:40,225]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736337/counts_unfiltered/cells_x_genes.nucleus.mtx\n",
            "[2024-07-08 12:03:34,359]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736337/counts_unfiltered/cells_x_genes.total.mtx\n",
            "[2024-07-08 12:04:19,448]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736337/counts_unfiltered/cells_x_genes.total.mtx\n",
            "[2024-07-08 12:05:21,542]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736337/counts_unfiltered/cells_x_genes.mature.mtx\n",
            "[2024-07-08 12:05:30,101]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736337/counts_unfiltered/cells_x_genes.nascent.mtx\n",
            "[2024-07-08 12:05:40,698]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736337/counts_unfiltered/cells_x_genes.ambiguous.mtx\n",
            "[2024-07-08 12:06:05,917]    INFO [count_nac] Combining matrices\n",
            "[2024-07-08 12:06:07,110]    INFO [count_nac] Writing matrices to h5ad /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736337/counts_unfiltered/adata.h5ad\n"
          ]
        }
      ],
      "source": [
        "!kb count -i index_nac.idx -g t2g_nac.txt -c1 cdna.txt -c2 nascent.txt --overwrite -x 10xv2 --workflow=nac --sum=total --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736337/ /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R1_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R2_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R1_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R2_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R1_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R2_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R1_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R2_003.fastq.gz\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "oXRyhCeGEDSF"
      },
      "outputs": [],
      "source": [
        "!bamtofastq /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/pool2_possorted_genome_bam.bam.1 /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "9kTUoob3Z2Q4",
        "outputId": "5fc7e19b-aa2a-49a5-e59b-ae131d8b0823"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[2024-07-08 14:33:44,870]    INFO [count_nac] Using index index_nac.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736338/ from\n",
            "[2024-07-08 14:33:44,870]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R1_001.fastq.gz\n",
            "[2024-07-08 14:33:44,870]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R2_001.fastq.gz\n",
            "[2024-07-08 14:33:44,870]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R1_002.fastq.gz\n",
            "[2024-07-08 14:33:44,870]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R2_002.fastq.gz\n",
            "[2024-07-08 14:33:44,870]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R1_003.fastq.gz\n",
            "[2024-07-08 14:33:44,870]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R2_003.fastq.gz\n",
            "[2024-07-08 14:33:44,870]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R1_004.fastq.gz\n",
            "[2024-07-08 14:33:44,870]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R2_004.fastq.gz\n",
            "[2024-07-08 14:33:44,870]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R1_001.fastq.gz\n",
            "[2024-07-08 14:33:44,871]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R2_001.fastq.gz\n",
            "[2024-07-08 14:33:44,871]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R1_002.fastq.gz\n",
            "[2024-07-08 14:33:44,871]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R2_002.fastq.gz\n",
            "[2024-07-08 14:33:44,871]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R1_003.fastq.gz\n",
            "[2024-07-08 14:33:44,871]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R2_003.fastq.gz\n",
            "[2024-07-08 14:33:44,871]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R1_004.fastq.gz\n",
            "[2024-07-08 14:33:44,871]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R2_004.fastq.gz\n",
            "[2024-07-08 15:29:42,636]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736338/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736338/tmp/output.s.bus\n",
            "[2024-07-08 15:33:10,222]    INFO [count_nac] On-list not provided\n",
            "[2024-07-08 15:33:10,223]    INFO [count_nac] Copying pre-packaged 10XV2 on-list to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736338/\n",
            "[2024-07-08 15:33:10,369]    INFO [count_nac] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736338/tmp/output.s.bus\n",
            "[2024-07-08 15:33:38,430]    INFO [count_nac] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736338/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736338/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736338/10x_version2_whitelist.txt\n",
            "[2024-07-08 15:35:09,224]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736338/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736338/output.unfiltered.bus\n"
          ]
        }
      ],
      "source": [
        "!kb count -i index_nac.idx -g t2g_nac.txt -c1 cdna.txt -c2 nascent.txt --overwrite -x 10xv2 --workflow=nac --sum=total --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX4736338/ /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R1_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R2_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R1_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R2_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R1_004.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R2_004.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R1_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R2_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R1_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R2_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R1_004.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R2_004.fastq.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "7ATP9o6n-ln7"
      },
      "source": [
        "# Same for Bastedo et al"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "xZdRKlJs-lIT",
        "outputId": "a2e949e0-9810-44a4-d274-6ee1656a52e6"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "--2024-08-03 15:02:57--  https://sra-pub-src-1.s3.amazonaws.com/SRR13203971/Pit_possorted_genome_bam.bam.1\n",
            "Resolving sra-pub-src-1.s3.amazonaws.com (sra-pub-src-1.s3.amazonaws.com)... 3.5.30.241, 54.231.171.169, 52.216.28.156, ...\n",
            "Connecting to sra-pub-src-1.s3.amazonaws.com (sra-pub-src-1.s3.amazonaws.com)|3.5.30.241|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 40072014800 (37G) [binary/octet-stream]\n",
            "Saving to: ‘/content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/pool1_possorted_genome_bam.bam.1’\n",
            "\n",
            "/content/drive/MyDr 100%[===================>]  37.32G  22.4MB/s    in 27m 13s \n",
            "\n",
            "2024-08-03 15:30:10 (23.4 MB/s) - ‘/content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/pool1_possorted_genome_bam.bam.1’ saved [40072014800/40072014800]\n",
            "\n",
            "Downloaded file to /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/pool1_possorted_genome_bam.bam.1\n"
          ]
        }
      ],
      "source": [
        "import os\n",
        "\n",
        "# Ensure the destination directory exists\n",
        "output_dir = \"/content/drive/MyDrive/pituitary_atlas/raw/SRX9637224\"\n",
        "os.makedirs(output_dir, exist_ok=True)\n",
        "# URL of the file to be downloaded\n",
        "url = \"https://sra-pub-src-1.s3.amazonaws.com/SRR13203971/Pit_possorted_genome_bam.bam.1\"\n",
        "# Path where the file will be saved\n",
        "output_file_path = os.path.join(output_dir, \"pool1_possorted_genome_bam.bam.1\")\n",
        "# Use wget to download the file\n",
        "!wget -O {output_file_path} {url}\n",
        "print(f\"Downloaded file to {output_file_path}\")"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "RhNet0Dl_39N",
        "outputId": "07d95b57-24f4-4e3f-e94c-4eb218386c87"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "bamtofastq v1.4.1\n",
            "Writing finished.  Observed 576030445 read pairs. Wrote 576030445 read pairs\n"
          ]
        }
      ],
      "source": [
        "!bamtofastq /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/pool1_possorted_genome_bam.bam.1 /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "UYNVPDh3AAsJ",
        "outputId": "2ef92d63-48a1-406c-cbd0-20e6a4c0a026"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[2024-08-11 15:30:12,030]    INFO [count_nac] Using index index_nac.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX9637224/ from\n",
            "[2024-08-11 15:30:12,030]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L001_R1_001.fastq.gz\n",
            "[2024-08-11 15:30:12,030]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L001_R2_001.fastq.gz\n",
            "[2024-08-11 15:30:12,030]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L001_R1_002.fastq.gz\n",
            "[2024-08-11 15:30:12,030]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L001_R2_002.fastq.gz\n",
            "[2024-08-11 15:30:12,030]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L001_R1_003.fastq.gz\n",
            "[2024-08-11 15:30:12,030]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L001_R2_003.fastq.gz\n",
            "[2024-08-11 15:30:12,030]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L002_R1_001.fastq.gz\n",
            "[2024-08-11 15:30:12,030]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L002_R2_001.fastq.gz\n",
            "[2024-08-11 15:30:12,030]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L002_R1_002.fastq.gz\n",
            "[2024-08-11 15:30:12,030]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L002_R2_002.fastq.gz\n",
            "[2024-08-11 15:30:12,030]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L002_R1_003.fastq.gz\n",
            "[2024-08-11 15:30:12,030]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L002_R2_003.fastq.gz\n",
            "[2024-08-11 15:30:12,031]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L003_R1_001.fastq.gz\n",
            "[2024-08-11 15:30:12,031]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L003_R2_001.fastq.gz\n",
            "[2024-08-11 15:30:12,031]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L003_R1_002.fastq.gz\n",
            "[2024-08-11 15:30:12,031]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L003_R2_002.fastq.gz\n",
            "[2024-08-11 15:30:12,031]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L003_R1_003.fastq.gz\n",
            "[2024-08-11 15:30:12,031]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L003_R2_003.fastq.gz\n",
            "[2024-08-11 15:30:12,031]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L004_R1_001.fastq.gz\n",
            "[2024-08-11 15:30:12,031]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L004_R2_001.fastq.gz\n",
            "[2024-08-11 15:30:12,031]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L004_R1_002.fastq.gz\n",
            "[2024-08-11 15:30:12,031]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L004_R2_002.fastq.gz\n",
            "[2024-08-11 15:30:12,031]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L004_R1_003.fastq.gz\n",
            "[2024-08-11 15:30:12,031]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L004_R2_003.fastq.gz\n",
            "[2024-08-11 15:59:40,364]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX9637224/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX9637224/tmp/output.s.bus\n",
            "[2024-08-11 16:02:44,718]    INFO [count_nac] On-list not provided\n",
            "[2024-08-11 16:02:44,719]    INFO [count_nac] Copying pre-packaged 10XV3 on-list to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX9637224/\n",
            "[2024-08-11 16:02:45,466]    INFO [count_nac] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX9637224/tmp/output.s.bus\n",
            "[2024-08-11 16:03:49,215]    INFO [count_nac] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX9637224/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX9637224/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX9637224/10x_version3_whitelist.txt\n",
            "[2024-08-11 16:05:29,712]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX9637224/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX9637224/output.unfiltered.bus\n",
            "[2024-08-11 16:07:34,861]    INFO [count_nac] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX9637224/counts_unfiltered/cells_x_genes from BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX9637224/output.unfiltered.bus\n",
            "[2024-08-11 16:13:24,868]    INFO [count_nac] Writing gene names to file /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX9637224/counts_unfiltered/cells_x_genes.genes.names.txt\n",
            "[2024-08-11 16:13:25,071] WARNING [count_nac] 1599 gene IDs do not have corresponding valid gene names. These genes will use their gene IDs instead.\n",
            "[2024-08-11 16:13:25,102]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX9637224/counts_unfiltered/cells_x_genes.cell.mtx\n",
            "[2024-08-11 16:14:09,103]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX9637224/counts_unfiltered/cells_x_genes.cell.mtx\n",
            "[2024-08-11 16:15:02,487]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX9637224/counts_unfiltered/cells_x_genes.nucleus.mtx\n",
            "[2024-08-11 16:15:42,551]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX9637224/counts_unfiltered/cells_x_genes.nucleus.mtx\n",
            "[2024-08-11 16:16:35,617]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX9637224/counts_unfiltered/cells_x_genes.total.mtx\n",
            "[2024-08-11 16:17:17,141]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX9637224/counts_unfiltered/cells_x_genes.total.mtx\n",
            "[2024-08-11 16:18:14,866]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX9637224/counts_unfiltered/cells_x_genes.mature.mtx\n",
            "[2024-08-11 16:18:17,744]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX9637224/counts_unfiltered/cells_x_genes.nascent.mtx\n",
            "[2024-08-11 16:18:20,325]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX9637224/counts_unfiltered/cells_x_genes.ambiguous.mtx\n",
            "[2024-08-11 16:18:23,766]    INFO [count_nac] Combining matrices\n",
            "[2024-08-11 16:18:25,916]    INFO [count_nac] Writing matrices to h5ad /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX9637224/counts_unfiltered/adata.h5ad\n"
          ]
        }
      ],
      "source": [
        "!kb count -i index_nac.idx -g t2g_nac.txt -c1 cdna.txt -c2 nascent.txt --overwrite -x 10XV3 --workflow=nac --sum=total --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX9637224/ /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L001_R1_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L001_R2_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L001_R1_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L001_R2_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L002_R1_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L002_R2_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L002_R1_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L002_R2_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L003_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L003_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L003_R1_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L003_R2_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L003_R1_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L003_R2_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L004_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L004_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L004_R1_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L004_R2_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L004_R1_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L004_R2_003.fastq.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "owAathCqAtCN"
      },
      "source": [
        "# Now Cheung 2020"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "XSsbac4cAsQs"
      },
      "outputs": [],
      "source": [
        "import os\n",
        "\n",
        "# Ensure the destination directory exists\n",
        "output_dir = \"/content/drive/MyDrive/pituitary_atlas/raw/SRX7377164\"\n",
        "os.makedirs(output_dir, exist_ok=True)\n",
        "# URL of the file to be downloaded\n",
        "url = \"https://sra-pub-src-2.s3.amazonaws.com/SRR10700816/Hesx1Cre_possorted_genome_bam.bam.1\"\n",
        "# Path where the file will be saved\n",
        "output_file_path = os.path.join(output_dir, \"pool1_possorted_genome_bam.bam.1\")\n",
        "# Use wget to download the file\n",
        "!wget -O {output_file_path} {url}\n",
        "print(f\"Downloaded file to {output_file_path}\")\n",
        "\n",
        "url2 = \"https://sra-pub-src-2.s3.amazonaws.com/SRR10700818/Hesx1Cre_dnRAR_possorted_genome_bam.bam.1\"\n",
        "output_dir = \"/content/drive/MyDrive/pituitary_atlas/raw/SRX7377166\"\n",
        "output_file_path2 = os.path.join(output_dir, \"pool2_possorted_genome_bam.bam.1\")\n",
        "!wget -O {output_file_path2} {url2}\n",
        "print(f\"Downloaded file to {output_file_path2}\")\n",
        "\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "2MXPWjSUCL1e"
      },
      "outputs": [],
      "source": [
        "!bamtofastq /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/pool1_possorted_genome_bam.bam.1 /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "_84g4FUpFDZK"
      },
      "outputs": [],
      "source": [
        "!kb count -i index_nac.idx -g t2g_nac.txt -c1 cdna.txt -c2 nascent.txt --overwrite -x 10xv2 --workflow=nac --sum=total --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX7377164/ /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R1_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R2_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R1_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R2_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R1_004.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R2_004.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R1_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R2_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R1_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R2_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R1_004.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R2_004.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "AeE0QCBGCMDw"
      },
      "outputs": [],
      "source": [
        "!bamtofastq /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/pool2_possorted_genome_bam.bam.1 /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "g3bs_AUaFE2D"
      },
      "outputs": [],
      "source": [
        "!kb count -i index_nac.idx -g t2g_nac.txt -c1 cdna.txt -c2 nascent.txt --overwrite -x 10xv2 --workflow=nac --sum=total --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX7377166/ /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R1_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R2_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R1_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R2_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R1_004.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R2_004.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R1_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R2_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R1_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R2_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R1_004.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R2_004.fastq.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "wrssB8wezrYS"
      },
      "source": [
        "# Found strange fastq files in De Vriendt et al. (2025) using SRA so getting files from arrayexpress instead."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "58w1iYODzvSK"
      },
      "outputs": [],
      "source": [
        "ftp_links = [\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892994/scrEXT356_mm10_S19_L001_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892995/scrEXT356_mm10_S3_L001_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892996/scrEXT356_mm10_S19_L002_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892996/scrEXT356_mm10_S19_L002_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892994/scrEXT356_mm10_S19_L001_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892995/scrEXT356_mm10_S3_L001_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892995/scrEXT356_mm10_S3_L001_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892994/scrEXT356_mm10_S19_L001_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892996/scrEXT356_mm10_S19_L002_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892998/scrEXT357_mm10_S4_L001_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12893000/scrEXT357_mm10_S20_L001_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12893000/scrEXT357_mm10_S20_L001_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892998/scrEXT357_mm10_S4_L001_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892991/scrEXT357_mm10_S20_L002_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892991/scrEXT357_mm10_S20_L002_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12893000/scrEXT357_mm10_S20_L001_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892991/scrEXT357_mm10_S20_L002_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892998/scrEXT357_mm10_S4_L001_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892993/scrEXT358_mm10_S11_L001_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892993/scrEXT358_mm10_S11_L001_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892993/scrEXT358_mm10_S11_L001_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892997/scrEXT364_mm10_S6_L002_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892988/scrEXT364_mm10_S6_L003_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892990/scrEXT364_mm10_S10_L002_I2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892988/scrEXT364_mm10_S6_L003_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892990/scrEXT364_mm10_S10_L002_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892992/scrEXT364_mm10_S10_L001_I2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892989/scrEXT364_mm10_S6_L001_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892999/scrEXT364_mm10_S6_L004_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892989/scrEXT364_mm10_S6_L001_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892992/scrEXT364_mm10_S10_L001_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892988/scrEXT364_mm10_S6_L003_I2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892990/scrEXT364_mm10_S10_L002_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892988/scrEXT364_mm10_S6_L003_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892989/scrEXT364_mm10_S6_L001_I2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892999/scrEXT364_mm10_S6_L004_I2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892997/scrEXT364_mm10_S6_L002_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892992/scrEXT364_mm10_S10_L001_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892989/scrEXT364_mm10_S6_L001_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892999/scrEXT364_mm10_S6_L004_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892997/scrEXT364_mm10_S6_L002_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892999/scrEXT364_mm10_S6_L004_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892990/scrEXT364_mm10_S10_L002_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892992/scrEXT364_mm10_S10_L001_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR128/ERR12892997/scrEXT364_mm10_S6_L002_I2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801862/scrEXT330_mm10_S1_L004_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801843/scrEXT330_mm10_S5_L001_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801862/scrEXT330_mm10_S1_L004_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801842/scrEXT330_mm10_S5_L002_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801856/scrEXT330_mm10_S1_L002_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801861/scrEXT330_mm10_S31_L002_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801856/scrEXT330_mm10_S1_L002_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801852/scrEXT330_mm10_S31_L001_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801856/scrEXT330_mm10_S1_L002_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801861/scrEXT330_mm10_S31_L002_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801846/scrEXT330_mm10_S1_L003_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801843/scrEXT330_mm10_S5_L001_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801861/scrEXT330_mm10_S31_L002_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801848/scrEXT330_mm10_S1_L001_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801852/scrEXT330_mm10_S31_L001_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801842/scrEXT330_mm10_S5_L002_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801846/scrEXT330_mm10_S1_L003_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801843/scrEXT330_mm10_S5_L001_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801848/scrEXT330_mm10_S1_L001_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801852/scrEXT330_mm10_S31_L001_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801848/scrEXT330_mm10_S1_L001_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801862/scrEXT330_mm10_S1_L004_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801842/scrEXT330_mm10_S5_L002_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801846/scrEXT330_mm10_S1_L003_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801855/scrEXT331_mm10_S2_L004_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801849/scrEXT331_mm10_S6_L002_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801857/scrEXT331_mm10_S2_L003_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801863/scrEXT331_mm10_S32_L002_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801857/scrEXT331_mm10_S2_L003_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801850/scrEXT331_mm10_S6_L001_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801858/scrEXT331_mm10_S2_L001_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801849/scrEXT331_mm10_S6_L002_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801859/scrEXT331_mm10_S2_L002_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801863/scrEXT331_mm10_S32_L002_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801859/scrEXT331_mm10_S2_L002_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801853/scrEXT331_mm10_S32_L001_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801855/scrEXT331_mm10_S2_L004_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801850/scrEXT331_mm10_S6_L001_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801859/scrEXT331_mm10_S2_L002_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801863/scrEXT331_mm10_S32_L002_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801855/scrEXT331_mm10_S2_L004_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801857/scrEXT331_mm10_S2_L003_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801850/scrEXT331_mm10_S6_L001_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801858/scrEXT331_mm10_S2_L001_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801853/scrEXT331_mm10_S32_L001_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801849/scrEXT331_mm10_S6_L002_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801858/scrEXT331_mm10_S2_L001_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801853/scrEXT331_mm10_S32_L001_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801841/scrEXT352_mm10_S15_L001_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801844/scrEXT352_mm10_S8_L002_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801845/scrEXT352_mm10_S8_L001_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801844/scrEXT352_mm10_S8_L002_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801841/scrEXT352_mm10_S15_L001_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801841/scrEXT352_mm10_S15_L001_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801845/scrEXT352_mm10_S8_L001_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801854/scrEXT352_mm10_S15_L002_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801854/scrEXT352_mm10_S15_L002_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801854/scrEXT352_mm10_S15_L002_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801845/scrEXT352_mm10_S8_L001_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801844/scrEXT352_mm10_S8_L002_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801847/scrEXT353_mm10_S16_L001_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801851/scrEXT353_mm10_S16_L002_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801840/scrEXT353_mm10_S9_L002_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801847/scrEXT353_mm10_S16_L001_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801860/scrEXT353_mm10_S9_L001_R2_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801847/scrEXT353_mm10_S16_L001_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801847/scrEXT353_mm10_S16_L001_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801851/scrEXT353_mm10_S16_L002_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801840/scrEXT353_mm10_S9_L002_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801851/scrEXT353_mm10_S16_L002_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801860/scrEXT353_mm10_S9_L001_I1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801860/scrEXT353_mm10_S9_L001_R1_001.fastq.gz\",\n",
        "    \"ftp://ftp.sra.ebi.ac.uk/vol1/run/ERR138/ERR13801840/scrEXT353_mm10_S9_L002_I1_001.fastq.gz\"\n",
        "]\n",
        "ftp_urls = list(set(ftp_links))"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 418
        },
        "id": "yNGzyAxuzvUz",
        "outputId": "da0bba1f-1a6d-4b4a-a67c-5e041db5c5f2"
      },
      "outputs": [
        {
          "ename": "KeyboardInterrupt",
          "evalue": "",
          "output_type": "error",
          "traceback": [
            "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
            "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
            "\u001b[0;32m<ipython-input-8-7755e01c3ecb>\u001b[0m in \u001b[0;36mmain\u001b[0;34m()\u001b[0m\n\u001b[1;32m     82\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 83\u001b[0;31m                 \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfuture\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mresult\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     84\u001b[0m                 \u001b[0mresults\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mresult\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/lib/python3.10/concurrent/futures/_base.py\u001b[0m in \u001b[0;36mresult\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    452\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 453\u001b[0;31m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_condition\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwait\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    454\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/lib/python3.10/threading.py\u001b[0m in \u001b[0;36mwait\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    319\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mtimeout\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 320\u001b[0;31m                 \u001b[0mwaiter\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0macquire\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    321\u001b[0m                 \u001b[0mgotit\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;31mKeyboardInterrupt\u001b[0m: ",
            "\nDuring handling of the above exception, another exception occurred:\n",
            "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
            "\u001b[0;32m<ipython-input-8-7755e01c3ecb>\u001b[0m in \u001b[0;36m<cell line: 104>\u001b[0;34m()\u001b[0m\n\u001b[1;32m    103\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    104\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0m__name__\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m\"__main__\"\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 105\u001b[0;31m     \u001b[0mmain\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
            "\u001b[0;32m<ipython-input-8-7755e01c3ecb>\u001b[0m in \u001b[0;36mmain\u001b[0;34m()\u001b[0m\n\u001b[1;32m     76\u001b[0m     \u001b[0mlogging\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0minfo\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34mf\"Starting download of {len(ftp_urls)} files...\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     77\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 78\u001b[0;31m     \u001b[0;32mwith\u001b[0m \u001b[0mThreadPoolExecutor\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmax_workers\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmax_workers\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mexecutor\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     79\u001b[0m         \u001b[0mfuture_to_url\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0mexecutor\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msubmit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdownload_file\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0murl\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0murl\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mftp_urls\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     80\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/lib/python3.10/concurrent/futures/_base.py\u001b[0m in \u001b[0;36m__exit__\u001b[0;34m(self, exc_type, exc_val, exc_tb)\u001b[0m\n\u001b[1;32m    647\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    648\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__exit__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mexc_type\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mexc_val\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mexc_tb\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 649\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshutdown\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mwait\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    650\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0;32mFalse\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    651\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/lib/python3.10/concurrent/futures/thread.py\u001b[0m in \u001b[0;36mshutdown\u001b[0;34m(self, wait, cancel_futures)\u001b[0m\n\u001b[1;32m    233\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mwait\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    234\u001b[0m             \u001b[0;32mfor\u001b[0m \u001b[0mt\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_threads\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 235\u001b[0;31m                 \u001b[0mt\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    236\u001b[0m     \u001b[0mshutdown\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__doc__\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_base\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mExecutor\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshutdown\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__doc__\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/lib/python3.10/threading.py\u001b[0m in \u001b[0;36mjoin\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m   1094\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1095\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mtimeout\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1096\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_wait_for_tstate_lock\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1097\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1098\u001b[0m             \u001b[0;31m# the behavior of a negative timeout isn't documented, but\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/lib/python3.10/threading.py\u001b[0m in \u001b[0;36m_wait_for_tstate_lock\u001b[0;34m(self, block, timeout)\u001b[0m\n\u001b[1;32m   1114\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1115\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1116\u001b[0;31m             \u001b[0;32mif\u001b[0m \u001b[0mlock\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0macquire\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mblock\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1117\u001b[0m                 \u001b[0mlock\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrelease\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1118\u001b[0m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_stop\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
          ]
        }
      ],
      "source": [
        "import os\n",
        "import shutil\n",
        "from ftplib import FTP\n",
        "from urllib.parse import urlparse\n",
        "from concurrent.futures import ThreadPoolExecutor\n",
        "import gzip\n",
        "from typing import List, Dict\n",
        "import logging\n",
        "from pathlib import Path\n",
        "import time\n",
        "\n",
        "# Configure logging\n",
        "logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')\n",
        "\n",
        "# Directory setup\n",
        "local_directory = '/content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/'\n",
        "Path(local_directory).mkdir(parents=True, exist_ok=True)\n",
        "\n",
        "# Clear directory contents if it exists\n",
        "for item in os.listdir(local_directory):\n",
        "    item_path = os.path.join(local_directory, item)\n",
        "    if os.path.isfile(item_path):\n",
        "        os.unlink(item_path)\n",
        "    elif os.path.isdir(item_path):\n",
        "        shutil.rmtree(item_path)\n",
        "\n",
        "\n",
        "\n",
        "def download_file(url: str) -> Dict:\n",
        "    \"\"\"\n",
        "    Download a single file from FTP server with error handling and retries\n",
        "    \"\"\"\n",
        "    parsed_url = urlparse(url)\n",
        "    filename = os.path.basename(parsed_url.path)\n",
        "    local_path = os.path.join(local_directory, filename)\n",
        "\n",
        "    max_retries = 3\n",
        "    retry_delay = 5  # seconds\n",
        "\n",
        "    for attempt in range(max_retries):\n",
        "        try:\n",
        "            with FTP(parsed_url.netloc) as ftp:\n",
        "                ftp.login()  # anonymous login\n",
        "\n",
        "                # Download file\n",
        "                with open(local_path, 'wb') as f:\n",
        "                    ftp.retrbinary(f'RETR {parsed_url.path}', f.write)\n",
        "\n",
        "                # Verify gzip integrity\n",
        "                try:\n",
        "                    with gzip.open(local_path, 'rb') as f:\n",
        "                        f.read(1)  # Try reading first byte to verify gzip integrity\n",
        "                except Exception as e:\n",
        "                    os.remove(local_path)  # Remove corrupted file\n",
        "                    raise ValueError(f\"Invalid gzip file: {str(e)}\")\n",
        "\n",
        "                logging.info(f\"Successfully downloaded: {filename}\")\n",
        "                return {\"success\": True, \"filename\": filename}\n",
        "\n",
        "        except Exception as e:\n",
        "            if attempt < max_retries - 1:\n",
        "                logging.warning(f\"Attempt {attempt + 1} failed for {filename}: {str(e)}. Retrying...\")\n",
        "                time.sleep(retry_delay * (attempt + 1))  # Exponential backoff\n",
        "            else:\n",
        "                logging.error(f\"Failed to download {filename} after {max_retries} attempts: {str(e)}\")\n",
        "                # Clean up any partial downloads\n",
        "                if os.path.exists(local_path):\n",
        "                    os.remove(local_path)\n",
        "                return {\"success\": False, \"filename\": filename, \"error\": str(e)}\n",
        "\n",
        "def main():\n",
        "    # Create a thread pool for parallel downloads\n",
        "    max_workers = 4  # Adjust based on available resources\n",
        "    results = []\n",
        "\n",
        "    logging.info(f\"Starting download of {len(ftp_urls)} files...\")\n",
        "\n",
        "    with ThreadPoolExecutor(max_workers=max_workers) as executor:\n",
        "        future_to_url = {executor.submit(download_file, url): url for url in ftp_urls}\n",
        "\n",
        "        for future in future_to_url:\n",
        "            try:\n",
        "                result = future.result()\n",
        "                results.append(result)\n",
        "            except Exception as e:\n",
        "                url = future_to_url[future]\n",
        "                logging.error(f\"Unexpected error for {url}: {str(e)}\")\n",
        "                results.append({\"success\": False, \"filename\": url, \"error\": str(e)})\n",
        "\n",
        "    # Summary\n",
        "    successful = sum(1 for r in results if r[\"success\"])\n",
        "    failed = len(results) - successful\n",
        "\n",
        "    logging.info(f\"\\nDownload Summary:\")\n",
        "    logging.info(f\"Successful downloads: {successful}\")\n",
        "    logging.info(f\"Failed downloads: {failed}\")\n",
        "\n",
        "    if failed > 0:\n",
        "        logging.info(\"\\nFailed files:\")\n",
        "        for result in results:\n",
        "            if not result[\"success\"]:\n",
        "                logging.info(f\"- {result['filename']}: {result.get('error', 'Unknown error')}\")\n",
        "\n",
        "if __name__ == \"__main__\":\n",
        "    main()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "jxdLG0FKKGWI",
        "outputId": "106fc4e1-413e-46e6-e59a-2ef5fdf93d32"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[2025-01-10 09:01:27,697]    INFO [count_nac] Using index index_nac.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202612/ from\n",
            "[2025-01-10 09:01:27,697]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L001_R1_001.fastq.gz\n",
            "[2025-01-10 09:01:27,697]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L001_R2_001.fastq.gz\n",
            "[2025-01-10 09:01:27,697]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L002_R1_001.fastq.gz\n",
            "[2025-01-10 09:01:27,697]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L002_R2_001.fastq.gz\n",
            "[2025-01-10 09:01:27,697]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L003_R1_001.fastq.gz\n",
            "[2025-01-10 09:01:27,697]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L003_R2_001.fastq.gz\n",
            "[2025-01-10 09:01:27,697]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L004_R1_001.fastq.gz\n",
            "[2025-01-10 09:01:27,697]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L004_R2_001.fastq.gz\n",
            "[2025-01-10 09:01:27,697]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S31_L001_R1_001.fastq.gz\n",
            "[2025-01-10 09:01:27,697]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S31_L001_R2_001.fastq.gz\n",
            "[2025-01-10 09:01:27,697]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S31_L002_R1_001.fastq.gz\n",
            "[2025-01-10 09:01:27,697]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S31_L002_R2_001.fastq.gz\n",
            "[2025-01-10 09:01:27,697]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S5_L001_R1_001.fastq.gz\n",
            "[2025-01-10 09:01:27,697]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S5_L001_R2_001.fastq.gz\n",
            "[2025-01-10 09:01:27,697]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S5_L002_R1_001.fastq.gz\n",
            "[2025-01-10 09:01:27,697]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S5_L002_R2_001.fastq.gz\n"
          ]
        }
      ],
      "source": [
        "!kb count -i index_nac.idx -g t2g_nac.txt -c1 cdna.txt -c2 nascent.txt --overwrite -x 10xv3 --workflow=nac --sum=total --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202612/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L003_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L003_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L004_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L004_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S31_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S31_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S31_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S31_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S5_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S5_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S5_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S5_L002_R2_001.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "background_save": true,
          "base_uri": "https://localhost:8080/"
        },
        "id": "rceGqttlKGtN",
        "outputId": "738c5e41-fede-40dc-912e-e00bcf7b41c7"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[2025-01-10 11:22:21,304]    INFO [count_nac] Using index index_nac.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202613/ from\n",
            "[2025-01-10 11:22:21,304]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L001_R1_001.fastq.gz\n",
            "[2025-01-10 11:22:21,304]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L001_R2_001.fastq.gz\n",
            "[2025-01-10 11:22:21,304]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L002_R1_001.fastq.gz\n",
            "[2025-01-10 11:22:21,304]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L002_R2_001.fastq.gz\n",
            "[2025-01-10 11:22:21,304]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L003_R1_001.fastq.gz\n",
            "[2025-01-10 11:22:21,304]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L003_R2_001.fastq.gz\n",
            "[2025-01-10 11:22:21,304]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L004_R1_001.fastq.gz\n",
            "[2025-01-10 11:22:21,304]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L004_R2_001.fastq.gz\n",
            "[2025-01-10 11:22:21,304]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S32_L001_R1_001.fastq.gz\n",
            "[2025-01-10 11:22:21,304]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S32_L001_R2_001.fastq.gz\n",
            "[2025-01-10 11:22:21,304]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S32_L002_R1_001.fastq.gz\n",
            "[2025-01-10 11:22:21,304]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S32_L002_R2_001.fastq.gz\n",
            "[2025-01-10 11:22:21,304]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S6_L001_R1_001.fastq.gz\n",
            "[2025-01-10 11:22:21,304]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S6_L001_R2_001.fastq.gz\n",
            "[2025-01-10 11:22:21,304]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S6_L002_R1_001.fastq.gz\n",
            "[2025-01-10 11:22:21,304]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S6_L002_R2_001.fastq.gz\n",
            "[2025-01-10 12:57:42,172]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202613/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202613/tmp/output.s.bus\n",
            "[2025-01-10 13:01:22,971]    INFO [count_nac] On-list not provided\n",
            "[2025-01-10 13:01:22,973]    INFO [count_nac] Copying pre-packaged 10XV3 on-list to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202613/\n",
            "[2025-01-10 13:01:23,851]    INFO [count_nac] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202613/tmp/output.s.bus\n",
            "[2025-01-10 13:02:08,552]    INFO [count_nac] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202613/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202613/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202613/10x_version3_whitelist.txt\n",
            "[2025-01-10 13:03:42,778]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202613/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202613/output.unfiltered.bus\n",
            "[2025-01-10 13:04:07,845]    INFO [count_nac] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202613/counts_unfiltered/cells_x_genes from BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202613/output.unfiltered.bus\n",
            "[2025-01-10 13:07:52,269]    INFO [count_nac] Writing gene names to file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202613/counts_unfiltered/cells_x_genes.genes.names.txt\n",
            "[2025-01-10 13:07:52,502] WARNING [count_nac] 1599 gene IDs do not have corresponding valid gene names. These genes will use their gene IDs instead.\n",
            "[2025-01-10 13:07:52,540]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202613/counts_unfiltered/cells_x_genes.cell.mtx\n",
            "[2025-01-10 13:08:21,502]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202613/counts_unfiltered/cells_x_genes.cell.mtx\n",
            "[2025-01-10 13:09:02,360]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202613/counts_unfiltered/cells_x_genes.nucleus.mtx\n",
            "[2025-01-10 13:09:36,242]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202613/counts_unfiltered/cells_x_genes.nucleus.mtx\n",
            "[2025-01-10 13:10:25,071]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202613/counts_unfiltered/cells_x_genes.total.mtx\n",
            "[2025-01-10 13:11:02,947]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202613/counts_unfiltered/cells_x_genes.total.mtx\n",
            "[2025-01-10 13:12:16,440]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202613/counts_unfiltered/cells_x_genes.mature.mtx\n",
            "[2025-01-10 13:12:20,604]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202613/counts_unfiltered/cells_x_genes.nascent.mtx\n",
            "[2025-01-10 13:12:24,376]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202613/counts_unfiltered/cells_x_genes.ambiguous.mtx\n",
            "[2025-01-10 13:12:29,032]    INFO [count_nac] Combining matrices\n",
            "[2025-01-10 13:12:32,388]    INFO [count_nac] Writing matrices to h5ad /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202613/counts_unfiltered/adata.h5ad\n"
          ]
        }
      ],
      "source": [
        "!kb count -i index_nac.idx -g t2g_nac.txt -c1 cdna.txt -c2 nascent.txt --overwrite -x 10xv3 --workflow=nac --sum=total --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202613/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L003_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L003_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L004_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L004_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S32_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S32_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S32_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S32_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S6_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S6_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S6_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S6_L002_R2_001.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "KfzIA_FCKGwg",
        "outputId": "eed923ec-0185-4522-c9a2-ceb69e6e7381"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[2025-01-10 13:12:56,404]    INFO [count_nac] Using index index_nac.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202614/ from\n",
            "[2025-01-10 13:12:56,404]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S15_L001_R1_001.fastq.gz\n",
            "[2025-01-10 13:12:56,405]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S15_L001_R2_001.fastq.gz\n",
            "[2025-01-10 13:12:56,405]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S15_L002_R1_001.fastq.gz\n",
            "[2025-01-10 13:12:56,405]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S15_L002_R2_001.fastq.gz\n",
            "[2025-01-10 13:12:56,405]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S8_L001_R1_001.fastq.gz\n",
            "[2025-01-10 13:12:56,405]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S8_L001_R2_001.fastq.gz\n",
            "[2025-01-10 13:12:56,405]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S8_L002_R1_001.fastq.gz\n",
            "[2025-01-10 13:12:56,405]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S8_L002_R2_001.fastq.gz\n"
          ]
        }
      ],
      "source": [
        "!kb count -i index_nac.idx -g t2g_nac.txt -c1 cdna.txt -c2 nascent.txt --overwrite -x 10xv3 --workflow=nac --sum=total --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202614/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S15_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S15_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S15_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S15_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S8_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S8_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S8_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S8_L002_R2_001.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "l47ulFBZKG05",
        "outputId": "ddf95657-7f84-4679-a9b9-a8677ac01ce8"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[2025-01-10 14:21:13,094]    INFO [count_nac] Using index index_nac.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202615/ from\n",
            "[2025-01-10 14:21:13,094]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S16_L001_R1_001.fastq.gz\n",
            "[2025-01-10 14:21:13,094]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S16_L001_R2_001.fastq.gz\n",
            "[2025-01-10 14:21:13,094]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S16_L002_R1_001.fastq.gz\n",
            "[2025-01-10 14:21:13,094]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S16_L002_R2_001.fastq.gz\n",
            "[2025-01-10 14:21:13,094]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S9_L001_R1_001.fastq.gz\n",
            "[2025-01-10 14:21:13,095]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S9_L001_R2_001.fastq.gz\n",
            "[2025-01-10 14:21:13,095]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S9_L002_R1_001.fastq.gz\n",
            "[2025-01-10 14:21:13,095]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S9_L002_R2_001.fastq.gz\n",
            "[2025-01-10 15:36:42,204]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202615/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202615/tmp/output.s.bus\n",
            "[2025-01-10 15:41:13,423]    INFO [count_nac] On-list not provided\n",
            "[2025-01-10 15:41:13,424]    INFO [count_nac] Copying pre-packaged 10XV3 on-list to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202615/\n",
            "[2025-01-10 15:41:14,274]    INFO [count_nac] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202615/tmp/output.s.bus\n",
            "[2025-01-10 15:42:45,111]    INFO [count_nac] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202615/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202615/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202615/10x_version3_whitelist.txt\n",
            "[2025-01-10 15:45:01,717]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202615/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202615/output.unfiltered.bus\n",
            "[2025-01-10 15:48:01,301]    INFO [count_nac] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202615/counts_unfiltered/cells_x_genes from BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202615/output.unfiltered.bus\n",
            "[2025-01-10 15:52:30,458]    INFO [count_nac] Writing gene names to file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202615/counts_unfiltered/cells_x_genes.genes.names.txt\n",
            "[2025-01-10 15:52:30,676] WARNING [count_nac] 1599 gene IDs do not have corresponding valid gene names. These genes will use their gene IDs instead.\n",
            "[2025-01-10 15:52:30,714]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202615/counts_unfiltered/cells_x_genes.cell.mtx\n",
            "[2025-01-10 15:53:34,052]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202615/counts_unfiltered/cells_x_genes.cell.mtx\n",
            "[2025-01-10 15:54:36,223]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202615/counts_unfiltered/cells_x_genes.nucleus.mtx\n",
            "[2025-01-10 15:55:32,782]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202615/counts_unfiltered/cells_x_genes.nucleus.mtx\n",
            "[2025-01-10 15:56:50,588]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202615/counts_unfiltered/cells_x_genes.total.mtx\n",
            "[2025-01-10 15:57:47,003]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202615/counts_unfiltered/cells_x_genes.total.mtx\n",
            "[2025-01-10 15:59:08,486]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202615/counts_unfiltered/cells_x_genes.mature.mtx\n",
            "[2025-01-10 15:59:12,129]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202615/counts_unfiltered/cells_x_genes.nascent.mtx\n",
            "[2025-01-10 15:59:15,907]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202615/counts_unfiltered/cells_x_genes.ambiguous.mtx\n",
            "[2025-01-10 15:59:20,538]    INFO [count_nac] Combining matrices\n",
            "[2025-01-10 15:59:24,686]    INFO [count_nac] Writing matrices to h5ad /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202615/counts_unfiltered/adata.h5ad\n"
          ]
        }
      ],
      "source": [
        "!kb count -i index_nac.idx -g t2g_nac.txt -c1 cdna.txt -c2 nascent.txt --overwrite -x 10xv3 --workflow=nac --sum=total --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX13202615/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S16_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S16_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S16_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S16_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S9_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S9_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S9_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S9_L002_R2_001.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "PmNTmCyN7-0M",
        "outputId": "c4cd1a69-77e6-426c-f691-4acfc2b61746"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[2025-01-11 02:42:05,701]    INFO [count_nac] Using index index_nac.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265374/ from\n",
            "[2025-01-11 02:42:05,701]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT356_mm10_S19_L001_R1_001.fastq.gz\n",
            "[2025-01-11 02:42:05,701]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT356_mm10_S19_L001_R2_001.fastq.gz\n",
            "[2025-01-11 02:42:05,701]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT356_mm10_S19_L002_R1_001.fastq.gz\n",
            "[2025-01-11 02:42:05,701]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT356_mm10_S19_L002_R2_001.fastq.gz\n",
            "[2025-01-11 02:42:05,701]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT356_mm10_S3_L001_R1_001.fastq.gz\n",
            "[2025-01-11 02:42:05,701]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT356_mm10_S3_L001_R2_001.fastq.gz\n",
            "[2025-01-11 03:41:58,760]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265374/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265374/tmp/output.s.bus\n",
            "[2025-01-11 03:47:00,571]    INFO [count_nac] On-list not provided\n",
            "[2025-01-11 03:47:00,572]    INFO [count_nac] Copying pre-packaged 10XV3 on-list to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265374/\n",
            "[2025-01-11 03:47:01,331]    INFO [count_nac] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265374/tmp/output.s.bus\n",
            "[2025-01-11 03:48:04,547]    INFO [count_nac] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265374/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265374/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265374/10x_version3_whitelist.txt\n",
            "[2025-01-11 03:50:03,271]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265374/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265374/output.unfiltered.bus\n",
            "[2025-01-11 03:53:36,429]    INFO [count_nac] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265374/counts_unfiltered/cells_x_genes from BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265374/output.unfiltered.bus\n",
            "[2025-01-11 03:57:33,033]    INFO [count_nac] Writing gene names to file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265374/counts_unfiltered/cells_x_genes.genes.names.txt\n",
            "[2025-01-11 03:57:33,229] WARNING [count_nac] 1599 gene IDs do not have corresponding valid gene names. These genes will use their gene IDs instead.\n",
            "[2025-01-11 03:57:33,260]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265374/counts_unfiltered/cells_x_genes.cell.mtx\n",
            "[2025-01-11 03:58:42,013]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265374/counts_unfiltered/cells_x_genes.cell.mtx\n",
            "[2025-01-11 04:00:14,594]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265374/counts_unfiltered/cells_x_genes.nucleus.mtx\n",
            "[2025-01-11 04:01:16,014]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265374/counts_unfiltered/cells_x_genes.nucleus.mtx\n",
            "[2025-01-11 04:02:39,628]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265374/counts_unfiltered/cells_x_genes.total.mtx\n",
            "[2025-01-11 04:03:55,358]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265374/counts_unfiltered/cells_x_genes.total.mtx\n",
            "[2025-01-11 04:05:36,366]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265374/counts_unfiltered/cells_x_genes.mature.mtx\n",
            "[2025-01-11 04:05:41,679]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265374/counts_unfiltered/cells_x_genes.nascent.mtx\n",
            "[2025-01-11 04:05:50,660]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265374/counts_unfiltered/cells_x_genes.ambiguous.mtx\n",
            "[2025-01-11 04:06:06,752]    INFO [count_nac] Combining matrices\n",
            "[2025-01-11 04:06:17,661]    INFO [count_nac] Writing matrices to h5ad /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265374/counts_unfiltered/adata.h5ad\n"
          ]
        }
      ],
      "source": [
        "!kb count -i index_nac.idx -g t2g_nac.txt -c1 cdna.txt -c2 nascent.txt --overwrite -x 10xv3 --workflow=nac --sum=total --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265374/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT356_mm10_S19_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT356_mm10_S19_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT356_mm10_S19_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT356_mm10_S19_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT356_mm10_S3_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT356_mm10_S3_L001_R2_001.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "JG-JzeOT7-5s",
        "outputId": "4ba80e88-da28-4584-8c7c-ea7372e713a6"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[2025-01-10 21:59:15,671]    INFO [count_nac] Using index index_nac.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265375/ from\n",
            "[2025-01-10 21:59:15,671]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT357_mm10_S20_L001_R1_001.fastq.gz\n",
            "[2025-01-10 21:59:15,671]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT357_mm10_S20_L001_R2_001.fastq.gz\n",
            "[2025-01-10 21:59:15,671]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT357_mm10_S20_L002_R1_001.fastq.gz\n",
            "[2025-01-10 21:59:15,671]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT357_mm10_S20_L002_R2_001.fastq.gz\n",
            "[2025-01-10 21:59:15,671]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT357_mm10_S4_L001_R1_001.fastq.gz\n",
            "[2025-01-10 21:59:15,672]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT357_mm10_S4_L001_R2_001.fastq.gz\n",
            "[2025-01-10 22:45:36,542]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265375/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265375/tmp/output.s.bus\n",
            "[2025-01-10 22:49:45,264]    INFO [count_nac] On-list not provided\n",
            "[2025-01-10 22:49:45,265]    INFO [count_nac] Copying pre-packaged 10XV3 on-list to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265375/\n",
            "[2025-01-10 22:49:46,097]    INFO [count_nac] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265375/tmp/output.s.bus\n",
            "[2025-01-10 22:50:46,121]    INFO [count_nac] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265375/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265375/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265375/10x_version3_whitelist.txt\n",
            "[2025-01-10 22:52:50,390]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265375/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265375/output.unfiltered.bus\n",
            "[2025-01-10 22:56:22,544]    INFO [count_nac] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265375/counts_unfiltered/cells_x_genes from BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265375/output.unfiltered.bus\n",
            "[2025-01-10 22:59:59,554]    INFO [count_nac] Writing gene names to file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265375/counts_unfiltered/cells_x_genes.genes.names.txt\n",
            "[2025-01-10 22:59:59,813] WARNING [count_nac] 1599 gene IDs do not have corresponding valid gene names. These genes will use their gene IDs instead.\n",
            "[2025-01-10 22:59:59,855]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265375/counts_unfiltered/cells_x_genes.cell.mtx\n",
            "[2025-01-10 23:01:06,098]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265375/counts_unfiltered/cells_x_genes.cell.mtx\n",
            "[2025-01-10 23:02:38,596]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265375/counts_unfiltered/cells_x_genes.nucleus.mtx\n",
            "[2025-01-10 23:03:35,549]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265375/counts_unfiltered/cells_x_genes.nucleus.mtx\n",
            "[2025-01-10 23:04:56,801]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265375/counts_unfiltered/cells_x_genes.total.mtx\n",
            "[2025-01-10 23:06:27,868]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265375/counts_unfiltered/cells_x_genes.total.mtx\n",
            "[2025-01-10 23:08:08,953]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265375/counts_unfiltered/cells_x_genes.mature.mtx\n",
            "[2025-01-10 23:08:13,094]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265375/counts_unfiltered/cells_x_genes.nascent.mtx\n",
            "[2025-01-10 23:08:16,477]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265375/counts_unfiltered/cells_x_genes.ambiguous.mtx\n",
            "[2025-01-10 23:08:23,344]    INFO [count_nac] Combining matrices\n",
            "[2025-01-10 23:08:27,974]    INFO [count_nac] Writing matrices to h5ad /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265375/counts_unfiltered/adata.h5ad\n"
          ]
        }
      ],
      "source": [
        "!kb count -i index_nac.idx -g t2g_nac.txt -c1 cdna.txt -c2 nascent.txt --overwrite -x 10xv3 --workflow=nac --sum=total --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265375/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT357_mm10_S20_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT357_mm10_S20_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT357_mm10_S20_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT357_mm10_S20_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT357_mm10_S4_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT357_mm10_S4_L001_R2_001.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "-chmahnN7_A3",
        "outputId": "b1631e8b-7e8c-4f9c-f984-e579fdc79045"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[2025-01-10 23:08:55,659]    INFO [count_nac] Using index index_nac.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265376/ from\n",
            "[2025-01-10 23:08:55,659]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT358_mm10_S11_L001_R1_001.fastq.gz\n",
            "[2025-01-10 23:08:55,659]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT358_mm10_S11_L001_R2_001.fastq.gz\n",
            "[2025-01-11 00:37:02,355]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265376/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265376/tmp/output.s.bus\n",
            "[2025-01-11 00:44:09,788]    INFO [count_nac] On-list not provided\n",
            "[2025-01-11 00:44:09,789]    INFO [count_nac] Copying pre-packaged 10XV3 on-list to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265376/\n",
            "[2025-01-11 00:44:10,627]    INFO [count_nac] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265376/tmp/output.s.bus\n",
            "[2025-01-11 00:46:03,473]    INFO [count_nac] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265376/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265376/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265376/10x_version3_whitelist.txt\n",
            "[2025-01-11 00:49:14,433]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265376/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265376/output.unfiltered.bus\n",
            "[2025-01-11 00:55:17,055]    INFO [count_nac] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265376/counts_unfiltered/cells_x_genes from BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265376/output.unfiltered.bus\n",
            "[2025-01-11 00:59:51,605]    INFO [count_nac] Writing gene names to file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265376/counts_unfiltered/cells_x_genes.genes.names.txt\n",
            "[2025-01-11 00:59:51,834] WARNING [count_nac] 1599 gene IDs do not have corresponding valid gene names. These genes will use their gene IDs instead.\n",
            "[2025-01-11 00:59:51,880]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265376/counts_unfiltered/cells_x_genes.cell.mtx\n",
            "[2025-01-11 01:00:46,991]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265376/counts_unfiltered/cells_x_genes.cell.mtx\n",
            "[2025-01-11 01:02:01,739]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265376/counts_unfiltered/cells_x_genes.nucleus.mtx\n",
            "[2025-01-11 01:02:57,232]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265376/counts_unfiltered/cells_x_genes.nucleus.mtx\n",
            "[2025-01-11 01:04:29,471]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265376/counts_unfiltered/cells_x_genes.total.mtx\n",
            "[2025-01-11 01:05:31,042]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265376/counts_unfiltered/cells_x_genes.total.mtx\n",
            "[2025-01-11 01:06:56,177]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265376/counts_unfiltered/cells_x_genes.mature.mtx\n",
            "[2025-01-11 01:07:00,868]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265376/counts_unfiltered/cells_x_genes.nascent.mtx\n",
            "[2025-01-11 01:07:05,030]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265376/counts_unfiltered/cells_x_genes.ambiguous.mtx\n",
            "[2025-01-11 01:07:10,779]    INFO [count_nac] Combining matrices\n",
            "[2025-01-11 01:07:14,992]    INFO [count_nac] Writing matrices to h5ad /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265376/counts_unfiltered/adata.h5ad\n"
          ]
        }
      ],
      "source": [
        "!kb count -i index_nac.idx -g t2g_nac.txt -c1 cdna.txt -c2 nascent.txt --overwrite -x 10xv3 --workflow=nac --sum=total --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265376/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT358_mm10_S11_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT358_mm10_S11_L001_R2_001.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ncVn5EgH7_JF",
        "outputId": "b5db960a-25ab-4cd9-e9c0-417af586a9e3"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[2025-01-11 01:07:42,443]    INFO [count_nac] Using index index_nac.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265377/ from\n",
            "[2025-01-11 01:07:42,443]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S10_L001_R1_001.fastq.gz\n",
            "[2025-01-11 01:07:42,443]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S10_L001_R2_001.fastq.gz\n",
            "[2025-01-11 01:07:42,443]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S10_L002_R1_001.fastq.gz\n",
            "[2025-01-11 01:07:42,443]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S10_L002_R2_001.fastq.gz\n",
            "[2025-01-11 01:07:42,443]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S6_L001_R1_001.fastq.gz\n",
            "[2025-01-11 01:07:42,444]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S6_L001_R2_001.fastq.gz\n",
            "[2025-01-11 01:07:42,444]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S6_L002_R1_001.fastq.gz\n",
            "[2025-01-11 01:07:42,444]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S6_L002_R2_001.fastq.gz\n",
            "[2025-01-11 01:07:42,444]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S6_L003_R1_001.fastq.gz\n",
            "[2025-01-11 01:07:42,444]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S6_L003_R2_001.fastq.gz\n",
            "[2025-01-11 01:07:42,444]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S6_L004_R1_001.fastq.gz\n",
            "[2025-01-11 01:07:42,444]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S6_L004_R2_001.fastq.gz\n",
            "[2025-01-11 02:12:57,101]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265377/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265377/tmp/output.s.bus\n",
            "[2025-01-11 02:19:00,200]    INFO [count_nac] On-list not provided\n",
            "[2025-01-11 02:19:00,201]    INFO [count_nac] Copying pre-packaged 10XV3 on-list to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265377/\n",
            "[2025-01-11 02:19:01,028]    INFO [count_nac] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265377/tmp/output.s.bus\n",
            "[2025-01-11 02:20:45,942]    INFO [count_nac] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265377/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265377/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265377/10x_version3_whitelist.txt\n",
            "[2025-01-11 02:23:31,312]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265377/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265377/output.unfiltered.bus\n",
            "[2025-01-11 02:28:18,300]    INFO [count_nac] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265377/counts_unfiltered/cells_x_genes from BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265377/output.unfiltered.bus\n",
            "[2025-01-11 02:33:24,730]    INFO [count_nac] Writing gene names to file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265377/counts_unfiltered/cells_x_genes.genes.names.txt\n",
            "[2025-01-11 02:33:24,950] WARNING [count_nac] 1599 gene IDs do not have corresponding valid gene names. These genes will use their gene IDs instead.\n",
            "[2025-01-11 02:33:24,992]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265377/counts_unfiltered/cells_x_genes.cell.mtx\n",
            "[2025-01-11 02:34:26,443]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265377/counts_unfiltered/cells_x_genes.cell.mtx\n",
            "[2025-01-11 02:35:48,402]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265377/counts_unfiltered/cells_x_genes.nucleus.mtx\n",
            "[2025-01-11 02:36:49,317]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265377/counts_unfiltered/cells_x_genes.nucleus.mtx\n",
            "[2025-01-11 02:38:11,983]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265377/counts_unfiltered/cells_x_genes.total.mtx\n",
            "[2025-01-11 02:39:20,601]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265377/counts_unfiltered/cells_x_genes.total.mtx\n",
            "[2025-01-11 02:41:16,958]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265377/counts_unfiltered/cells_x_genes.mature.mtx\n",
            "[2025-01-11 02:41:21,473]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265377/counts_unfiltered/cells_x_genes.nascent.mtx\n",
            "[2025-01-11 02:41:25,063]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265377/counts_unfiltered/cells_x_genes.ambiguous.mtx\n",
            "[2025-01-11 02:41:31,826]    INFO [count_nac] Combining matrices\n",
            "[2025-01-11 02:41:35,964]    INFO [count_nac] Writing matrices to h5ad /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265377/counts_unfiltered/adata.h5ad\n"
          ]
        }
      ],
      "source": [
        "!kb count -i index_nac.idx -g t2g_nac.txt -c1 cdna.txt -c2 nascent.txt --overwrite -x 10xv3 --workflow=nac --sum=total --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX12265377/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S10_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S10_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S10_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S10_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S6_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S6_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S6_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S6_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S6_L003_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S6_L003_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S6_L004_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S6_L004_R2_001.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "nGbZJWU-lx-r"
      },
      "outputs": [],
      "source": []
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "EDOZsI3RlyBe"
      },
      "outputs": [],
      "source": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "WpBuVgFPbh7W"
      },
      "source": [
        "# Found strange fastq files in Vennekens et al. (2021) using SRA so getting files from arrayexpress instead."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "q6KHWOEZ1EYg",
        "outputId": "a5599ccb-b7c8-41ac-a7c2-8ed67e036e58"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/?C=N;O=D...Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/?C=S;O=A...Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/?C=M;O=A...\n",
            "\n",
            "\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/?C=D;O=A...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/E-MTAB-10021.sdrf.txt...Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/E-MTAB-10021.idf.txt...\n",
            "\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT036_S9_L001_I1_001.fastq.gz...Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT036_S9_L001_R1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT036_S9_L001_R2_001.fastq.gz...\n",
            "\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT036_S9_L002_I1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT036_S9_L002_R1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT036_S9_L002_R2_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT036_mm10_S9_L003_I1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT036_mm10_S9_L003_R1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT036_mm10_S9_L003_R2_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT036_mm10_S9_L004_I1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT036_mm10_S9_L004_R1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT036_mm10_S9_L004_R2_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT036_mm10_S9_L005_I1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT036_mm10_S9_L005_R1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT036_mm10_S9_L005_R2_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT036_mm10_S9_L006_I1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT036_mm10_S9_L006_R1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT036_mm10_S9_L006_R2_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT037_S10_L001_I1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT037_S10_L001_R1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT037_S10_L001_R2_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT037_S10_L002_I1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT037_S10_L002_R1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT037_S10_L002_R2_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT037_mm10_S10_L003_I1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT037_mm10_S10_L003_R1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT037_mm10_S10_L003_R2_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT037_mm10_S10_L004_I1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT037_mm10_S10_L004_R1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT037_mm10_S10_L004_R2_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT037_mm10_S10_L005_I1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT037_mm10_S10_L005_R1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT037_mm10_S10_L005_R2_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT037_mm10_S10_L006_I1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT037_mm10_S10_L006_R1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT037_mm10_S10_L006_R2_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT144_mm10_S13_L001_I1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT144_mm10_S13_L001_R1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT144_mm10_S13_L001_R2_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT144_mm10_S13_L002_I1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT144_mm10_S13_L002_R1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT144_mm10_S13_L002_R2_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT144_mm10_S27_L001_I1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT144_mm10_S27_L001_R1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT144_mm10_S27_L001_R2_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT144_mm10_S27_L002_I1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT144_mm10_S27_L002_R1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT144_mm10_S27_L002_R2_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT144_mm10_S27_L003_I1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT144_mm10_S27_L003_R1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT144_mm10_S27_L003_R2_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT144_mm10_S27_L004_I1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT144_mm10_S27_L004_R1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT144_mm10_S27_L004_R2_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT145_mm10_S9_L001_I1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT145_mm10_S9_L001_R1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT145_mm10_S9_L001_R2_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT145_mm10_S9_L002_I1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT145_mm10_S9_L002_R1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT145_mm10_S9_L002_R2_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT145_mm10_S14_L001_I1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT145_mm10_S14_L001_R1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT145_mm10_S14_L001_R2_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT145_mm10_S14_L002_I1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT145_mm10_S14_L002_R1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT145_mm10_S14_L002_R2_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT145_mm10_S28_L001_I1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT145_mm10_S28_L001_R1_001.fastq.gz...\n",
            "Downloading https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/scrEXT145_mm10_S28_L001_R2_001.fastq.gz...\n"
          ]
        }
      ],
      "source": [
        "import os\n",
        "import requests\n",
        "from concurrent.futures import ThreadPoolExecutor\n",
        "from bs4 import BeautifulSoup\n",
        "\n",
        "# Step 2: Create a local directory to store the downloaded files\n",
        "local_directory = '/content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs'\n",
        "os.makedirs(local_directory, exist_ok=True)\n",
        "\n",
        "# Step 3: Define the download function and use parallel processing\n",
        "base_url = \"https://ftp.ebi.ac.uk/biostudies/fire/E-MTAB-/021/E-MTAB-10021/Files/\"\n",
        "\n",
        "def get_file_links(url):\n",
        "    \"\"\"Fetches all file links from the given directory URL.\"\"\"\n",
        "    response = requests.get(url)\n",
        "    soup = BeautifulSoup(response.content, 'html.parser')\n",
        "    links = [link.get('href') for link in soup.find_all('a') if link.get('href')]\n",
        "    return [url + link for link in links if not link.endswith('/')]\n",
        "\n",
        "def download_file(url):\n",
        "    \"\"\"Downloads a single file from the given URL.\"\"\"\n",
        "    local_path = os.path.join(local_directory, os.path.basename(url))\n",
        "    if not os.path.exists(local_path):\n",
        "        print(f\"Downloading {url}...\")\n",
        "        response = requests.get(url, stream=True)\n",
        "        with open(local_path, 'wb') as f:\n",
        "            for chunk in response.iter_content(chunk_size=8192):\n",
        "                if chunk:\n",
        "                    f.write(chunk)\n",
        "    else:\n",
        "        print(f\"{local_path} already exists, skipping download.\")\n",
        "\n",
        "# Get all file links\n",
        "file_links = get_file_links(base_url)\n",
        "\n",
        "# Use ThreadPoolExecutor to download files in parallel\n",
        "with ThreadPoolExecutor(max_workers=10) as executor:\n",
        "    executor.map(download_file, file_links)\n",
        "\n",
        "# Verify the download\n",
        "# List the downloaded files\n",
        "downloaded_files = os.listdir(local_directory)\n",
        "downloaded_files\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "PTz8CfkVMXDe"
      },
      "source": [
        "# Aligning these Vennekens et al files now"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "dpA0pPlHv64v"
      },
      "outputs": [],
      "source": [
        "import subprocess\n",
        "import shutil\n",
        "import os\n",
        "for SRA_ID in [\"ERX4978537\",\"ERX4978538\",\"ERX4978539\",\"ERX4978540\",\n",
        "               \"ERX4978541\",\"ERX4978542\",\"ERX4978543\",\"ERX4978544\"]:\n",
        "#if no folder in nac\n",
        "  if not os.path.exists('/content/drive/MyDrive/pituitary_atlas/processed/nac/' + SRA_ID):\n",
        "    os.mkdir('/content/drive/MyDrive/pituitary_atlas/processed/nac/' + SRA_ID)\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "aiXuCaKAMa4v"
      },
      "source": [
        "# We can see it is v2 library."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "hICViSXszFEq",
        "outputId": "9cd5616e-bbc4-4822-a84f-b145bc12178d"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "@NB501171:232:HJL3KBGX7:3:11401:5056:1022 1:N:0:GNTCCTCA\n",
            "GCAATCAAGCACCGTCACCGCCCCTG\n",
            "+\n",
            "AAAAAEEEEAAEEEEE/E/EAE///E\n",
            "@NB501171:232:HJL3KBGX7:3:11401:20483:1023 1:N:0:GNTCCTCA\n",
            "GTAACGTAGCGCCTTGTAATCCGTAT\n",
            "+\n",
            "AAAAAEEEEEAEAEEEEEAEEEEEEE\n",
            "@NB501171:232:HJL3KBGX7:3:11401:9431:1027 1:N:0:GTTCCTCA\n",
            "CCACGGATCCACTCCATTTTTCCCTT\n"
          ]
        }
      ],
      "source": [
        "!gunzip /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_mm10_S10_L003_R1_001.fastq.gz\n",
        "!head /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_mm10_S10_L003_R1_001.fastq"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "QWPhciuHv6E1",
        "outputId": "cb2ff4b1-d06a-4fb4-8cd5-453ee30cb847"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[2024-07-11 15:03:42,972]    INFO [count_nac] Using index index_nac.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978539/ from\n",
            "[2024-07-11 15:03:42,973]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S13_L001_R1_001.fastq.gz\n",
            "[2024-07-11 15:03:42,973]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S13_L001_R2_001.fastq.gz\n",
            "[2024-07-11 15:03:42,973]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S13_L002_R1_001.fastq.gz\n",
            "[2024-07-11 15:03:42,973]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S13_L002_R2_001.fastq.gz\n",
            "[2024-07-11 15:03:42,973]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L001_R1_001.fastq.gz\n",
            "[2024-07-11 15:03:42,973]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L001_R2_001.fastq.gz\n",
            "[2024-07-11 15:03:42,973]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L002_R1_001.fastq.gz\n",
            "[2024-07-11 15:03:42,973]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L002_R2_001.fastq.gz\n",
            "[2024-07-11 15:03:42,973]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L003_R1_001.fastq.gz\n",
            "[2024-07-11 15:03:42,973]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L003_R2_001.fastq.gz\n",
            "[2024-07-11 15:03:42,973]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L004_R1_001.fastq.gz\n",
            "[2024-07-11 15:03:42,973]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L004_R2_001.fastq.gz\n",
            "[2024-07-11 20:22:23,334]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978539/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978539/tmp/output.s.bus\n",
            "[2024-07-11 20:32:02,939]    INFO [count_nac] On-list not provided\n",
            "[2024-07-11 20:32:02,942]    INFO [count_nac] Copying pre-packaged 10XV2 on-list to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978539/\n",
            "[2024-07-11 20:32:03,112]    INFO [count_nac] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978539/tmp/output.s.bus\n",
            "[2024-07-11 20:32:44,222]    INFO [count_nac] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978539/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978539/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978539/10x_version2_whitelist.txt\n",
            "[2024-07-11 20:34:22,273]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978539/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978539/output.unfiltered.bus\n",
            "[2024-07-11 20:34:47,037]    INFO [count_nac] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978539/counts_unfiltered/cells_x_genes from BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978539/output.unfiltered.bus\n",
            "[2024-07-11 20:41:33,788]    INFO [count_nac] Writing gene names to file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978539/counts_unfiltered/cells_x_genes.genes.names.txt\n",
            "[2024-07-11 20:41:34,233] WARNING [count_nac] 1599 gene IDs do not have corresponding valid gene names. These genes will use their gene IDs instead.\n",
            "[2024-07-11 20:41:34,304]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978539/counts_unfiltered/cells_x_genes.cell.mtx\n",
            "[2024-07-11 20:42:18,425]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978539/counts_unfiltered/cells_x_genes.cell.mtx\n",
            "[2024-07-11 20:43:23,094]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978539/counts_unfiltered/cells_x_genes.nucleus.mtx\n",
            "[2024-07-11 20:44:16,635]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978539/counts_unfiltered/cells_x_genes.nucleus.mtx\n",
            "[2024-07-11 20:45:30,207]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978539/counts_unfiltered/cells_x_genes.total.mtx\n",
            "[2024-07-11 20:46:25,204]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978539/counts_unfiltered/cells_x_genes.total.mtx\n",
            "[2024-07-11 20:47:39,051]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978539/counts_unfiltered/cells_x_genes.mature.mtx\n",
            "[2024-07-11 20:47:45,197]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978539/counts_unfiltered/cells_x_genes.nascent.mtx\n",
            "[2024-07-11 20:47:58,403]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978539/counts_unfiltered/cells_x_genes.ambiguous.mtx\n",
            "[2024-07-11 20:48:42,259]    INFO [count_nac] Combining matrices\n",
            "[2024-07-11 20:48:51,994]    INFO [count_nac] Writing matrices to h5ad /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978539/counts_unfiltered/adata.h5ad\n",
            "[2024-07-11 20:49:21,893]    INFO [count_nac] Using index index_nac.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978540/ from\n",
            "[2024-07-11 20:49:21,893]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S14_L001_R1_001.fastq.gz\n",
            "[2024-07-11 20:49:21,893]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S14_L001_R2_001.fastq.gz\n",
            "[2024-07-11 20:49:21,893]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S14_L002_R1_001.fastq.gz\n",
            "[2024-07-11 20:49:21,893]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S14_L002_R2_001.fastq.gz\n",
            "[2024-07-11 20:49:21,893]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L001_R1_001.fastq.gz\n",
            "[2024-07-11 20:49:21,893]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L001_R2_001.fastq.gz\n",
            "[2024-07-11 20:49:21,893]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L002_R1_001.fastq.gz\n",
            "[2024-07-11 20:49:21,893]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L002_R2_001.fastq.gz\n",
            "[2024-07-11 20:49:21,893]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L003_R1_001.fastq.gz\n",
            "[2024-07-11 20:49:21,893]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L003_R2_001.fastq.gz\n",
            "[2024-07-11 20:49:21,893]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L004_R1_001.fastq.gz\n",
            "[2024-07-11 20:49:21,893]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L004_R2_001.fastq.gz\n",
            "[2024-07-11 20:49:21,893]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S9_L001_R1_001.fastq.gz\n",
            "[2024-07-11 20:49:21,893]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S9_L001_R2_001.fastq.gz\n",
            "[2024-07-11 20:49:21,893]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S9_L002_R1_001.fastq.gz\n",
            "[2024-07-11 20:49:21,893]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S9_L002_R2_001.fastq.gz\n",
            "[2024-07-11 21:11:56,879]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978540/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978540/tmp/output.s.bus\n",
            "[2024-07-11 21:12:15,819]    INFO [count_nac] On-list not provided\n",
            "[2024-07-11 21:12:15,819]    INFO [count_nac] Copying pre-packaged 10XV2 on-list to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978540/\n",
            "[2024-07-11 21:12:15,996]    INFO [count_nac] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978540/tmp/output.s.bus\n",
            "[2024-07-11 21:12:24,417]    INFO [count_nac] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978540/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978540/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978540/10x_version2_whitelist.txt\n",
            "[2024-07-11 21:12:50,074]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978540/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978540/output.unfiltered.bus\n",
            "[2024-07-11 21:13:04,712]    INFO [count_nac] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978540/counts_unfiltered/cells_x_genes from BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978540/output.unfiltered.bus\n",
            "[2024-07-11 21:15:09,039]    INFO [count_nac] Writing gene names to file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978540/counts_unfiltered/cells_x_genes.genes.names.txt\n",
            "[2024-07-11 21:15:09,375] WARNING [count_nac] 1599 gene IDs do not have corresponding valid gene names. These genes will use their gene IDs instead.\n",
            "[2024-07-11 21:15:09,418]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978540/counts_unfiltered/cells_x_genes.cell.mtx\n",
            "[2024-07-11 21:15:25,399]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978540/counts_unfiltered/cells_x_genes.cell.mtx\n",
            "[2024-07-11 21:15:47,492]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978540/counts_unfiltered/cells_x_genes.nucleus.mtx\n",
            "[2024-07-11 21:16:06,897]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978540/counts_unfiltered/cells_x_genes.nucleus.mtx\n",
            "[2024-07-11 21:16:33,280]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978540/counts_unfiltered/cells_x_genes.total.mtx\n",
            "[2024-07-11 21:16:53,838]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978540/counts_unfiltered/cells_x_genes.total.mtx\n",
            "[2024-07-11 21:17:22,054]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978540/counts_unfiltered/cells_x_genes.mature.mtx\n",
            "[2024-07-11 21:17:25,074]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978540/counts_unfiltered/cells_x_genes.nascent.mtx\n",
            "[2024-07-11 21:17:30,076]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978540/counts_unfiltered/cells_x_genes.ambiguous.mtx\n",
            "[2024-07-11 21:17:42,295]    INFO [count_nac] Combining matrices\n",
            "[2024-07-11 21:17:42,918]    INFO [count_nac] Writing matrices to h5ad /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978540/counts_unfiltered/adata.h5ad\n",
            "[2024-07-11 21:17:58,323]    INFO [count_nac] Using index index_nac.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978541/ from\n",
            "[2024-07-11 21:17:58,324]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S10_L001_R1_001.fastq.gz\n",
            "[2024-07-11 21:17:58,324]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S10_L001_R2_001.fastq.gz\n",
            "[2024-07-11 21:17:58,324]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S10_L002_R1_001.fastq.gz\n",
            "[2024-07-11 21:17:58,324]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S10_L002_R2_001.fastq.gz\n",
            "[2024-07-11 21:17:58,324]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S15_L001_R1_001.fastq.gz\n",
            "[2024-07-11 21:17:58,324]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S15_L001_R2_001.fastq.gz\n",
            "[2024-07-11 21:17:58,324]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S15_L002_R1_001.fastq.gz\n",
            "[2024-07-11 21:17:58,324]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S15_L002_R2_001.fastq.gz\n",
            "[2024-07-11 21:17:58,324]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L001_R1_001.fastq.gz\n",
            "[2024-07-11 21:17:58,324]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L001_R2_001.fastq.gz\n",
            "[2024-07-11 21:17:58,324]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L002_R1_001.fastq.gz\n",
            "[2024-07-11 21:17:58,324]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L002_R2_001.fastq.gz\n",
            "[2024-07-11 21:17:58,324]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L003_R1_001.fastq.gz\n",
            "[2024-07-11 21:17:58,324]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L003_R2_001.fastq.gz\n",
            "[2024-07-11 21:17:58,324]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L004_R1_001.fastq.gz\n",
            "[2024-07-11 21:17:58,324]    INFO [count_nac]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L004_R2_001.fastq.gz\n",
            "[2024-07-11 21:37:33,762]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978541/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978541/tmp/output.s.bus\n",
            "[2024-07-11 21:37:51,300]    INFO [count_nac] On-list not provided\n",
            "[2024-07-11 21:37:51,300]    INFO [count_nac] Copying pre-packaged 10XV2 on-list to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978541/\n",
            "[2024-07-11 21:37:51,431]    INFO [count_nac] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978541/tmp/output.s.bus\n",
            "[2024-07-11 21:37:58,349]    INFO [count_nac] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978541/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978541/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978541/10x_version2_whitelist.txt\n",
            "[2024-07-11 21:38:20,898]    INFO [count_nac] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978541/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978541/output.unfiltered.bus\n",
            "[2024-07-11 21:38:30,424]    INFO [count_nac] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978541/counts_unfiltered/cells_x_genes from BUS file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978541/output.unfiltered.bus\n",
            "[2024-07-11 21:40:29,275]    INFO [count_nac] Writing gene names to file /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978541/counts_unfiltered/cells_x_genes.genes.names.txt\n",
            "[2024-07-11 21:40:29,620] WARNING [count_nac] 1599 gene IDs do not have corresponding valid gene names. These genes will use their gene IDs instead.\n",
            "[2024-07-11 21:40:29,666]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978541/counts_unfiltered/cells_x_genes.cell.mtx\n",
            "[2024-07-11 21:40:44,276]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978541/counts_unfiltered/cells_x_genes.cell.mtx\n",
            "[2024-07-11 21:41:04,406]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978541/counts_unfiltered/cells_x_genes.nucleus.mtx\n",
            "[2024-07-11 21:41:20,991]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978541/counts_unfiltered/cells_x_genes.nucleus.mtx\n",
            "[2024-07-11 21:41:43,329]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978541/counts_unfiltered/cells_x_genes.total.mtx\n",
            "[2024-07-11 21:42:00,849]    INFO [count_nac] Summing matrices into /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978541/counts_unfiltered/cells_x_genes.total.mtx\n",
            "[2024-07-11 21:42:51,707]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978541/counts_unfiltered/cells_x_genes.mature.mtx\n",
            "[2024-07-11 21:42:54,470]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978541/counts_unfiltered/cells_x_genes.nascent.mtx\n",
            "[2024-07-11 21:42:58,142]    INFO [count_nac] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978541/counts_unfiltered/cells_x_genes.ambiguous.mtx\n",
            "[2024-07-11 21:43:10,032]    INFO [count_nac] Combining matrices\n",
            "[2024-07-11 21:43:10,629]    INFO [count_nac] Writing matrices to h5ad /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978541/counts_unfiltered/adata.h5ad\n"
          ]
        }
      ],
      "source": [
        "!kb count -i index_nac.idx -g t2g_nac.txt -c1 cdna.txt -c2 nascent.txt --overwrite -x 10xv2 --workflow=nac --sum=total --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978537/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_S9_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_S9_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_S9_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_S9_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_mm10_S9_L003_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_mm10_S9_L003_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_mm10_S9_L004_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_mm10_S9_L004_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_mm10_S9_L005_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_mm10_S9_L005_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_mm10_S9_L006_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_mm10_S9_L006_R2_001.fastq.gz\n",
        "\n",
        "!kb count -i index_nac.idx -g t2g_nac.txt -c1 cdna.txt -c2 nascent.txt --overwrite -x 10xv2 --workflow=nac --sum=total --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978538/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_S10_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_S10_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_S10_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_S10_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_mm10_S10_L003_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_mm10_S10_L003_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_mm10_S10_L004_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_mm10_S10_L004_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_mm10_S10_L005_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_mm10_S10_L005_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_mm10_S10_L006_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_mm10_S10_L006_R2_001.fastq.gz\n",
        "\n",
        "!kb count -i index_nac.idx -g t2g_nac.txt -c1 cdna.txt -c2 nascent.txt --overwrite -x 10xv2 --workflow=nac --sum=total --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978539/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S13_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S13_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S13_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S13_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L003_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L003_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L004_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L004_R2_001.fastq.gz\n",
        "\n",
        "\n",
        "\n",
        "!kb count -i index_nac.idx -g t2g_nac.txt -c1 cdna.txt -c2 nascent.txt --overwrite -x 10xv2 --workflow=nac --sum=total --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978540/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S14_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S14_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S14_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S14_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L003_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L003_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L004_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L004_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S9_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S9_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S9_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S9_L002_R2_001.fastq.gz\n",
        "\n",
        "\n",
        "\n",
        "!kb count -i index_nac.idx -g t2g_nac.txt -c1 cdna.txt -c2 nascent.txt --overwrite -x 10xv2 --workflow=nac --sum=total --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978541/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S10_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S10_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S10_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S10_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S15_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S15_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S15_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S15_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L003_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L003_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L004_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L004_R2_001.fastq.gz\n",
        "\n",
        "!kb count -i index_nac.idx -g t2g_nac.txt -c1 cdna.txt -c2 nascent.txt --overwrite -x 10xv2 --workflow=nac --sum=total --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978542/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S11_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S11_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S11_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S11_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S16_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S16_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S16_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S16_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S30_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S30_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S30_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S30_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S30_L003_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S30_L003_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S30_L004_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S30_L004_R2_001.fastq.gz\n",
        "\n",
        "!kb count -i index_nac.idx -g t2g_nac.txt -c1 cdna.txt -c2 nascent.txt --overwrite -x 10xv2 --workflow=nac --sum=total --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978543/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S12_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S12_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S12_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S12_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S17_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S17_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S17_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S17_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S31_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S31_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S31_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S31_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S31_L003_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S31_L003_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S31_L004_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S31_L004_R2_001.fastq.gz\n",
        "\n",
        "\n",
        "!kb count -i index_nac.idx -g t2g_nac.txt -c1 cdna.txt -c2 nascent.txt --overwrite -x 10xv2 --workflow=nac --sum=total --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/nac/ERX4978544/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S13_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S13_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S13_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S13_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S18_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S18_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S18_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S18_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S32_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S32_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S32_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S32_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S32_L003_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S32_L003_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S32_L004_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S32_L004_R2_001.fastq.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "qjws18_RHmt8"
      },
      "source": [
        "# Aligning strange datasets again, using tcc workflow"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "VgoXMXkeQWSN"
      },
      "source": [
        "# First aligning Cheung et al files\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ags6eDg8QYVE",
        "outputId": "43327aff-5e72-403c-e65b-1a90fd0ba147"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[2024-07-12 09:32:27,398]    INFO [count] Using index index_std.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX4736337/ from\n",
            "[2024-07-12 09:32:27,399]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R1_001.fastq.gz\n",
            "[2024-07-12 09:32:27,399]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R2_001.fastq.gz\n",
            "[2024-07-12 09:32:27,399]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R1_002.fastq.gz\n",
            "[2024-07-12 09:32:27,399]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R2_002.fastq.gz\n",
            "[2024-07-12 09:32:27,399]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R1_003.fastq.gz\n",
            "[2024-07-12 09:32:27,399]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R2_003.fastq.gz\n",
            "[2024-07-12 09:32:27,399]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R1_001.fastq.gz\n",
            "[2024-07-12 09:32:27,399]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R2_001.fastq.gz\n",
            "[2024-07-12 09:32:27,399]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R1_002.fastq.gz\n",
            "[2024-07-12 09:32:27,399]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R2_002.fastq.gz\n",
            "[2024-07-12 09:32:27,399]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R1_003.fastq.gz\n",
            "[2024-07-12 09:32:27,399]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R2_003.fastq.gz\n",
            "[2024-07-12 09:58:58,293]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX4736337/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX4736337/tmp/output.s.bus\n",
            "[2024-07-12 10:01:10,781]    INFO [count] On-list not provided\n",
            "[2024-07-12 10:01:10,782]    INFO [count] Copying pre-packaged 10XV2 on-list to /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX4736337/\n",
            "[2024-07-12 10:01:10,913]    INFO [count] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX4736337/tmp/output.s.bus\n",
            "[2024-07-12 10:01:29,553]    INFO [count] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX4736337/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX4736337/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX4736337/10x_version2_whitelist.txt\n",
            "[2024-07-12 10:02:28,273]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX4736337/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX4736337/output.unfiltered.bus\n",
            "[2024-07-12 10:04:25,546]    INFO [count] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX4736337/counts_unfiltered/cells_x_tcc from BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX4736337/output.unfiltered.bus\n",
            "[2024-07-12 10:05:47,875]    INFO [count] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX4736337/counts_unfiltered/cells_x_tcc.mtx\n",
            "[2024-07-12 10:06:25,156]    INFO [count] Writing matrix to h5ad /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX4736337/counts_unfiltered/adata.h5ad\n"
          ]
        }
      ],
      "source": [
        "!kb count -i index_std.idx -g t2g_std.txt --overwrite -x 10xv2 --tcc --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX4736337/ /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R1_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R2_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R1_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R2_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R1_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R2_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R1_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736337/fastqs/7wo_B6N_males_Pool1_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R2_003.fastq.gz\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Vdrf9aW9e542",
        "outputId": "8d254b60-4793-45f5-c7dd-3cb1d743683d"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[2024-07-12 10:28:05,627]    INFO [count] Using index index_std.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX4736338/ from\n",
            "[2024-07-12 10:28:05,627]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R1_001.fastq.gz\n",
            "[2024-07-12 10:28:05,627]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R2_001.fastq.gz\n",
            "[2024-07-12 10:28:05,627]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R1_002.fastq.gz\n",
            "[2024-07-12 10:28:05,627]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R2_002.fastq.gz\n",
            "[2024-07-12 10:28:05,627]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R1_003.fastq.gz\n",
            "[2024-07-12 10:28:05,627]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R2_003.fastq.gz\n",
            "[2024-07-12 10:28:05,627]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R1_004.fastq.gz\n",
            "[2024-07-12 10:28:05,627]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R2_004.fastq.gz\n",
            "[2024-07-12 10:28:05,627]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R1_001.fastq.gz\n",
            "[2024-07-12 10:28:05,627]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R2_001.fastq.gz\n",
            "[2024-07-12 10:28:05,627]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R1_002.fastq.gz\n",
            "[2024-07-12 10:28:05,627]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R2_002.fastq.gz\n",
            "[2024-07-12 10:28:05,627]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R1_003.fastq.gz\n",
            "[2024-07-12 10:28:05,627]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R2_003.fastq.gz\n",
            "[2024-07-12 10:28:05,627]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R1_004.fastq.gz\n",
            "[2024-07-12 10:28:05,627]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R2_004.fastq.gz\n",
            "[2024-07-12 11:01:37,545]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX4736338/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX4736338/tmp/output.s.bus\n",
            "[2024-07-12 11:04:22,280]    INFO [count] On-list not provided\n",
            "[2024-07-12 11:04:22,280]    INFO [count] Copying pre-packaged 10XV2 on-list to /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX4736338/\n",
            "[2024-07-12 11:04:22,453]    INFO [count] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX4736338/tmp/output.s.bus\n",
            "[2024-07-12 11:04:44,600]    INFO [count] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX4736338/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX4736338/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX4736338/10x_version2_whitelist.txt\n",
            "[2024-07-12 11:06:01,774]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX4736338/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX4736338/output.unfiltered.bus\n",
            "[2024-07-12 11:08:12,074]    INFO [count] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX4736338/counts_unfiltered/cells_x_tcc from BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX4736338/output.unfiltered.bus\n",
            "[2024-07-12 11:09:46,624]    INFO [count] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX4736338/counts_unfiltered/cells_x_tcc.mtx\n",
            "[2024-07-12 11:10:30,853]    INFO [count] Writing matrix to h5ad /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX4736338/counts_unfiltered/adata.h5ad\n"
          ]
        }
      ],
      "source": [
        "!kb count -i index_std.idx -g t2g_std.txt --overwrite -x 10xv2 --tcc --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX4736338/ /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R1_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R2_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R1_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R2_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R1_004.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L001_R2_004.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R1_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R2_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R1_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R2_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R1_004.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX4736338/fastqs/7wo_B6N_males_Pool2_mm10v7b_MissingLibrary_1_HW73WBBXX/bamtofastq_S1_L002_R2_004.fastq.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "DX-_GoqWNGTj"
      },
      "source": [
        "# Cheung 2020 now"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "0R3BkDrLNEZi",
        "outputId": "63e29f37-188f-433d-9c09-022d2e57034c"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[2024-09-13 11:54:02,994]    INFO [count] Using index index_std.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX7377164/ from\n",
            "[2024-09-13 11:54:02,994]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R1_001.fastq.gz\n",
            "[2024-09-13 11:54:02,994]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R2_001.fastq.gz\n",
            "[2024-09-13 11:54:02,994]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R1_002.fastq.gz\n",
            "[2024-09-13 11:54:02,994]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R2_002.fastq.gz\n",
            "[2024-09-13 11:54:02,994]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R1_003.fastq.gz\n",
            "[2024-09-13 11:54:02,994]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R2_003.fastq.gz\n",
            "[2024-09-13 11:54:02,994]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R1_004.fastq.gz\n",
            "[2024-09-13 11:54:02,994]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R2_004.fastq.gz\n",
            "[2024-09-13 11:54:02,994]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R1_001.fastq.gz\n",
            "[2024-09-13 11:54:02,995]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R2_001.fastq.gz\n",
            "[2024-09-13 11:54:02,995]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R1_002.fastq.gz\n",
            "[2024-09-13 11:54:02,995]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R2_002.fastq.gz\n",
            "[2024-09-13 11:54:02,995]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R1_003.fastq.gz\n",
            "[2024-09-13 11:54:02,995]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R2_003.fastq.gz\n",
            "[2024-09-13 11:54:02,995]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R1_004.fastq.gz\n",
            "[2024-09-13 11:54:02,995]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R2_004.fastq.gz\n",
            "[2024-09-13 12:17:41,506]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX7377164/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX7377164/tmp/output.s.bus\n",
            "[2024-09-13 12:19:34,653]    INFO [count] On-list not provided\n",
            "[2024-09-13 12:19:34,655]    INFO [count] Copying pre-packaged 10XV2 on-list to /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX7377164/\n",
            "[2024-09-13 12:19:34,768]    INFO [count] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX7377164/tmp/output.s.bus\n",
            "[2024-09-13 12:19:48,902]    INFO [count] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX7377164/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX7377164/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX7377164/10x_version2_whitelist.txt\n",
            "[2024-09-13 12:20:39,409]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX7377164/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX7377164/output.unfiltered.bus\n",
            "[2024-09-13 12:21:46,860]    INFO [count] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX7377164/counts_unfiltered/cells_x_tcc from BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX7377164/output.unfiltered.bus\n",
            "[2024-09-13 12:22:41,270]    INFO [count] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX7377164/counts_unfiltered/cells_x_tcc.mtx\n",
            "[2024-09-13 12:22:44,152]    INFO [count] Writing matrix to h5ad /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX7377164/counts_unfiltered/adata.h5ad\n"
          ]
        }
      ],
      "source": [
        "!kb count -i index_std.idx -g t2g_std.txt --overwrite -x 10xv2 --tcc --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX7377164/ /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R1_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R2_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R1_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R2_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R1_004.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R2_004.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R1_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R2_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R1_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R2_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R1_004.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377164/fastqs/P4_Hesx1Cre_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R2_004.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "5BKkiYgENRDV",
        "outputId": "e41d26cf-c7a4-4f92-9383-c0874c137507"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[2024-09-13 12:22:58,308]    INFO [count] Using index index_std.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX7377166/ from\n",
            "[2024-09-13 12:22:58,308]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R1_001.fastq.gz\n",
            "[2024-09-13 12:22:58,308]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R2_001.fastq.gz\n",
            "[2024-09-13 12:22:58,308]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R1_002.fastq.gz\n",
            "[2024-09-13 12:22:58,309]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R2_002.fastq.gz\n",
            "[2024-09-13 12:22:58,309]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R1_003.fastq.gz\n",
            "[2024-09-13 12:22:58,309]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R2_003.fastq.gz\n",
            "[2024-09-13 12:22:58,309]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R1_004.fastq.gz\n",
            "[2024-09-13 12:22:58,309]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R2_004.fastq.gz\n",
            "[2024-09-13 12:22:58,309]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R1_001.fastq.gz\n",
            "[2024-09-13 12:22:58,309]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R2_001.fastq.gz\n",
            "[2024-09-13 12:22:58,309]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R1_002.fastq.gz\n",
            "[2024-09-13 12:22:58,309]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R2_002.fastq.gz\n",
            "[2024-09-13 12:22:58,309]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R1_003.fastq.gz\n",
            "[2024-09-13 12:22:58,309]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R2_003.fastq.gz\n",
            "[2024-09-13 12:22:58,309]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R1_004.fastq.gz\n",
            "[2024-09-13 12:22:58,309]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R2_004.fastq.gz\n",
            "[2024-09-13 12:46:18,161]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX7377166/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX7377166/tmp/output.s.bus\n",
            "[2024-09-13 12:48:51,567]    INFO [count] On-list not provided\n",
            "[2024-09-13 12:48:51,568]    INFO [count] Copying pre-packaged 10XV2 on-list to /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX7377166/\n",
            "[2024-09-13 12:48:51,686]    INFO [count] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX7377166/tmp/output.s.bus\n",
            "[2024-09-13 12:49:05,918]    INFO [count] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX7377166/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX7377166/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX7377166/10x_version2_whitelist.txt\n",
            "[2024-09-13 12:49:51,418]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX7377166/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX7377166/output.unfiltered.bus\n",
            "[2024-09-13 12:50:54,060]    INFO [count] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX7377166/counts_unfiltered/cells_x_tcc from BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX7377166/output.unfiltered.bus\n",
            "[2024-09-13 12:51:41,838]    INFO [count] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX7377166/counts_unfiltered/cells_x_tcc.mtx\n",
            "[2024-09-13 12:51:44,660]    INFO [count] Writing matrix to h5ad /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX7377166/counts_unfiltered/adata.h5ad\n"
          ]
        }
      ],
      "source": [
        "!kb count -i index_std.idx -g t2g_std.txt --overwrite -x 10xv2 --tcc --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX7377166/ /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R1_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R2_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R1_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R2_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R1_004.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L004_R2_004.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R1_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R2_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R1_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R2_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R1_004.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX7377166/fastqs/P4_Hesx1Cre_dnRAR_v3_expect10k_0_1_HTYCGBBXX/bamtofastq_S1_L005_R2_004.fastq.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "MJYH1DApNWYK"
      },
      "source": [
        "# Now Bastedo"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "hFv6UL6fNUrl",
        "outputId": "04ff875a-133b-4fe7-8d8b-6e633483bd9b"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[2024-09-13 12:52:02,071]    INFO [count] Using index index_std.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX9637224/ from\n",
            "[2024-09-13 12:52:02,072]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L001_R1_001.fastq.gz\n",
            "[2024-09-13 12:52:02,072]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L001_R2_001.fastq.gz\n",
            "[2024-09-13 12:52:02,072]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L001_R1_002.fastq.gz\n",
            "[2024-09-13 12:52:02,072]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L001_R2_002.fastq.gz\n",
            "[2024-09-13 12:52:02,072]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L001_R1_003.fastq.gz\n",
            "[2024-09-13 12:52:02,072]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L001_R2_003.fastq.gz\n",
            "[2024-09-13 12:52:02,072]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L002_R1_001.fastq.gz\n",
            "[2024-09-13 12:52:02,072]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L002_R2_001.fastq.gz\n",
            "[2024-09-13 12:52:02,072]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L002_R1_002.fastq.gz\n",
            "[2024-09-13 12:52:02,072]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L002_R2_002.fastq.gz\n",
            "[2024-09-13 12:52:02,072]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L002_R1_003.fastq.gz\n",
            "[2024-09-13 12:52:02,072]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L002_R2_003.fastq.gz\n",
            "[2024-09-13 12:52:02,072]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L003_R1_001.fastq.gz\n",
            "[2024-09-13 12:52:02,072]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L003_R2_001.fastq.gz\n",
            "[2024-09-13 12:52:02,072]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L003_R1_002.fastq.gz\n",
            "[2024-09-13 12:52:02,072]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L003_R2_002.fastq.gz\n",
            "[2024-09-13 12:52:02,072]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L003_R1_003.fastq.gz\n",
            "[2024-09-13 12:52:02,072]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L003_R2_003.fastq.gz\n",
            "[2024-09-13 12:52:02,072]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L004_R1_001.fastq.gz\n",
            "[2024-09-13 12:52:02,072]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L004_R2_001.fastq.gz\n",
            "[2024-09-13 12:52:02,072]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L004_R1_002.fastq.gz\n",
            "[2024-09-13 12:52:02,072]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L004_R2_002.fastq.gz\n",
            "[2024-09-13 12:52:02,072]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L004_R1_003.fastq.gz\n",
            "[2024-09-13 12:52:02,072]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L004_R2_003.fastq.gz\n",
            "[2024-09-13 13:13:12,147]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX9637224/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX9637224/tmp/output.s.bus\n",
            "[2024-09-13 13:15:41,253]    INFO [count] On-list not provided\n",
            "[2024-09-13 13:15:41,254]    INFO [count] Copying pre-packaged 10XV3 on-list to /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX9637224/\n",
            "[2024-09-13 13:15:42,108]    INFO [count] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX9637224/tmp/output.s.bus\n",
            "[2024-09-13 13:16:21,391]    INFO [count] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX9637224/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX9637224/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX9637224/10x_version3_whitelist.txt\n",
            "[2024-09-13 13:17:39,456]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX9637224/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX9637224/output.unfiltered.bus\n",
            "[2024-09-13 13:19:07,744]    INFO [count] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX9637224/counts_unfiltered/cells_x_tcc from BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX9637224/output.unfiltered.bus\n",
            "[2024-09-13 13:20:56,967]    INFO [count] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX9637224/counts_unfiltered/cells_x_tcc.mtx\n",
            "[2024-09-13 13:21:02,665]    INFO [count] Writing matrix to h5ad /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX9637224/counts_unfiltered/adata.h5ad\n"
          ]
        }
      ],
      "source": [
        "!kb count -i index_std.idx -g t2g_std.txt --overwrite -x 10xv3 --tcc --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX9637224/ /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L001_R1_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L001_R2_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L001_R1_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L001_R2_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L002_R1_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L002_R2_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L002_R1_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L002_R2_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L003_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L003_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L003_R1_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L003_R2_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L003_R1_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L003_R2_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L004_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L004_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L004_R1_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L004_R2_002.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L004_R1_003.fastq.gz /content/drive/MyDrive/pituitary_atlas/raw/SRX9637224/fastqs/sc62__RNA_WS20200904_8WeekPituitary_0_1_HY3YGBGXF/bamtofastq_S1_L004_R2_003.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "g0LDVUn7NU9b"
      },
      "outputs": [],
      "source": []
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "Xy3nJKuINIZX"
      },
      "outputs": [],
      "source": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "TOaAbKp3tMU_"
      },
      "source": [
        "Now repeat for Vennekens datasets"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "WMMPmtaZtLx3",
        "outputId": "2a37867e-43e9-4db6-9b9f-3d1d77b847c0"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[2024-07-12 11:36:43,725]    INFO [count] Using index index_std.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978537/ from\n",
            "[2024-07-12 11:36:43,725]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_S9_L001_R1_001.fastq.gz\n",
            "[2024-07-12 11:36:43,725]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_S9_L001_R2_001.fastq.gz\n",
            "[2024-07-12 11:36:43,725]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_S9_L002_R1_001.fastq.gz\n",
            "[2024-07-12 11:36:43,725]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_S9_L002_R2_001.fastq.gz\n",
            "[2024-07-12 11:36:43,725]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_mm10_S9_L003_R1_001.fastq.gz\n",
            "[2024-07-12 11:36:43,725]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_mm10_S9_L003_R2_001.fastq.gz\n",
            "[2024-07-12 11:36:43,725]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_mm10_S9_L004_R1_001.fastq.gz\n",
            "[2024-07-12 11:36:43,725]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_mm10_S9_L004_R2_001.fastq.gz\n",
            "[2024-07-12 11:36:43,725]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_mm10_S9_L005_R1_001.fastq.gz\n",
            "[2024-07-12 11:36:43,725]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_mm10_S9_L005_R2_001.fastq.gz\n",
            "[2024-07-12 11:36:43,726]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_mm10_S9_L006_R1_001.fastq.gz\n",
            "[2024-07-12 11:36:43,726]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_mm10_S9_L006_R2_001.fastq.gz\n",
            "[2024-07-12 11:54:29,432]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978537/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978537/tmp/output.s.bus\n",
            "[2024-07-12 11:56:16,369]    INFO [count] On-list not provided\n",
            "[2024-07-12 11:56:16,369]    INFO [count] Copying pre-packaged 10XV2 on-list to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978537/\n",
            "[2024-07-12 11:56:16,581]    INFO [count] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978537/tmp/output.s.bus\n",
            "[2024-07-12 11:56:35,430]    INFO [count] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978537/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978537/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978537/10x_version2_whitelist.txt\n",
            "[2024-07-12 11:57:21,126]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978537/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978537/output.unfiltered.bus\n",
            "[2024-07-12 11:58:28,270]    INFO [count] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978537/counts_unfiltered/cells_x_tcc from BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978537/output.unfiltered.bus\n",
            "[2024-07-12 11:59:34,672]    INFO [count] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978537/counts_unfiltered/cells_x_tcc.mtx\n",
            "[2024-07-12 12:00:05,369]    INFO [count] Writing matrix to h5ad /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978537/counts_unfiltered/adata.h5ad\n",
            "[2024-07-12 12:00:19,764]    INFO [count] Using index index_std.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978538/ from\n",
            "[2024-07-12 12:00:19,764]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_S10_L001_R1_001.fastq.gz\n",
            "[2024-07-12 12:00:19,764]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_S10_L001_R2_001.fastq.gz\n",
            "[2024-07-12 12:00:19,764]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_S10_L002_R1_001.fastq.gz\n",
            "[2024-07-12 12:00:19,764]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_S10_L002_R2_001.fastq.gz\n",
            "[2024-07-12 12:00:19,764]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_mm10_S10_L003_R1_001.fastq.gz\n",
            "[2024-07-12 12:00:19,764]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_mm10_S10_L003_R2_001.fastq.gz\n",
            "[2024-07-12 12:00:19,764]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_mm10_S10_L004_R1_001.fastq.gz\n",
            "[2024-07-12 12:00:19,764]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_mm10_S10_L004_R2_001.fastq.gz\n",
            "[2024-07-12 12:00:19,764]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_mm10_S10_L005_R1_001.fastq.gz\n",
            "[2024-07-12 12:00:19,764]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_mm10_S10_L005_R2_001.fastq.gz\n",
            "[2024-07-12 12:00:19,764]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_mm10_S10_L006_R1_001.fastq.gz\n",
            "[2024-07-12 12:00:19,764]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_mm10_S10_L006_R2_001.fastq.gz\n",
            "[2024-07-12 12:26:01,050]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978538/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978538/tmp/output.s.bus\n",
            "[2024-07-12 12:27:47,974]    INFO [count] On-list not provided\n",
            "[2024-07-12 12:27:47,974]    INFO [count] Copying pre-packaged 10XV2 on-list to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978538/\n",
            "[2024-07-12 12:27:48,111]    INFO [count] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978538/tmp/output.s.bus\n",
            "[2024-07-12 12:28:00,339]    INFO [count] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978538/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978538/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978538/10x_version2_whitelist.txt\n",
            "[2024-07-12 12:28:43,435]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978538/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978538/output.unfiltered.bus\n",
            "[2024-07-12 12:28:57,070]    INFO [count] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978538/counts_unfiltered/cells_x_tcc from BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978538/output.unfiltered.bus\n",
            "[2024-07-12 12:30:08,991]    INFO [count] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978538/counts_unfiltered/cells_x_tcc.mtx\n",
            "[2024-07-12 12:30:38,723]    INFO [count] Writing matrix to h5ad /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978538/counts_unfiltered/adata.h5ad\n",
            "[2024-07-12 12:30:53,185]    INFO [count] Using index index_std.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978539/ from\n",
            "[2024-07-12 12:30:53,185]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S13_L001_R1_001.fastq.gz\n",
            "[2024-07-12 12:30:53,186]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S13_L001_R2_001.fastq.gz\n",
            "[2024-07-12 12:30:53,186]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S13_L002_R1_001.fastq.gz\n",
            "[2024-07-12 12:30:53,186]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S13_L002_R2_001.fastq.gz\n",
            "[2024-07-12 12:30:53,186]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L001_R1_001.fastq.gz\n",
            "[2024-07-12 12:30:53,186]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L001_R2_001.fastq.gz\n",
            "[2024-07-12 12:30:53,186]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L002_R1_001.fastq.gz\n",
            "[2024-07-12 12:30:53,186]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L002_R2_001.fastq.gz\n",
            "[2024-07-12 12:30:53,186]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L003_R1_001.fastq.gz\n",
            "[2024-07-12 12:30:53,186]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L003_R2_001.fastq.gz\n",
            "[2024-07-12 12:30:53,186]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L004_R1_001.fastq.gz\n",
            "[2024-07-12 12:30:53,186]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L004_R2_001.fastq.gz\n",
            "[2024-07-12 15:00:29,604]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978539/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978539/tmp/output.s.bus\n",
            "[2024-07-12 15:08:40,898]    INFO [count] On-list not provided\n",
            "[2024-07-12 15:08:40,899]    INFO [count] Copying pre-packaged 10XV2 on-list to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978539/\n",
            "[2024-07-12 15:08:41,079]    INFO [count] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978539/tmp/output.s.bus\n",
            "[2024-07-12 15:09:05,930]    INFO [count] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978539/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978539/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978539/10x_version2_whitelist.txt\n",
            "[2024-07-12 15:10:32,738]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978539/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978539/output.unfiltered.bus\n",
            "[2024-07-12 15:11:45,895]    INFO [count] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978539/counts_unfiltered/cells_x_tcc from BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978539/output.unfiltered.bus\n",
            "[2024-07-12 15:13:18,191]    INFO [count] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978539/counts_unfiltered/cells_x_tcc.mtx\n",
            "[2024-07-12 15:14:02,723]    INFO [count] Writing matrix to h5ad /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978539/counts_unfiltered/adata.h5ad\n",
            "[2024-07-12 15:14:39,638]    INFO [count] Using index index_std.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978540/ from\n",
            "[2024-07-12 15:14:39,639]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S14_L001_R1_001.fastq.gz\n",
            "[2024-07-12 15:14:39,639]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S14_L001_R2_001.fastq.gz\n",
            "[2024-07-12 15:14:39,639]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S14_L002_R1_001.fastq.gz\n",
            "[2024-07-12 15:14:39,639]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S14_L002_R2_001.fastq.gz\n",
            "[2024-07-12 15:14:39,639]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L001_R1_001.fastq.gz\n",
            "[2024-07-12 15:14:39,640]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L001_R2_001.fastq.gz\n",
            "[2024-07-12 15:14:39,640]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L002_R1_001.fastq.gz\n",
            "[2024-07-12 15:14:39,640]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L002_R2_001.fastq.gz\n",
            "[2024-07-12 15:14:39,640]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L003_R1_001.fastq.gz\n",
            "[2024-07-12 15:14:39,640]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L003_R2_001.fastq.gz\n",
            "[2024-07-12 15:14:39,640]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L004_R1_001.fastq.gz\n",
            "[2024-07-12 15:14:39,640]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L004_R2_001.fastq.gz\n",
            "[2024-07-12 15:14:39,640]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S9_L001_R1_001.fastq.gz\n",
            "[2024-07-12 15:14:39,640]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S9_L001_R2_001.fastq.gz\n",
            "[2024-07-12 15:14:39,640]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S9_L002_R1_001.fastq.gz\n",
            "[2024-07-12 15:14:39,640]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S9_L002_R2_001.fastq.gz\n",
            "[2024-07-12 15:24:28,188]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978540/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978540/tmp/output.s.bus\n",
            "[2024-07-12 15:25:09,376]    INFO [count] On-list not provided\n",
            "[2024-07-12 15:25:09,376]    INFO [count] Copying pre-packaged 10XV2 on-list to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978540/\n",
            "[2024-07-12 15:25:09,510]    INFO [count] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978540/tmp/output.s.bus\n",
            "[2024-07-12 15:25:18,236]    INFO [count] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978540/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978540/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978540/10x_version2_whitelist.txt\n",
            "[2024-07-12 15:26:05,069]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978540/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978540/output.unfiltered.bus\n",
            "[2024-07-12 15:26:17,399]    INFO [count] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978540/counts_unfiltered/cells_x_tcc from BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978540/output.unfiltered.bus\n",
            "[2024-07-12 15:26:54,265]    INFO [count] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978540/counts_unfiltered/cells_x_tcc.mtx\n",
            "[2024-07-12 15:27:12,114]    INFO [count] Writing matrix to h5ad /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978540/counts_unfiltered/adata.h5ad\n",
            "[2024-07-12 15:27:27,720]    INFO [count] Using index index_std.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978541/ from\n",
            "[2024-07-12 15:27:27,721]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S10_L001_R1_001.fastq.gz\n",
            "[2024-07-12 15:27:27,721]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S10_L001_R2_001.fastq.gz\n",
            "[2024-07-12 15:27:27,721]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S10_L002_R1_001.fastq.gz\n",
            "[2024-07-12 15:27:27,721]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S10_L002_R2_001.fastq.gz\n",
            "[2024-07-12 15:27:27,721]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S15_L001_R1_001.fastq.gz\n",
            "[2024-07-12 15:27:27,721]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S15_L001_R2_001.fastq.gz\n",
            "[2024-07-12 15:27:27,721]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S15_L002_R1_001.fastq.gz\n",
            "[2024-07-12 15:27:27,721]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S15_L002_R2_001.fastq.gz\n",
            "[2024-07-12 15:27:27,721]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L001_R1_001.fastq.gz\n",
            "[2024-07-12 15:27:27,721]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L001_R2_001.fastq.gz\n",
            "[2024-07-12 15:27:27,721]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L002_R1_001.fastq.gz\n",
            "[2024-07-12 15:27:27,721]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L002_R2_001.fastq.gz\n",
            "[2024-07-12 15:27:27,721]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L003_R1_001.fastq.gz\n",
            "[2024-07-12 15:27:27,721]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L003_R2_001.fastq.gz\n",
            "[2024-07-12 15:27:27,721]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L004_R1_001.fastq.gz\n",
            "[2024-07-12 15:27:27,721]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L004_R2_001.fastq.gz\n",
            "[2024-07-12 15:35:45,280]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978541/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978541/tmp/output.s.bus\n",
            "[2024-07-12 15:36:00,021]    INFO [count] On-list not provided\n",
            "[2024-07-12 15:36:00,021]    INFO [count] Copying pre-packaged 10XV2 on-list to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978541/\n",
            "[2024-07-12 15:36:00,299]    INFO [count] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978541/tmp/output.s.bus\n",
            "[2024-07-12 15:36:13,344]    INFO [count] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978541/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978541/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978541/10x_version2_whitelist.txt\n",
            "[2024-07-12 15:36:47,037]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978541/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978541/output.unfiltered.bus\n",
            "[2024-07-12 15:36:57,364]    INFO [count] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978541/counts_unfiltered/cells_x_tcc from BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978541/output.unfiltered.bus\n",
            "[2024-07-12 15:37:30,580]    INFO [count] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978541/counts_unfiltered/cells_x_tcc.mtx\n",
            "[2024-07-12 15:37:46,589]    INFO [count] Writing matrix to h5ad /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978541/counts_unfiltered/adata.h5ad\n",
            "[2024-07-12 15:37:57,159]    INFO [count] Using index index_std.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978542/ from\n",
            "[2024-07-12 15:37:57,160]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S11_L001_R1_001.fastq.gz\n",
            "[2024-07-12 15:37:57,160]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S11_L001_R2_001.fastq.gz\n",
            "[2024-07-12 15:37:57,160]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S11_L002_R1_001.fastq.gz\n",
            "[2024-07-12 15:37:57,160]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S11_L002_R2_001.fastq.gz\n",
            "[2024-07-12 15:37:57,160]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S16_L001_R1_001.fastq.gz\n",
            "[2024-07-12 15:37:57,160]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S16_L001_R2_001.fastq.gz\n",
            "[2024-07-12 15:37:57,160]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S16_L002_R1_001.fastq.gz\n",
            "[2024-07-12 15:37:57,160]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S16_L002_R2_001.fastq.gz\n",
            "[2024-07-12 15:37:57,160]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S30_L001_R1_001.fastq.gz\n",
            "[2024-07-12 15:37:57,160]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S30_L001_R2_001.fastq.gz\n",
            "[2024-07-12 15:37:57,160]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S30_L002_R1_001.fastq.gz\n",
            "[2024-07-12 15:37:57,160]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S30_L002_R2_001.fastq.gz\n",
            "[2024-07-12 15:37:57,160]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S30_L003_R1_001.fastq.gz\n",
            "[2024-07-12 15:37:57,160]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S30_L003_R2_001.fastq.gz\n",
            "[2024-07-12 15:37:57,160]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S30_L004_R1_001.fastq.gz\n",
            "[2024-07-12 15:37:57,160]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S30_L004_R2_001.fastq.gz\n",
            "[2024-07-12 15:48:29,671]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978542/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978542/tmp/output.s.bus\n",
            "[2024-07-12 15:48:46,517]    INFO [count] On-list not provided\n",
            "[2024-07-12 15:48:46,518]    INFO [count] Copying pre-packaged 10XV2 on-list to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978542/\n",
            "[2024-07-12 15:48:46,677]    INFO [count] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978542/tmp/output.s.bus\n",
            "[2024-07-12 15:48:53,895]    INFO [count] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978542/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978542/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978542/10x_version2_whitelist.txt\n",
            "[2024-07-12 15:49:16,845]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978542/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978542/output.unfiltered.bus\n",
            "[2024-07-12 15:49:28,177]    INFO [count] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978542/counts_unfiltered/cells_x_tcc from BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978542/output.unfiltered.bus\n",
            "[2024-07-12 15:50:03,793]    INFO [count] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978542/counts_unfiltered/cells_x_tcc.mtx\n",
            "[2024-07-12 15:50:20,653]    INFO [count] Writing matrix to h5ad /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978542/counts_unfiltered/adata.h5ad\n",
            "[2024-07-12 15:50:31,229]    INFO [count] Using index index_std.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978543/ from\n",
            "[2024-07-12 15:50:31,229]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S12_L001_R1_001.fastq.gz\n",
            "[2024-07-12 15:50:31,229]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S12_L001_R2_001.fastq.gz\n",
            "[2024-07-12 15:50:31,229]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S12_L002_R1_001.fastq.gz\n",
            "[2024-07-12 15:50:31,229]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S12_L002_R2_001.fastq.gz\n",
            "[2024-07-12 15:50:31,229]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S17_L001_R1_001.fastq.gz\n",
            "[2024-07-12 15:50:31,229]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S17_L001_R2_001.fastq.gz\n",
            "[2024-07-12 15:50:31,229]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S17_L002_R1_001.fastq.gz\n",
            "[2024-07-12 15:50:31,229]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S17_L002_R2_001.fastq.gz\n",
            "[2024-07-12 15:50:31,229]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S31_L001_R1_001.fastq.gz\n",
            "[2024-07-12 15:50:31,229]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S31_L001_R2_001.fastq.gz\n",
            "[2024-07-12 15:50:31,229]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S31_L002_R1_001.fastq.gz\n",
            "[2024-07-12 15:50:31,229]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S31_L002_R2_001.fastq.gz\n",
            "[2024-07-12 15:50:31,229]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S31_L003_R1_001.fastq.gz\n",
            "[2024-07-12 15:50:31,229]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S31_L003_R2_001.fastq.gz\n",
            "[2024-07-12 15:50:31,230]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S31_L004_R1_001.fastq.gz\n",
            "[2024-07-12 15:50:31,230]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S31_L004_R2_001.fastq.gz\n",
            "[2024-07-12 16:00:59,661]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978543/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978543/tmp/output.s.bus\n",
            "[2024-07-12 16:01:22,518]    INFO [count] On-list not provided\n",
            "[2024-07-12 16:01:22,519]    INFO [count] Copying pre-packaged 10XV2 on-list to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978543/\n",
            "[2024-07-12 16:01:22,657]    INFO [count] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978543/tmp/output.s.bus\n",
            "[2024-07-12 16:01:30,075]    INFO [count] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978543/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978543/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978543/10x_version2_whitelist.txt\n",
            "[2024-07-12 16:01:52,625]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978543/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978543/output.unfiltered.bus\n",
            "[2024-07-12 16:02:02,554]    INFO [count] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978543/counts_unfiltered/cells_x_tcc from BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978543/output.unfiltered.bus\n",
            "[2024-07-12 16:02:39,006]    INFO [count] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978543/counts_unfiltered/cells_x_tcc.mtx\n",
            "[2024-07-12 16:02:56,401]    INFO [count] Writing matrix to h5ad /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978543/counts_unfiltered/adata.h5ad\n",
            "[2024-07-12 16:03:07,077]    INFO [count] Using index index_std.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978544/ from\n",
            "[2024-07-12 16:03:07,077]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S13_L001_R1_001.fastq.gz\n",
            "[2024-07-12 16:03:07,077]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S13_L001_R2_001.fastq.gz\n",
            "[2024-07-12 16:03:07,077]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S13_L002_R1_001.fastq.gz\n",
            "[2024-07-12 16:03:07,077]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S13_L002_R2_001.fastq.gz\n",
            "[2024-07-12 16:03:07,077]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S18_L001_R1_001.fastq.gz\n",
            "[2024-07-12 16:03:07,077]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S18_L001_R2_001.fastq.gz\n",
            "[2024-07-12 16:03:07,077]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S18_L002_R1_001.fastq.gz\n",
            "[2024-07-12 16:03:07,077]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S18_L002_R2_001.fastq.gz\n",
            "[2024-07-12 16:03:07,077]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S32_L001_R1_001.fastq.gz\n",
            "[2024-07-12 16:03:07,077]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S32_L001_R2_001.fastq.gz\n",
            "[2024-07-12 16:03:07,077]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S32_L002_R1_001.fastq.gz\n",
            "[2024-07-12 16:03:07,077]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S32_L002_R2_001.fastq.gz\n",
            "[2024-07-12 16:03:07,078]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S32_L003_R1_001.fastq.gz\n",
            "[2024-07-12 16:03:07,078]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S32_L003_R2_001.fastq.gz\n",
            "[2024-07-12 16:03:07,078]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S32_L004_R1_001.fastq.gz\n",
            "[2024-07-12 16:03:07,078]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S32_L004_R2_001.fastq.gz\n",
            "[2024-07-12 16:12:49,908]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978544/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978544/tmp/output.s.bus\n",
            "[2024-07-12 16:13:36,004]    INFO [count] On-list not provided\n",
            "[2024-07-12 16:13:36,004]    INFO [count] Copying pre-packaged 10XV2 on-list to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978544/\n",
            "[2024-07-12 16:13:36,303]    INFO [count] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978544/tmp/output.s.bus\n",
            "[2024-07-12 16:13:44,221]    INFO [count] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978544/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978544/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978544/10x_version2_whitelist.txt\n",
            "[2024-07-12 16:14:07,872]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978544/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978544/output.unfiltered.bus\n",
            "[2024-07-12 16:14:15,694]    INFO [count] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978544/counts_unfiltered/cells_x_tcc from BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978544/output.unfiltered.bus\n",
            "[2024-07-12 16:14:51,782]    INFO [count] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978544/counts_unfiltered/cells_x_tcc.mtx\n",
            "[2024-07-12 16:15:16,040]    INFO [count] Writing matrix to h5ad /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978544/counts_unfiltered/adata.h5ad\n"
          ]
        }
      ],
      "source": [
        "!kb count -i index_std.idx -g t2g_std.txt --overwrite -x 10xv2 --tcc --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978537/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_S9_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_S9_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_S9_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_S9_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_mm10_S9_L003_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_mm10_S9_L003_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_mm10_S9_L004_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_mm10_S9_L004_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_mm10_S9_L005_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_mm10_S9_L005_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_mm10_S9_L006_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT036_mm10_S9_L006_R2_001.fastq.gz\n",
        "\n",
        "!kb count -i index_std.idx -g t2g_std.txt --overwrite -x 10xv2 --tcc --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978538/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_S10_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_S10_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_S10_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_S10_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_mm10_S10_L003_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_mm10_S10_L003_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_mm10_S10_L004_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_mm10_S10_L004_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_mm10_S10_L005_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_mm10_S10_L005_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_mm10_S10_L006_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT037_mm10_S10_L006_R2_001.fastq.gz\n",
        "\n",
        "!kb count -i index_std.idx -g t2g_std.txt --overwrite -x 10xv2 --tcc --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978539/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S13_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S13_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S13_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S13_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L003_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L003_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L004_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT144_mm10_S27_L004_R2_001.fastq.gz\n",
        "\n",
        "\n",
        "\n",
        "!kb count -i index_std.idx -g t2g_std.txt --overwrite -x 10xv2 --tcc --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978540/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S14_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S14_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S14_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S14_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L003_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L003_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L004_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S28_L004_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S9_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S9_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S9_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT145_mm10_S9_L002_R2_001.fastq.gz\n",
        "\n",
        "\n",
        "\n",
        "!kb count -i index_std.idx -g t2g_std.txt --overwrite -x 10xv2 --tcc --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978541/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S10_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S10_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S10_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S10_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S15_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S15_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S15_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S15_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L003_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L003_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L004_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT146_mm10_S29_L004_R2_001.fastq.gz\n",
        "\n",
        "!kb count -i index_std.idx -g t2g_std.txt --overwrite -x 10xv2 --tcc --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978542/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S11_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S11_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S11_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S11_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S16_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S16_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S16_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S16_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S30_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S30_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S30_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S30_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S30_L003_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S30_L003_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S30_L004_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT147_mm10_S30_L004_R2_001.fastq.gz\n",
        "\n",
        "!kb count -i index_std.idx -g t2g_std.txt --overwrite -x 10xv2 --tcc --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978543/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S12_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S12_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S12_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S12_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S17_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S17_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S17_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S17_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S31_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S31_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S31_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S31_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S31_L003_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S31_L003_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S31_L004_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT148_mm10_S31_L004_R2_001.fastq.gz\n",
        "\n",
        "\n",
        "!kb count -i index_std.idx -g t2g_std.txt --overwrite -x 10xv2 --tcc --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX4978544/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S13_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S13_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S13_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S13_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S18_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S18_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S18_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S18_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S32_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S32_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S32_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S32_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S32_L003_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S32_L003_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S32_L004_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/scrEXT149_mm10_S32_L004_R2_001.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "tr6hIKzJe68c"
      },
      "outputs": [],
      "source": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "ajdH4lqjlmEx"
      },
      "source": [
        "# Now repeat for De Vriendt dataset"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "ciWgw0Z2e6_W"
      },
      "outputs": [],
      "source": []
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "MwBhJqgml09v",
        "outputId": "ba3b8f13-1d10-4c5e-aed8-391a152a5a1a"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[2025-01-11 09:01:55,513]    INFO [count] Using index index_std.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202612/ from\n",
            "[2025-01-11 09:01:55,513]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L001_R1_001.fastq.gz\n",
            "[2025-01-11 09:01:55,513]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L001_R2_001.fastq.gz\n",
            "[2025-01-11 09:01:55,513]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L002_R1_001.fastq.gz\n",
            "[2025-01-11 09:01:55,513]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L002_R2_001.fastq.gz\n",
            "[2025-01-11 09:01:55,513]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L003_R1_001.fastq.gz\n",
            "[2025-01-11 09:01:55,513]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L003_R2_001.fastq.gz\n",
            "[2025-01-11 09:01:55,513]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L004_R1_001.fastq.gz\n",
            "[2025-01-11 09:01:55,513]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L004_R2_001.fastq.gz\n",
            "[2025-01-11 09:01:55,513]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S31_L001_R1_001.fastq.gz\n",
            "[2025-01-11 09:01:55,513]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S31_L001_R2_001.fastq.gz\n",
            "[2025-01-11 09:01:55,513]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S31_L002_R1_001.fastq.gz\n",
            "[2025-01-11 09:01:55,513]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S31_L002_R2_001.fastq.gz\n",
            "[2025-01-11 09:01:55,513]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S5_L001_R1_001.fastq.gz\n",
            "[2025-01-11 09:01:55,513]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S5_L001_R2_001.fastq.gz\n",
            "[2025-01-11 09:01:55,513]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S5_L002_R1_001.fastq.gz\n",
            "[2025-01-11 09:01:55,513]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S5_L002_R2_001.fastq.gz\n",
            "[2025-01-11 09:40:53,193]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202612/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202612/tmp/output.s.bus\n",
            "[2025-01-11 09:42:36,285]    INFO [count] On-list not provided\n",
            "[2025-01-11 09:42:36,286]    INFO [count] Copying pre-packaged 10XV3 on-list to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202612/\n",
            "[2025-01-11 09:42:37,079]    INFO [count] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202612/tmp/output.s.bus\n",
            "[2025-01-11 09:43:00,121]    INFO [count] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202612/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202612/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202612/10x_version3_whitelist.txt\n",
            "[2025-01-11 09:43:44,705]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202612/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202612/output.unfiltered.bus\n",
            "[2025-01-11 09:44:35,904]    INFO [count] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202612/counts_unfiltered/cells_x_tcc from BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202612/output.unfiltered.bus\n",
            "[2025-01-11 09:45:09,829]    INFO [count] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202612/counts_unfiltered/cells_x_tcc.mtx\n",
            "[2025-01-11 09:45:16,177]    INFO [count] Writing matrix to h5ad /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202612/counts_unfiltered/adata.h5ad\n"
          ]
        }
      ],
      "source": [
        "!kb count -i index_std.idx -g t2g_std.txt --overwrite -x 10xv3 --tcc --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202612/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L003_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L003_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L004_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S1_L004_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S31_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S31_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S31_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S31_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S5_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S5_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S5_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT330_mm10_S5_L002_R2_001.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "background_save": true,
          "base_uri": "https://localhost:8080/"
        },
        "id": "VSqoy7hGl09x",
        "outputId": "3ee909ca-110e-4868-db61-6e35fd0c96b5"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[2025-01-11 09:45:41,596]    INFO [count] Using index index_std.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202613/ from\n",
            "[2025-01-11 09:45:41,596]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L001_R1_001.fastq.gz\n",
            "[2025-01-11 09:45:41,596]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L001_R2_001.fastq.gz\n",
            "[2025-01-11 09:45:41,597]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L002_R1_001.fastq.gz\n",
            "[2025-01-11 09:45:41,597]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L002_R2_001.fastq.gz\n",
            "[2025-01-11 09:45:41,597]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L003_R1_001.fastq.gz\n",
            "[2025-01-11 09:45:41,597]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L003_R2_001.fastq.gz\n",
            "[2025-01-11 09:45:41,597]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L004_R1_001.fastq.gz\n",
            "[2025-01-11 09:45:41,598]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L004_R2_001.fastq.gz\n",
            "[2025-01-11 09:45:41,598]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S32_L001_R1_001.fastq.gz\n",
            "[2025-01-11 09:45:41,598]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S32_L001_R2_001.fastq.gz\n",
            "[2025-01-11 09:45:41,598]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S32_L002_R1_001.fastq.gz\n",
            "[2025-01-11 09:45:41,598]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S32_L002_R2_001.fastq.gz\n",
            "[2025-01-11 09:45:41,598]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S6_L001_R1_001.fastq.gz\n",
            "[2025-01-11 09:45:41,598]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S6_L001_R2_001.fastq.gz\n",
            "[2025-01-11 09:45:41,599]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S6_L002_R1_001.fastq.gz\n",
            "[2025-01-11 09:45:41,599]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S6_L002_R2_001.fastq.gz\n",
            "[2025-01-11 10:31:59,556]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202613/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202613/tmp/output.s.bus\n",
            "[2025-01-11 10:34:20,528]    INFO [count] On-list not provided\n",
            "[2025-01-11 10:34:20,529]    INFO [count] Copying pre-packaged 10XV3 on-list to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202613/\n",
            "[2025-01-11 10:34:22,120]    INFO [count] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202613/tmp/output.s.bus\n",
            "[2025-01-11 10:34:53,180]    INFO [count] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202613/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202613/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202613/10x_version3_whitelist.txt\n",
            "[2025-01-11 10:35:46,082]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202613/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202613/output.unfiltered.bus\n",
            "[2025-01-11 10:36:58,327]    INFO [count] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202613/counts_unfiltered/cells_x_tcc from BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202613/output.unfiltered.bus\n",
            "[2025-01-11 10:37:43,760]    INFO [count] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202613/counts_unfiltered/cells_x_tcc.mtx\n",
            "[2025-01-11 10:37:48,355]    INFO [count] Writing matrix to h5ad /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202613/counts_unfiltered/adata.h5ad\n"
          ]
        }
      ],
      "source": [
        "!kb count -i index_std.idx -g t2g_std.txt --overwrite -x 10xv3 --tcc --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202613/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L003_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L003_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L004_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S2_L004_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S32_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S32_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S32_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S32_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S6_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S6_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S6_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT331_mm10_S6_L002_R2_001.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "background_save": true
        },
        "id": "2aJOB9CTl09x",
        "outputId": "6910e8a0-1cf4-4fa5-aebc-59b514405cc8"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[2025-01-11 10:38:05,597]    INFO [count] Using index index_std.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202614/ from\n",
            "[2025-01-11 10:38:05,597]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S15_L001_R1_001.fastq.gz\n",
            "[2025-01-11 10:38:05,597]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S15_L001_R2_001.fastq.gz\n",
            "[2025-01-11 10:38:05,597]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S15_L002_R1_001.fastq.gz\n",
            "[2025-01-11 10:38:05,597]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S15_L002_R2_001.fastq.gz\n",
            "[2025-01-11 10:38:05,597]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S8_L001_R1_001.fastq.gz\n",
            "[2025-01-11 10:38:05,597]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S8_L001_R2_001.fastq.gz\n",
            "[2025-01-11 10:38:05,597]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S8_L002_R1_001.fastq.gz\n",
            "[2025-01-11 10:38:05,597]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S8_L002_R2_001.fastq.gz\n",
            "[2025-01-11 11:03:59,742]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202614/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202614/tmp/output.s.bus\n",
            "[2025-01-11 11:05:42,537]    INFO [count] On-list not provided\n",
            "[2025-01-11 11:05:42,537]    INFO [count] Copying pre-packaged 10XV3 on-list to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202614/\n",
            "[2025-01-11 11:05:43,287]    INFO [count] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202614/tmp/output.s.bus\n",
            "[2025-01-11 11:06:14,048]    INFO [count] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202614/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202614/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202614/10x_version3_whitelist.txt\n",
            "[2025-01-11 11:07:05,755]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202614/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202614/output.unfiltered.bus\n",
            "[2025-01-11 11:08:10,683]    INFO [count] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202614/counts_unfiltered/cells_x_tcc from BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202614/output.unfiltered.bus\n",
            "[2025-01-11 11:08:51,909]    INFO [count] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202614/counts_unfiltered/cells_x_tcc.mtx\n",
            "[2025-01-11 11:08:55,677]    INFO [count] Writing matrix to h5ad /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202614/counts_unfiltered/adata.h5ad\n"
          ]
        }
      ],
      "source": [
        "!kb count -i index_std.idx -g t2g_std.txt --overwrite -x 10xv3 --tcc --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202614/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S15_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S15_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S15_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S15_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S8_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S8_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S8_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT352_mm10_S8_L002_R2_001.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "background_save": true
        },
        "id": "5C5E22X1l09x",
        "outputId": "4bebb74f-53be-48dd-e7cc-d3f97c233879"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[2025-01-11 11:09:10,947]    INFO [count] Using index index_std.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202615/ from\n",
            "[2025-01-11 11:09:10,948]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S16_L001_R1_001.fastq.gz\n",
            "[2025-01-11 11:09:10,948]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S16_L001_R2_001.fastq.gz\n",
            "[2025-01-11 11:09:10,948]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S16_L002_R1_001.fastq.gz\n",
            "[2025-01-11 11:09:10,948]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S16_L002_R2_001.fastq.gz\n",
            "[2025-01-11 11:09:10,948]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S9_L001_R1_001.fastq.gz\n",
            "[2025-01-11 11:09:10,948]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S9_L001_R2_001.fastq.gz\n",
            "[2025-01-11 11:09:10,948]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S9_L002_R1_001.fastq.gz\n",
            "[2025-01-11 11:09:10,948]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S9_L002_R2_001.fastq.gz\n",
            "[2025-01-11 11:46:21,799]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202615/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202615/tmp/output.s.bus\n",
            "[2025-01-11 11:49:11,333]    INFO [count] On-list not provided\n",
            "[2025-01-11 11:49:11,333]    INFO [count] Copying pre-packaged 10XV3 on-list to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202615/\n",
            "[2025-01-11 11:49:12,103]    INFO [count] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202615/tmp/output.s.bus\n",
            "[2025-01-11 11:49:53,485]    INFO [count] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202615/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202615/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202615/10x_version3_whitelist.txt\n",
            "[2025-01-11 11:51:05,328]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202615/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202615/output.unfiltered.bus\n",
            "[2025-01-11 11:52:53,951]    INFO [count] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202615/counts_unfiltered/cells_x_tcc from BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202615/output.unfiltered.bus\n",
            "[2025-01-11 11:53:52,772]    INFO [count] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202615/counts_unfiltered/cells_x_tcc.mtx\n",
            "[2025-01-11 11:53:57,389]    INFO [count] Writing matrix to h5ad /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202615/counts_unfiltered/adata.h5ad\n"
          ]
        }
      ],
      "source": [
        "!kb count -i index_std.idx -g t2g_std.txt --overwrite -x 10xv3 --tcc --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX13202615/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S16_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S16_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S16_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S16_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S9_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S9_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S9_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT353_mm10_S9_L002_R2_001.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "background_save": true
        },
        "id": "ea_7t0Vel09x",
        "outputId": "bd21f8f4-3480-4915-ec69-2e67c7199ef3"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[2025-01-11 11:54:14,428]    INFO [count] Using index index_std.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265374/ from\n",
            "[2025-01-11 11:54:14,428]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT356_mm10_S19_L001_R1_001.fastq.gz\n",
            "[2025-01-11 11:54:14,428]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT356_mm10_S19_L001_R2_001.fastq.gz\n",
            "[2025-01-11 11:54:14,428]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT356_mm10_S19_L002_R1_001.fastq.gz\n",
            "[2025-01-11 11:54:14,428]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT356_mm10_S19_L002_R2_001.fastq.gz\n",
            "[2025-01-11 11:54:14,428]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT356_mm10_S3_L001_R1_001.fastq.gz\n",
            "[2025-01-11 11:54:14,428]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT356_mm10_S3_L001_R2_001.fastq.gz\n",
            "[2025-01-11 12:26:51,022]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265374/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265374/tmp/output.s.bus\n",
            "[2025-01-11 12:30:59,428]    INFO [count] On-list not provided\n",
            "[2025-01-11 12:30:59,429]    INFO [count] Copying pre-packaged 10XV3 on-list to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265374/\n",
            "[2025-01-11 12:31:00,168]    INFO [count] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265374/tmp/output.s.bus\n",
            "[2025-01-11 12:31:53,867]    INFO [count] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265374/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265374/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265374/10x_version3_whitelist.txt\n",
            "[2025-01-11 12:33:41,766]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265374/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265374/output.unfiltered.bus\n",
            "[2025-01-11 12:36:55,559]    INFO [count] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265374/counts_unfiltered/cells_x_tcc from BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265374/output.unfiltered.bus\n",
            "[2025-01-11 12:38:45,531]    INFO [count] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265374/counts_unfiltered/cells_x_tcc.mtx\n",
            "[2025-01-11 12:38:58,555]    INFO [count] Writing matrix to h5ad /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265374/counts_unfiltered/adata.h5ad\n"
          ]
        }
      ],
      "source": [
        "!kb count -i index_std.idx -g t2g_std.txt --overwrite -x 10xv3 --tcc --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265374/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT356_mm10_S19_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT356_mm10_S19_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT356_mm10_S19_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT356_mm10_S19_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT356_mm10_S3_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT356_mm10_S3_L001_R2_001.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "background_save": true
        },
        "id": "m7bMdwIvl09x",
        "outputId": "7a106550-5637-4841-e192-832c823a2591"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[2025-01-11 12:39:21,603]    INFO [count] Using index index_std.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265375/ from\n",
            "[2025-01-11 12:39:21,603]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT357_mm10_S20_L001_R1_001.fastq.gz\n",
            "[2025-01-11 12:39:21,603]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT357_mm10_S20_L001_R2_001.fastq.gz\n",
            "[2025-01-11 12:39:21,604]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT357_mm10_S20_L002_R1_001.fastq.gz\n",
            "[2025-01-11 12:39:21,604]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT357_mm10_S20_L002_R2_001.fastq.gz\n",
            "[2025-01-11 12:39:21,604]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT357_mm10_S4_L001_R1_001.fastq.gz\n",
            "[2025-01-11 12:39:21,604]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT357_mm10_S4_L001_R2_001.fastq.gz\n",
            "[2025-01-11 13:07:50,219]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265375/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265375/tmp/output.s.bus\n",
            "[2025-01-11 13:12:08,846]    INFO [count] On-list not provided\n",
            "[2025-01-11 13:12:08,846]    INFO [count] Copying pre-packaged 10XV3 on-list to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265375/\n",
            "[2025-01-11 13:12:09,605]    INFO [count] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265375/tmp/output.s.bus\n",
            "[2025-01-11 13:13:05,309]    INFO [count] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265375/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265375/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265375/10x_version3_whitelist.txt\n",
            "[2025-01-11 13:14:46,308]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265375/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265375/output.unfiltered.bus\n",
            "[2025-01-11 13:17:31,133]    INFO [count] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265375/counts_unfiltered/cells_x_tcc from BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265375/output.unfiltered.bus\n",
            "[2025-01-11 13:18:56,742]    INFO [count] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265375/counts_unfiltered/cells_x_tcc.mtx\n",
            "[2025-01-11 13:19:02,966]    INFO [count] Writing matrix to h5ad /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265375/counts_unfiltered/adata.h5ad\n"
          ]
        }
      ],
      "source": [
        "!kb count -i index_std.idx -g t2g_std.txt --overwrite -x 10xv3 --tcc --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265375/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT357_mm10_S20_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT357_mm10_S20_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT357_mm10_S20_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT357_mm10_S20_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT357_mm10_S4_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT357_mm10_S4_L001_R2_001.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "background_save": true
        },
        "id": "9nHA6kmKl09x",
        "outputId": "292cf626-c129-4c97-fb75-12a5444d3946"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "[2025-01-11 13:19:21,969]    INFO [count] Using index index_std.idx to generate BUS file to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265376/ from\n",
            "[2025-01-11 13:19:21,969]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT358_mm10_S11_L001_R1_001.fastq.gz\n",
            "[2025-01-11 13:19:21,969]    INFO [count]         /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT358_mm10_S11_L001_R2_001.fastq.gz\n",
            "[2025-01-11 14:03:54,323]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265376/output.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265376/tmp/output.s.bus\n",
            "[2025-01-11 14:10:02,209]    INFO [count] On-list not provided\n",
            "[2025-01-11 14:10:02,210]    INFO [count] Copying pre-packaged 10XV3 on-list to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265376/\n",
            "[2025-01-11 14:10:03,039]    INFO [count] Inspecting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265376/tmp/output.s.bus\n",
            "[2025-01-11 14:11:27,923]    INFO [count] Correcting BUS records in /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265376/tmp/output.s.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265376/tmp/output.s.c.bus with on-list /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265376/10x_version3_whitelist.txt\n",
            "[2025-01-11 14:14:17,197]    INFO [count] Sorting BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265376/tmp/output.s.c.bus to /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265376/output.unfiltered.bus\n",
            "[2025-01-11 14:19:22,176]    INFO [count] Generating count matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265376/counts_unfiltered/cells_x_tcc from BUS file /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265376/output.unfiltered.bus\n",
            "[2025-01-11 14:21:31,136]    INFO [count] Reading matrix /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265376/counts_unfiltered/cells_x_tcc.mtx\n",
            "[2025-01-11 14:21:37,149]    INFO [count] Writing matrix to h5ad /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265376/counts_unfiltered/adata.h5ad\n"
          ]
        }
      ],
      "source": [
        "!kb count -i index_std.idx -g t2g_std.txt --overwrite -x 10xv3 --tcc --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265376/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT358_mm10_S11_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT358_mm10_S11_L001_R2_001.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "0VSrMl9bl09x"
      },
      "outputs": [],
      "source": [
        "!kb count -i index_std.idx -g t2g_std.txt --overwrite -x 10xv3 --tcc --h5ad -o /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX12265377/ /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S10_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S10_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S10_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S10_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S6_L001_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S6_L001_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S6_L002_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S6_L002_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S6_L003_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S6_L003_R2_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S6_L004_R1_001.fastq.gz /content/drive/MyDrive/pituitary_atlas/Unstructured/fastqs/devriendt/scrEXT364_mm10_S6_L004_R2_001.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "gwPmNCOse7Ce"
      },
      "outputs": [],
      "source": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "nOGWL7nVz_fS"
      },
      "source": [
        "# Loop for TCC processing"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "v6ojif4GJdyt"
      },
      "outputs": [],
      "source": [
        "import os\n",
        "datasets_already_downloaded = os.listdir('/content/drive/MyDrive/pituitary_atlas/processed/tcc/')"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Hi3vAVHHGpyZ",
        "outputId": "db5495c8-43e8-48c6-eea2-8deb52e162f9"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "SRA: SRX29174030 . Experiment type is not sc: sn\n",
            "SRA: SRX29174029 . Experiment type is not sc: sn\n",
            "SRA: SRX29174028 . Experiment type is not sc: sn\n",
            "SRA: SRX29174027 . Experiment type is not sc: sn\n",
            "\n",
            "  SRA_ID: SRX25982299\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX25982300\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX25982301\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX25982298\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX25982297\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX25982296\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX25982295\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX25982294\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX26708357\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "SRA: SRX25736229 . Experiment type is not sc: sn\n",
            "SRA: SRX25736228 . Experiment type is not sc: sn\n",
            "SRA: SRX25736227 . Experiment type is not sc: sn\n",
            "SRA: SRX25736226 . Experiment type is not sc: sn\n",
            "SRA: SRX25736225 . Experiment type is not sc: sn\n",
            "SRA: SRX25736224 . Experiment type is not sc: sn\n",
            "SRA: SRX25736223 . Experiment type is not sc: sn\n",
            "SRA: SRX25736222 . Experiment type is not sc: sn\n",
            "SRA: SRX25736221 . Experiment type is not sc: sn\n",
            "SRA: SRX25736220 . Experiment type is not sc: sn\n",
            "SRA: SRX25736219 . Experiment type is not sc: sn\n",
            "\n",
            "  SRA_ID: ERX12265377\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: ERX12265376\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: ERX12265375\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: ERX12265374\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: ERX13202615\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: ERX13202614\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: ERX13202613\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: ERX13202612\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX18716675\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX18716674\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX18716673\n",
            "  experiment_type: sc\n",
            "  tech: 10XV2\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX18716672\n",
            "  experiment_type: sc\n",
            "  tech: 10XV2\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "SRA: SRX21902370 . Experiment type is not sc: multi_rna\n",
            "SRA: SRX21902369 . Experiment type is not sc: multi_rna\n",
            "SRA: SRX21902368 . Experiment type is not sc: multi_rna\n",
            "SRA: SRX21902367 . Experiment type is not sc: multi_rna\n",
            "SRA: SRX21902366 . Experiment type is not sc: multi_rna\n",
            "SRA: SRX21902365 . Experiment type is not sc: multi_rna\n",
            "SRA: SRX21902364 . Experiment type is not sc: multi_rna\n",
            "SRA: SRX21902356 . Experiment type is not sc: multi_rna\n",
            "SRA: SRX21902355 . Experiment type is not sc: multi_rna\n",
            "SRA: SRX21902354 . Experiment type is not sc: multi_rna\n",
            "SRA: SRX21902353 . Experiment type is not sc: multi_rna\n",
            "SRA: SRX21902352 . Experiment type is not sc: multi_rna\n",
            "SRA: SRX21902351 . Experiment type is not sc: multi_rna\n",
            "SRA: SRX21902350 . Experiment type is not sc: multi_rna\n",
            "SRA: SRX21902349 . Experiment type is not sc: multi_rna\n",
            "SRA: SRX21902348 . Experiment type is not sc: multi_rna\n",
            "\n",
            "  SRA_ID: SRX19698100\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX19698101\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX19698102\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX25839821\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX25839820\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX21636477\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX21636476\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX22219346\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX22219345\n",
            "  experiment_type: sc\n",
            "  tech: 10XV2\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX24151439\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX24151438\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX24151437\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX24151436\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX24151435\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX24151434\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX24151433\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX24151432\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX25532499\n",
            "  experiment_type: sc\n",
            "  tech: 10XV2\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX25532498\n",
            "  experiment_type: sc\n",
            "  tech: 10XV2\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX7377166\n",
            "  experiment_type: sc\n",
            "  tech: 10XV2\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX7377164\n",
            "  experiment_type: sc\n",
            "  tech: 10XV2\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "SRA: SRX20679984 . Experiment type is not sc: multi_rna\n",
            "\n",
            "  SRA_ID: SRX9637224\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX8497094\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX8497093\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX8497092\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX8602212\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX8602211\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX15581371\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX15581370\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX18224109\n",
            "  experiment_type: sc\n",
            "  tech: 10XV2\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX18224108\n",
            "  experiment_type: sc\n",
            "  tech: 10XV2\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX18224107\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX18224106\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX18224105\n",
            "  experiment_type: sc\n",
            "  tech: 10XV2\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX18224104\n",
            "  experiment_type: sc\n",
            "  tech: 10XV2\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "SRA: SRX13290062 . Experiment type is not sc: sn\n",
            "SRA: SRX13290061 . Experiment type is not sc: sn\n",
            "SRA: SRX13290060 . Experiment type is not sc: sn\n",
            "SRA: SRX13290059 . Experiment type is not sc: sn\n",
            "SRA: SRX13290058 . Experiment type is not sc: sn\n",
            "SRA: SRX13290057 . Experiment type is not sc: sn\n",
            "\n",
            "  SRA_ID: SRX21986116\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX21986115\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX21986114\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX21986113\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX21986112\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX21986111\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX21986110\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX21986109\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX21986108\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX21986107\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX21986106\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX21986105\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX21170825\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX21170824\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX21170823\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: ERX7593484\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: ERX7593483\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: ERX7593482\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: ERX7593481\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: ERX4978544\n",
            "  experiment_type: sc\n",
            "  tech: 10XV2\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: ERX4978543\n",
            "  experiment_type: sc\n",
            "  tech: 10XV2\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: ERX4978542\n",
            "  experiment_type: sc\n",
            "  tech: 10XV2\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: ERX4978541\n",
            "  experiment_type: sc\n",
            "  tech: 10XV2\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: ERX4978540\n",
            "  experiment_type: sc\n",
            "  tech: 10XV2\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: ERX4978539\n",
            "  experiment_type: sc\n",
            "  tech: 10XV2\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: ERX4978538\n",
            "  experiment_type: sc\n",
            "  tech: 10XV2\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: ERX4978537\n",
            "  experiment_type: sc\n",
            "  tech: 10XV2\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX21241801\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX21241800\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX21241799\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX21241798\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX9528565\n",
            "  experiment_type: sc\n",
            "  tech: 10XV2\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX9528564\n",
            "  experiment_type: sc\n",
            "  tech: 10XV2\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX7874746\n",
            "  experiment_type: sc\n",
            "  tech: DROPSEQ\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX7874745\n",
            "  experiment_type: sc\n",
            "  tech: DROPSEQ\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX7874744\n",
            "  experiment_type: sc\n",
            "  tech: DROPSEQ\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX7874743\n",
            "  experiment_type: sc\n",
            "  tech: DROPSEQ\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX7874742\n",
            "  experiment_type: sc\n",
            "  tech: DROPSEQ\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX7874741\n",
            "  experiment_type: sc\n",
            "  tech: DROPSEQ\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX7874740\n",
            "  experiment_type: sc\n",
            "  tech: DROPSEQ\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX7874739\n",
            "  experiment_type: sc\n",
            "  tech: DROPSEQ\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX7874738\n",
            "  experiment_type: sc\n",
            "  tech: 10XV2\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX7874737\n",
            "  experiment_type: sc\n",
            "  tech: 10XV2\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX6699503\n",
            "  experiment_type: sc\n",
            "  tech: 10XV2\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX6699502\n",
            "  experiment_type: sc\n",
            "  tech: 10XV2\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX5292053\n",
            "  experiment_type: sc\n",
            "  tech: 10XV2\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX4736338\n",
            "  experiment_type: sc\n",
            "  tech: 10XV2\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX4736337\n",
            "  experiment_type: sc\n",
            "  tech: 10XV2\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "SRA: SRX8489842 . Experiment type is not sc: sn\n",
            "SRA: SRX8489841 . Experiment type is not sc: sn\n",
            "SRA: SRX8489840 . Experiment type is not sc: sn\n",
            "SRA: SRX8489839 . Experiment type is not sc: sn\n",
            "SRA: SRX8489838 . Experiment type is not sc: sn\n",
            "SRA: SRX8489837 . Experiment type is not sc: sn\n",
            "SRA: SRX8489836 . Experiment type is not sc: sn\n",
            "SRA: SRX8489835 . Experiment type is not sc: sn\n",
            "SRA: SRX8489834 . Experiment type is not sc: sn\n",
            "\n",
            "  SRA_ID: SRX8489824\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX8489823\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX8489822\n",
            "  experiment_type: sc\n",
            "  tech: 10XV2\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX8489821\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX8489820\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX8489819\n",
            "  experiment_type: sc\n",
            "  tech: 10XV3\n",
            "  \n",
            "SRA_ID already aligned for tcc\n",
            "\n",
            "  SRA_ID: SRX8489818\n",
            "  experiment_type: sc\n",
            "  tech: 10XV2\n",
            "  \n",
            "SRA_ID already aligned for tcc\n"
          ]
        }
      ],
      "source": [
        "#iterating through the table and getting datasets\n",
        "import subprocess\n",
        "for i in range(len(df)-1,-1,-1):\n",
        "  SRA_ID = df.iloc[i]['SRA_ID']\n",
        "  experiment_type = df.iloc[i]['Modality']\n",
        "  tech = df.iloc[i]['10X version']\n",
        "  #if experiment_type is not sc print SRA_ID and continue\n",
        "  if experiment_type != \"sc\":\n",
        "    print(f\"SRA: {SRA_ID} . Experiment type is not sc: {experiment_type}\")\n",
        "    continue\n",
        "  #print all of these\n",
        "  print(\"\"\"\n",
        "  SRA_ID: {}\n",
        "  experiment_type: {}\n",
        "  tech: {}\n",
        "  \"\"\".format(SRA_ID,experiment_type,tech))\n",
        "\n",
        "  #check if SRA_ID is already downloaded\n",
        "  if SRA_ID in datasets_already_downloaded:\n",
        "    print(\"SRA_ID already aligned for tcc\")\n",
        "    continue\n",
        "\n",
        "  #if no folder in tcc\n",
        "  if not os.path.exists('/content/drive/MyDrive/pituitary_atlas/processed/tcc/' + SRA_ID):\n",
        "    os.mkdir('/content/drive/MyDrive/pituitary_atlas/processed/tcc/' + SRA_ID)\n",
        "\n",
        "  directory = '/content/drive/MyDrive/pituitary_atlas/raw/' + SRA_ID\n",
        "  files = os.listdir(directory)\n",
        "  all_fastqs = []\n",
        "  for file in files:\n",
        "      if file.endswith('.fastq.gz'):\n",
        "          all_fastqs.append(directory + '/' + file)\n",
        "  #if there are fastq files with the same string before .sralite, only keep the two largest of them in the list\n",
        "  #does any of the files have lite in name\n",
        "  lite_name = True if any(x for x in all_fastqs if 'lite' in x) else False\n",
        "  if lite_name:\n",
        "    id_tags = [x.split('lite')[0] for x in all_fastqs]\n",
        "\n",
        "  else:\n",
        "    id_tags = [x.split('_R1_')[0] for x in all_fastqs if '_R1_' in x]\n",
        "\n",
        "  print(id_tags)\n",
        "  all_fastqs_final = []\n",
        "  unique_tags= list(set(id_tags))\n",
        "  for tag in unique_tags:\n",
        "    fastqs_with_tag = [x for x in all_fastqs if tag in x]\n",
        "    if len(fastqs_with_tag) > 1:\n",
        "      fastqs_with_tag.sort(key=lambda x: os.path.getsize(x), reverse=True)\n",
        "      all_fastqs_final.append(fastqs_with_tag[1])\n",
        "      all_fastqs_final.append(fastqs_with_tag[0])\n",
        "\n",
        "\n",
        "  all_fastqs = all_fastqs_final\n",
        "  #if there arent at least 2 fastqs then continue\n",
        "  if len(all_fastqs) < 2:\n",
        "    print(\"Not enough fastqs\")\n",
        "    continue\n",
        "\n",
        "  if experiment_type == \"sc\":\n",
        "    print(\"Running kb count\")\n",
        "    output_dir = f\"/content/drive/MyDrive/pituitary_atlas/processed/tcc/{SRA_ID}/\"\n",
        "    kbcount_tcc_command1 = f\"kb count -i index_std.idx -g t2g_std.txt --overwrite -x {tech} --tcc --h5ad -o {output_dir} {' '.join(all_fastqs)}\"\n",
        "    print(kbcount_tcc_command1)\n",
        "    subprocess.run(kbcount_tcc_command1, shell=True)\n",
        "\n",
        "\n",
        "  #elif experiment_type == \"multi_rna\":\n",
        "\n",
        "    #from all_fastqs remove ones that contain \"atac\"\n",
        "  #  all_fastqs = [x for x in all_fastqs if \"atac\" not in x]\n",
        "\n",
        "\n",
        "  #  print(\"Running kb count\")\n",
        "  #  output_dir = f\"/content/drive/MyDrive/pituitary_atlas/processed/tcc/{SRA_ID}/\"\n",
        "  #  kbcount_tcc_command1 = f\"kb count -i index_std.idx -g t2g_std.txt --overwrite -x {tech} --tcc --h5ad -w /content/10xMultiome/gex_737K-arc-v1.txt -o {output_dir} {' '.join(all_fastqs)}\"\n",
        "  #  print(kbcount_tcc_command1)\n",
        "  #  subprocess.run(kbcount_tcc_command1, shell=True)\n",
        "  else:\n",
        "    print(f\"Experiment not single cell: {experiment_type}\")\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "_sK20uZb1oop"
      },
      "source": [
        "# Downloading new Cheung et al (2025) ageing data"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "vEYYGAsJ1sAF",
        "outputId": "401448d7-1019-4e34-cc23-39619f98500e"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "/content/drive/MyDrive/pituitary_atlas/raw/SRX29174027\n",
            "\n",
            "2025-06-27T14:44:17 prefetch.2.11.3: Current preference is set to retrieve SRA Normalized Format files with full base quality scores.\n",
            "2025-06-27T14:44:18 prefetch.2.11.3: 1) Downloading 'SRR33975506'...\n",
            "2025-06-27T14:44:18 prefetch.2.11.3: SRA Normalized Format file is being retrieved, if this is different from your preference, it may be due to current file availability.\n",
            "2025-06-27T14:44:18 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2025-06-27T15:16:08 prefetch.2.11.3:  HTTPS download succeed\n",
            "2025-06-27T15:16:08 prefetch.2.11.3:   verifying 'SRR33975506'...\n",
            "2025-06-27T15:18:30 prefetch.2.11.3:  'SRR33975506' is valid\n",
            "2025-06-27T15:18:30 prefetch.2.11.3: 1) 'SRR33975506' was downloaded successfully\n",
            "2025-06-27T15:18:31 prefetch.2.11.3: 'SRR33975506' has 0 unresolved dependencies\n",
            "total 4.0K\n",
            "drwx------ 2 root root 4.0K Jun 27 15:18 SRR33975506\n"
          ]
        }
      ],
      "source": [
        "# 2. Define the target directory in Google Drive\n",
        "output_dir = \"/content/drive/MyDrive/pituitary_atlas/raw/SRX29174027/\"\n",
        "\n",
        "# 3. Create the directory if it doesn't exist\n",
        "# The -p flag ensures that parent directories are also created if they don't exist\n",
        "!mkdir -p \"$output_dir\"\n",
        "\n",
        "# 4. Change to the target directory for prefetch\n",
        "# This is important so prefetch downloads directly into the desired location\n",
        "%cd \"$output_dir\"\n",
        "\n",
        "# 5. Pre-fetch the .sra files\n",
        "# --max-size 0 means no size limit\n",
        "# Ensure you are in the correct directory before running prefetch\n",
        "!prefetch --max-size 0 SRR33975506\n",
        "\n",
        "# Optional: Verify the downloaded files\n",
        "!ls -lh \"$output_dir\""
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "YVZw8ziJUx2S",
        "outputId": "47c839bb-e337-46b1-e363-80ed26819ad5"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "/content/drive/MyDrive/pituitary_atlas/raw/SRX29174027/SRR33975506\n",
            "Checking for SRR33975506.sra in /content/drive/MyDrive/pituitary_atlas/raw/SRX29174027/SRR33975506/:\n",
            "-rw------- 1 root root 26G Jun 27 15:16 SRR33975506.sra\n",
            "\n",
            "Starting fasterq-dump...\n"
          ]
        }
      ],
      "source": [
        "# Define the SRA file path (where your SRR33975506.sra is located)\n",
        "# Based on your prompt, it seems SRR33975506.sra is DIRECTLY inside this folder.\n",
        "sra_input_dir = \"/content/drive/MyDrive/pituitary_atlas/raw/SRX29174027/SRR33975506/\"\n",
        "\n",
        "# Define the final output directory for FASTQ files (can be the same as sra_input_dir if you prefer)\n",
        "fastq_output_dir = \"/content/drive/MyDrive/pituitary_atlas/raw/SRX29174027/SRR33975506/fastqs/\" # Let's put FASTQs in a separate 'fastqs' folder for organization\n",
        "\n",
        "# Define a temporary directory *on Google Drive* for fasterq-dump's intermediate files\n",
        "# This is CRITICAL for large files to avoid exhausting Colab's local storage\n",
        "temp_dir_on_drive = \"/content/drive/MyDrive/pituitary_atlas/raw/SRX29174027/SRR33975506/temp_fasterq_dump/\" # Create a dedicated temp folder on Drive\n",
        "\n",
        "# Create all necessary directories\n",
        "!mkdir -p \"$sra_input_dir\"\n",
        "!mkdir -p \"$fastq_output_dir\"\n",
        "!mkdir -p \"$temp_dir_on_drive\" # Create the temp directory on Drive\n",
        "\n",
        "# Change the current working directory to where the SRA files are.\n",
        "# This makes it easier for fasterq-dump to find them by just their name.\n",
        "%cd \"$sra_input_dir\"\n",
        "\n",
        "# Verify the SRA file exists in the current directory\n",
        "print(f\"Checking for SRR33975506.sra in {sra_input_dir}:\")\n",
        "!ls -lh SRR33975506.sra\n",
        "\n",
        "!fastq-dump SRR33975506.sra -O \"$fastq_output_dir\"  --split-files --gzip\n",
        "\n",
        "print(\"\\nFinished fasterq-dump.\")\n",
        "\n",
        "# Optional: Verify the generated FASTQ files\n",
        "print(f\"\\nFASTQ files generated in {fastq_output_dir}:\")\n",
        "!ls -lh \"$fastq_output_dir\"/*.fastq\n",
        "\n",
        "# Optional: Clean up temporary files on Drive if successful\n",
        "# Be careful with rm -rf! Only uncomment if you are sure.\n",
        "# !rm -rf \"$temp_dir_on_drive\"/*\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "iSAaKpIW2iMv",
        "outputId": "815647f6-3a35-4376-e19d-bf5b7a16266f"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "/content/drive/MyDrive/pituitary_atlas/raw/SRX29174028\n",
            "\n",
            "2025-06-27T15:18:33 prefetch.2.11.3: Current preference is set to retrieve SRA Normalized Format files with full base quality scores.\n",
            "2025-06-27T15:18:34 prefetch.2.11.3: 1) Downloading 'SRR33975505'...\n",
            "2025-06-27T15:18:34 prefetch.2.11.3: SRA Normalized Format file is being retrieved, if this is different from your preference, it may be due to current file availability.\n",
            "2025-06-27T15:18:34 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2025-06-27T15:49:11 prefetch.2.11.3:  HTTPS download succeed\n",
            "2025-06-27T15:49:11 prefetch.2.11.3:   verifying 'SRR33975505'...\n",
            "2025-06-27T15:51:37 prefetch.2.11.3:  'SRR33975505' is valid\n",
            "2025-06-27T15:51:37 prefetch.2.11.3: 1) 'SRR33975505' was downloaded successfully\n",
            "2025-06-27T15:51:38 prefetch.2.11.3: 'SRR33975505' has 0 unresolved dependencies\n",
            "total 4.0K\n",
            "drwx------ 2 root root 4.0K Jun 27 15:51 SRR33975505\n"
          ]
        }
      ],
      "source": [
        "# 2. Define the target directory in Google Drive\n",
        "output_dir = \"/content/drive/MyDrive/pituitary_atlas/raw/SRX29174028/\"\n",
        "\n",
        "# 3. Create the directory if it doesn't exist\n",
        "# The -p flag ensures that parent directories are also created if they don't exist\n",
        "!mkdir -p \"$output_dir\"\n",
        "\n",
        "# 4. Change to the target directory for prefetch\n",
        "# This is important so prefetch downloads directly into the desired location\n",
        "%cd \"$output_dir\"\n",
        "\n",
        "# 5. Pre-fetch the .sra files\n",
        "# --max-size 0 means no size limit\n",
        "# Ensure you are in the correct directory before running prefetch\n",
        "!prefetch --max-size 0 SRR33975505\n",
        "\n",
        "# Optional: Verify the downloaded files\n",
        "!ls -lh \"$output_dir\""
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "rjsp-5zzZYlj",
        "outputId": "92519c65-3c75-49e1-ef9c-44cdf74af371"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "/content/drive/MyDrive/pituitary_atlas/raw/SRX29174028/SRR33975505\n",
            "1900-01-00T00:00:00 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1: storage exhausted while writing file within file system module - system bad file descriptor error fd='11'\n",
            "1900-01-00T00:00:00 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1: storage exhausted while writing file within file system module - system bad file descriptor error fd='26'\n",
            "1900-01-00T00:00:00 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1: storage exhausted while writing file within file system module - system bad file descriptor error fd='17'\n",
            "1900-01-00T00:00:00 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1: storage exhausted while writing file within file system module - system bad file descriptor error fd='24'\n",
            "1900-01-00T00:00:00 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1: storage exhausted while writing file within file system module - system bad file descriptor error fd='19'\n",
            "1900-01-00T00:00:00 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1: storage exhausted while writing file within file system module - system bad file descriptor error fd='21'\n",
            "1900-01-00T00:00:00 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1: storage exhausted while writing file within file system module - system bad file descriptor error fd='15'\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1: storage exhausted while writing file within file system module - system bad file descriptor error fd='14'\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1: storage exhausted while writing file within file system module - system bad file descriptor error fd='17'\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1 rcFS,rcFile,rcWriting,rcStorage,rcExhausted)\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:108:KSysFileDestroy_v1:  unknown while destroying file within file system module - unknown system error 'No space left on device(28)'\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:108:KSysFileDestroy_v1 rcFS,rcFile,rcDestroying,rcNoObj,rcUnknown)\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1: storage exhausted while writing file within file system module - system bad file descriptor error fd='21'\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1 rcFS,rcFile,rcWriting,rcStorage,rcExhausted)\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1: storage exhausted while writing file within file system module - system bad file descriptor error fd='11'\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1 rcFS,rcFile,rcWriting,rcStorage,rcExhausted)\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1: storage exhausted while writing file within file system module - system bad file descriptor error fd='15'\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1 rcFS,rcFile,rcWriting,rcStorage,rcExhausted)\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1: storage exhausted while writing file within file system module - system bad file descriptor error fd='19'\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1 rcFS,rcFile,rcWriting,rcStorage,rcExhausted)\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1: storage exhausted while writing file within file system module - system bad file descriptor error fd='24'\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1 rcFS,rcFile,rcWriting,rcStorage,rcExhausted)\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:108:KSysFileDestroy_v1:  unknown while destroying file within file system module - unknown system error 'No space left on device(28)'\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:108:KSysFileDestroy_v1 rcFS,rcFile,rcDestroying,rcNoObj,rcUnknown)\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:108:KSysFileDestroy_v1:  unknown while destroying file within file system module - unknown system error 'No space left on device(28)'\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:108:KSysFileDestroy_v1:  unknown while destroying file within file system module - unknown system error 'No space left on device(28)'\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:108:KSysFileDestroy_v1 rcFS,rcFile,rcDestroying,rcNoObj,rcUnknown)\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:108:KSysFileDestroy_v1 rcFS,rcFile,rcDestroying,rcNoObj,rcUnknown)\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1: storage exhausted while writing file within file system module - system bad file descriptor error fd='20'\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:108:KSysFileDestroy_v1:  unknown while destroying file within file system module - unknown system error 'No space left on device(28)'\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1 rcFS,rcFile,rcWriting,rcStorage,rcExhausted)\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1 rcFS,rcFile,rcDestroying,rcNoObj,rcUnknown)\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:108:KSysFileDestroy_v1:  unknown while destroying file within file system module - unknown system error 'No space left on device(28)'\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1 rcFS,rcFile,rcDestroying,rcNoObj,rcUnknown)\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1: storage exhausted while writing file within file system module - system bad file descriptor error fd='16'\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1 rcFS,rcFile,rcWriting,rcStorage,rcExhausted)\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1: storage exhausted while writing file within file system module - system bad file descriptor error fd='14'\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1 rcFS,rcFile,rcWriting,rcStorage,rcExhausted)\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1: storage exhausted while writing file within file system module - system bad file descriptor error fd='26'\n",
            "2025-06-27T22:07:44 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1 rcFS,rcFile,rcWriting,rcStorage,rcExhausted)\n",
            "fasterq-dump quit with error code 3\n"
          ]
        }
      ],
      "source": [
        "output_dir = \"/content/drive/MyDrive/pituitary_atlas/raw/SRX29174028/SRR33975505/\"\n",
        "\n",
        "# 3. Create the directory if it doesn't exist\n",
        "# The -p flag ensures that parent directories are also created if they don't exist\n",
        "!mkdir -p \"$output_dir\"\n",
        "\n",
        "# 4. Change to the target directory for prefetch\n",
        "# This is important so prefetch downloads directly into the desired location\n",
        "%cd \"$output_dir\"\n",
        "!fasterq-dump SRR33975505.sra -O \"$output_dir\" -e 8 --split-files"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "of-WV0wl1sFp",
        "outputId": "3961ced9-7bb4-40c4-982e-80b3d31c20cd"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "/content/drive/MyDrive/pituitary_atlas/raw/SRX29174029\n",
            "\n",
            "2025-06-27T15:51:40 prefetch.2.11.3: Current preference is set to retrieve SRA Normalized Format files with full base quality scores.\n",
            "2025-06-27T15:51:41 prefetch.2.11.3: 1) Downloading 'SRR33975504'...\n",
            "2025-06-27T15:51:41 prefetch.2.11.3: SRA Normalized Format file is being retrieved, if this is different from your preference, it may be due to current file availability.\n",
            "2025-06-27T15:51:41 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2025-06-27T16:25:17 prefetch.2.11.3:  HTTPS download succeed\n",
            "2025-06-27T16:25:17 prefetch.2.11.3:   verifying 'SRR33975504'...\n"
          ]
        }
      ],
      "source": [
        "# 2. Define the target directory in Google Drive\n",
        "output_dir = \"/content/drive/MyDrive/pituitary_atlas/raw/SRX29174029/\"\n",
        "\n",
        "# 3. Create the directory if it doesn't exist\n",
        "# The -p flag ensures that parent directories are also created if they don't exist\n",
        "!mkdir -p \"$output_dir\"\n",
        "\n",
        "# 4. Change to the target directory for prefetch\n",
        "# This is important so prefetch downloads directly into the desired location\n",
        "%cd \"$output_dir\"\n",
        "\n",
        "# 5. Pre-fetch the .sra files\n",
        "# --max-size 0 means no size limit\n",
        "# Ensure you are in the correct directory before running prefetch\n",
        "!prefetch --max-size 0 SRR33975504\n",
        "\n",
        "# Optional: Verify the downloaded files\n",
        "!ls -lh \"$output_dir\""
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ZWWKwqAzZda8",
        "outputId": "c4d619e1-efb8-4968-847f-0bf22886663e"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "/content/drive/MyDrive/pituitary_atlas/raw/SRX29174029/SRR33975504\n",
            "2025-06-27T22:21:30 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1: storage exhausted while writing file within file system module - system bad file descriptor error fd='14'\n",
            "2025-06-27T22:21:30 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:108:KSysFileDestroy_v1:  unknown while destroying file within file system module - unknown system error 'No space left on device(28)'\n",
            "2025-06-27T22:21:30 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:108:KSysFileDestroy_v1 rcFS,rcFile,rcDestroying,rcNoObj,rcUnknown)\n",
            "2025-06-27T22:21:30 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:108:KSysFileDestroy_v1:  unknown while destroying file within file system module - unknown system error 'No space left on device(28)'\n",
            "2025-06-27T22:21:30 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:108:KSysFileDestroy_v1 rcFS,rcFile,rcDestroying,rcNoObj,rcUnknown)\n",
            "2025-06-27T22:21:30 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1: storage exhausted while writing file within file system module - system bad file descriptor error fd='14'\n",
            "2025-06-27T22:21:30 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1 rcFS,rcFile,rcWriting,rcStorage,rcExhausted)\n",
            "2025-06-27T22:21:30 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:108:KSysFileDestroy_v1:  unknown while destroying file within file system module - unknown system error 'No space left on device(28)'\n",
            "2025-06-27T22:21:30 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:108:KSysFileDestroy_v1 rcFS,rcFile,rcDestroying,rcNoObj,rcUnknown)\n",
            "2025-06-27T22:21:30 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:108:KSysFileDestroy_v1:  unknown while destroying file within file system module - unknown system error 'No space left on device(28)'\n",
            "2025-06-27T22:21:30 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:108:KSysFileDestroy_v1 rcFS,rcFile,rcDestroying,rcNoObj,rcUnknown)\n",
            "2025-06-27T22:21:30 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1: storage exhausted while writing file within file system module - system bad file descriptor error fd='12'\n",
            "2025-06-27T22:21:30 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1 rcFS,rcFile,rcWriting,rcStorage,rcExhausted)\n",
            "2025-06-27T22:21:30 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1: storage exhausted while writing file within file system module - system bad file descriptor error fd='19'\n",
            "2025-06-27T22:21:30 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1 rcFS,rcFile,rcWriting,rcStorage,rcExhausted)\n",
            "2025-06-27T22:21:30 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:108:KSysFileDestroy_v1:  unknown while destroying file within file system module - unknown system error 'No space left on device(28)'\n",
            "2025-06-27T22:21:30 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:108:KSysFileDestroy_v1 rcFS,rcFile,rcDestroying,rcNoObj,rcUnknown)\n",
            "fasterq-dump quit with error code 3\n"
          ]
        }
      ],
      "source": [
        "output_dir = \"/content/drive/MyDrive/pituitary_atlas/raw/SRX29174029/SRR33975504/\"\n",
        "\n",
        "# 3. Create the directory if it doesn't exist\n",
        "# The -p flag ensures that parent directories are also created if they don't exist\n",
        "!mkdir -p \"$output_dir\"\n",
        "\n",
        "# 4. Change to the target directory for prefetch\n",
        "# This is important so prefetch downloads directly into the desired location\n",
        "%cd \"$output_dir\"\n",
        "!fasterq-dump SRR33975504.sra -O \"$output_dir\" -e 8 --split-files\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "uMGomof01sIN"
      },
      "outputs": [],
      "source": [
        "# 2. Define the target directory in Google Drive\n",
        "output_dir = \"/content/drive/MyDrive/pituitary_atlas/raw/SRX29174030/\"\n",
        "\n",
        "# 3. Create the directory if it doesn't exist\n",
        "# The -p flag ensures that parent directories are also created if they don't exist\n",
        "!mkdir -p \"$output_dir\"\n",
        "\n",
        "# 4. Change to the target directory for prefetch\n",
        "# This is important so prefetch downloads directly into the desired location\n",
        "%cd \"$output_dir\"\n",
        "\n",
        "# 5. Pre-fetch the .sra files\n",
        "# --max-size 0 means no size limit\n",
        "# Ensure you are in the correct directory before running prefetch\n",
        "!prefetch --max-size 0 SRR33975503\n",
        "\n",
        "# Optional: Verify the downloaded files\n",
        "!ls -lh \"$output_dir\""
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "BKPdePu0ZeAE",
        "outputId": "80a66c43-64d7-4c30-e268-54b59f23156c"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "/content/drive/MyDrive/pituitary_atlas/raw/SRX29174030/SRR33975503\n",
            "2025-06-27T22:33:03 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1: storage exhausted while writing file within file system module - system bad file descriptor error fd='18'\n",
            "2025-06-27T22:33:03 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:108:KSysFileDestroy_v1:  unknown while destroying file within file system module - unknown system error 'No space left on device(28)'\n",
            "2025-06-27T22:33:03 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:108:KSysFileDestroy_v1 rcFS,rcFile,rcDestroying,rcNoObj,rcUnknown)\n",
            "2025-06-27T22:33:03 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1: storage exhausted while writing file within file system module - system bad file descriptor error fd='21'\n",
            "2025-06-27T22:33:03 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1 rcFS,rcFile,rcWriting,rcStorage,rcExhausted)\n",
            "2025-06-27T22:33:03 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:108:KSysFileDestroy_v1: storage exhausted while writing file within file system module - system bad file descriptor error fd='23'\n",
            "2025-06-27T22:33:03 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:108:KSysFileDestroy_v1 rcFS,rcFile,rcWriting,rcStorage,rcExhausted)\n",
            "2025-06-27T22:33:03 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:108:KSysFileDestroy_v1:  unknown while destroying file within file system module - unknown system error 'No space left on device(28)'\n",
            "2025-06-27T22:33:03 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:108:KSysFileDestroy_v1 rcFS,rcFile,rcDestroying,rcNoObj,rcUnknown)\n",
            "2025-06-27T22:33:03 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1: storage exhausted while writing file within file system module - system bad file descriptor error fd='20'\n",
            "2025-06-27T22:33:03 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1 rcFS,rcFile,rcWriting,rcStorage,rcExhausted)\n",
            "2025-06-27T22:33:03 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1: storage exhausted while writing file within file system module - system bad file descriptor error fd='18'\n",
            "2025-06-27T22:33:03 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1 rcFS,rcFile,rcWriting,rcStorage,rcExhausted)\n",
            "2025-06-27T22:33:03 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:108:KSysFileDestroy_v1:  unknown while destroying file within file system module - unknown system error 'No space left on device(28)'\n",
            "2025-06-27T22:33:03 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1 rcFS,rcFile,rcDestroying,rcNoObj,rcUnknown)\n",
            "2025-06-27T22:33:03 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1: storage exhausted while writing file within file system module - system bad file descriptor error fd='24'\n",
            "2025-06-27T22:33:03 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:464:KSysFileWrite_v1 rcFS,rcFile,rcWriting,rcStorage,rcExhausted)\n",
            "2025-06-27T22:33:03 fasterq-dump.2.11.3 err: libs/kfs/unix/sysfile.c:108:KSysFileDestroy_v1:  unknown while destroying file within file system module - unknown system error 'No space left on device(28)'\n",
            "2025-06-27T22:33:03 fasterq-dump.2.11.3 err: join_results.c destroy_join_printer() release_file() -> RC(libs/kfs/unix/sysfile.c:108:KSysFileDestroy_v1 rcFS,rcFile,rcDestroying,rcNoObj,rcUnknown)\n",
            "fasterq-dump quit with error code 3\n"
          ]
        }
      ],
      "source": [
        "output_dir = \"/content/drive/MyDrive/pituitary_atlas/raw/SRX29174030/SRR33975503/\"\n",
        "\n",
        "# 3. Create the directory if it doesn't exist\n",
        "# The -p flag ensures that parent directories are also created if they don't exist\n",
        "!mkdir -p \"$output_dir\"\n",
        "\n",
        "# 4. Change to the target directory for prefetch\n",
        "# This is important so prefetch downloads directly into the desired location\n",
        "%cd \"$output_dir\"\n",
        "!fasterq-dump SRR33975503.sra -O \"$output_dir\" -e 8 --split-files\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "FnBYv_aR1sSk"
      },
      "outputs": [],
      "source": []
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "zgF1zNFp1sVG"
      },
      "outputs": [],
      "source": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "p602-62yFMXR"
      },
      "source": [
        "# Compress all bus files!"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "M6tgs8qF_j60",
        "outputId": "e255f2ed-69e1-45d2-8a85-bd09dbc14d1d"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "-- The C compiler identification is GNU 11.4.0\n",
            "-- The CXX compiler identification is GNU 11.4.0\n",
            "-- Detecting C compiler ABI info\n",
            "-- Detecting C compiler ABI info - done\n",
            "-- Check for working C compiler: /usr/bin/cc - skipped\n",
            "-- Detecting C compile features\n",
            "-- Detecting C compile features - done\n",
            "-- Detecting CXX compiler ABI info\n",
            "-- Detecting CXX compiler ABI info - done\n",
            "-- Check for working CXX compiler: /usr/bin/c++ - skipped\n",
            "-- Detecting CXX compile features\n",
            "-- Detecting CXX compile features - done\n",
            "-- Performing Test CMAKE_HAVE_LIBC_PTHREAD\n",
            "-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success\n",
            "-- Found Threads: TRUE\n",
            "-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found version \"1.2.11\")\n",
            "-- Configuring done (2.1s)\n",
            "-- Generating done (0.0s)\n",
            "-- Build files have been written to: /content/bustools/build\n",
            "[  3%] Building CXX object src/CMakeFiles/bustools_core.dir/BUSData.cpp.o\n",
            "[  7%] Building CXX object src/CMakeFiles/bustools_core.dir/Common.cpp.o\n",
            "[ 11%] Building CXX object src/CMakeFiles/bustools_core.dir/bustools_capture.cpp.o\n",
            "[ 15%] Building CXX object src/CMakeFiles/bustools_core.dir/bustools_clusterhist.cpp.o\n",
            "[ 19%] Building CXX object src/CMakeFiles/bustools_core.dir/bustools_collapse.cpp.o\n",
            "[ 23%] Building CXX object src/CMakeFiles/bustools_core.dir/bustools_compress.cpp.o\n",
            "[ 26%] Building CXX object src/CMakeFiles/bustools_core.dir/bustools_correct.cpp.o\n",
            "[ 30%] Building CXX object src/CMakeFiles/bustools_core.dir/bustools_count.cpp.o\n",
            "[ 34%] Building CXX object src/CMakeFiles/bustools_core.dir/bustools_decompress.cpp.o\n",
            "[ 38%] Building CXX object src/CMakeFiles/bustools_core.dir/bustools_extract.cpp.o\n",
            "[ 42%] Building CXX object src/CMakeFiles/bustools_core.dir/bustools_inspect.cpp.o\n",
            "[ 46%] Building CXX object src/CMakeFiles/bustools_core.dir/bustools_linker.cpp.o\n",
            "[ 50%] Building CXX object src/CMakeFiles/bustools_core.dir/bustools_main.cpp.o\n",
            "[ 53%] Building CXX object src/CMakeFiles/bustools_core.dir/bustools_mash.cpp.o\n",
            "[ 57%] Building CXX object src/CMakeFiles/bustools_core.dir/bustools_merge.cpp.o\n",
            "[ 61%] Building CXX object src/CMakeFiles/bustools_core.dir/bustools_predict.cpp.o\n",
            "[ 65%] Building CXX object src/CMakeFiles/bustools_core.dir/bustools_project.cpp.o\n",
            "[ 69%] Building CXX object src/CMakeFiles/bustools_core.dir/bustools_sort.cpp.o\n",
            "[ 73%] Building CXX object src/CMakeFiles/bustools_core.dir/bustools_text.cpp.o\n",
            "[ 76%] Building CXX object src/CMakeFiles/bustools_core.dir/bustools_umicorrect.cpp.o\n",
            "[ 80%] Building CXX object src/CMakeFiles/bustools_core.dir/bustools_whitelist.cpp.o\n",
            "[ 84%] Building CXX object src/CMakeFiles/bustools_core.dir/hash.cpp.o\n",
            "[ 88%] Building C object src/CMakeFiles/bustools_core.dir/roaring.c.o\n",
            "[ 92%] Linking CXX static library libbustools_core.a\n",
            "[ 92%] Built target bustools_core\n",
            "[ 96%] Building CXX object src/CMakeFiles/bustools.dir/bustools_main.cpp.o\n",
            "[100%] Linking CXX executable bustools\n",
            "[100%] Built target bustools\n",
            "[ 92%] Built target bustools_core\n",
            "[100%] Built target bustools\n",
            "Install the project...\n",
            "-- Install configuration: \"Release\"\n",
            "-- Installing: /usr/local/bin/bustools\n"
          ]
        },
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "Cloning into 'bustools'...\n",
            "CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):\n",
            "  Compatibility with CMake < 3.5 will be removed from a future version of\n",
            "  CMake.\n",
            "\n",
            "  Update the VERSION argument <min> value or use a ...<max> suffix to tell\n",
            "  CMake that the project does not need compatibility with older versions.\n",
            "\n",
            "\n",
            "release mode\n"
          ]
        }
      ],
      "source": [
        "%%bash\n",
        "git clone https://github.com/BUStools/bustools && \\\n",
        "cd bustools && \\\n",
        "mkdir build && \\\n",
        "cd build && \\\n",
        "cmake .. && \\\n",
        "make && \\\n",
        "make install"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "kU1RrhhF0FTR",
        "outputId": "7b860665-2974-4116-f32c-342146c26409"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX8489818/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX8489818/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX8489819/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX8489819/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX8489820/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX8489820/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRR6364452/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRR6364453/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRR6364454/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRR6364455/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/ERR1869557/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/ERR1869558/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/ERR1869559/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/ERR1869560/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/ERR1869561/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/ERR1869562/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX4736337/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX4736337/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX4736338/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX4736338/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/ERX4978537/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/ERX4978537/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/ERX4978538/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/ERX4978538/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/ERX4978539/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/ERX4978539/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/ERX4978540/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/ERX4978540/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/ERX4978541/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/ERX4978541/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/ERX4978542/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/ERX4978542/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/ERX4978543/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/ERX4978543/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/ERX4978544/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/ERX4978544/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX8489821/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX8489821/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX8489822/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX8489822/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX8489823/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX8489823/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX8489824/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX8489824/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX5292053/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX5292053/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX6699502/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX6699502/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX6699503/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX6699503/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX7874737/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX7874737/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX7874738/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX7874738/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX7874739/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX7874739/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX7874740/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX7874740/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX7874741/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX7874741/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX7874742/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX7874742/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX7874743/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX7874743/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX7874744/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX7874744/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX7874745/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX7874745/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX7874746/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX7874746/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX9528564/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX9528564/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX9528565/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX9528565/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21241798/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21241798/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21241799/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21241799/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21241801/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21241801/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/ERX7593481/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/ERX7593481/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/ERX7593482/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/ERX7593482/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/ERX7593483/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/ERX7593483/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/ERX7593484/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/ERX7593484/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21170823/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21170823/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21170824/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21170824/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21986105/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21986105/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21986106/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21986106/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21986107/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21986107/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21986108/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21986108/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21986109/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21986109/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21986110/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21986110/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21986111/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21986111/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21986112/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21986112/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21986113/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21986113/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21986114/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21986114/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21986115/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21986115/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21986116/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21986116/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21241800/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21241800/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX18224104/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX18224104/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX18224105/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX18224105/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX18224106/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX18224106/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX18224107/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX18224107/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX18224108/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX18224108/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX18224109/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX18224109/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX15581370/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX15581370/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX15581371/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX15581371/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX8602211/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX8602211/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX8602212/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX8602212/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX8497092/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX8497092/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX8497093/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX8497093/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX8497094/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX8497094/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX9637224/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX9637224/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX7377164/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX7377164/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX7377166/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX7377166/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/MEMB-BRF1-MUT1/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/MEMB-BRF1-MUT1/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/MMP3-CONT1/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/MMP3-CONT1/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/MMP3-MUT1/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/MMP3-MUT1/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/MMYAPCont-C1/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/MMYAPCont-C1/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/MMYAPCont-C2/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/MMYAPCont-C2/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/MMYAPTetO-M1/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/MMYAPTetO-M1/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/MMYAPTetO-M2/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/MMYAPTetO-M2/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/MF2_MM5_CONTA/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/MF2_MM5_CONTA/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/sox2egfp_p3/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/sox2egfp_p3/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/sox2egfp_p15/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/sox2egfp_p15/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/sox2egfp_p56/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/sox2egfp_p56/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/sox2egfp_14m/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/sox2egfp_14m/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/e145wls_ctrl/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/e145wls_ctrl/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/e145wls_mut/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/e145wls_mut/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX24151432/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX24151432/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX24151433/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX24151433/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX24151434/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX24151434/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX24151435/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX24151435/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX24151436/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX24151436/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX24151437/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX24151437/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX24151438/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX24151438/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX24151439/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX24151439/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX22219346/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX22219346/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21636476/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21636476/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21636477/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/tcc/SRX21636477/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489818/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489818/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489819/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489819/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489820/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489820/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489821/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489821/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489822/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489822/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489823/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489823/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489824/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489824/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489834/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489834/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489835/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489835/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489836/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489836/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489837/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489837/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489838/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489838/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489839/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489839/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489841/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489841/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489840/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489840/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX5292053/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX5292053/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX6699502/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX6699502/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX6699503/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX4736337/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX4736337/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX9528564/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX9528564/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX4736338/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX4736338/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489842/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8489842/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX7874737/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX7874737/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX7874738/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX7874738/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX7874739/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX7874739/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX7874740/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX7874740/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX7874741/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX7874741/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX7874742/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX7874742/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX7874743/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX7874743/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX7874744/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX7874744/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX7874745/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX7874745/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX7874746/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX7874746/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX9528565/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX9528565/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21241798/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21241798/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21241799/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21241799/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21241800/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21241800/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21241801/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21241801/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21986115/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21986115/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21986114/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21986114/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21986113/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21986113/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21986112/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21986112/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21986111/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21986111/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21986105/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21986105/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21986110/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21986110/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21986106/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21986106/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21986109/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21986109/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21986107/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21986107/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21986108/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21986108/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21986116/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21986116/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21170824/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21170824/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21170823/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21170823/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/ERX7593484/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/ERX7593484/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/ERX7593483/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/ERX7593483/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/ERX7593482/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/ERX7593482/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/ERX7593481/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/ERX7593481/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/ERX4978537/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/ERX4978537/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/ERX4978538/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/ERX4978538/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/ERX4978542/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/ERX4978542/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/ERX4978539/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/ERX4978539/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/ERX4978543/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/ERX4978543/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/ERX4978544/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/ERX4978544/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/ERX4978540/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/ERX4978540/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/ERX4978541/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/ERX4978541/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8497094/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8497094/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8497093/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8497093/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8497092/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8497092/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8602212/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8602212/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8602211/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX8602211/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX15581371/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX15581371/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX15581370/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX15581370/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX18224109/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX18224109/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX18224108/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX18224108/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX18224107/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX18224107/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX18224106/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX18224106/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX18224105/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX18224105/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX18224104/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX18224104/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX13290057/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX13290057/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX13290058/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX13290058/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX13290059/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX13290059/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX13290060/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX13290060/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX13290061/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX13290061/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX13290062/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX13290062/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX11179262/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX11179262/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX11179262/tmp/output.s.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX11179262/tmp/output.s.c.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/MF2_MM5_CONTA/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/MF2_MM5_CONTA/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/MMYAPTetO-M2/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/MMYAPTetO-M2/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/MMYAPTetO-M1/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/MMYAPTetO-M1/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/MMYAPCont-C2/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/MMYAPCont-C2/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/MMYAPCont-C1/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/MMYAPCont-C1/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/MMP3-MUT1/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/MMP3-MUT1/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/MMP3-CONT1/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/MMP3-CONT1/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/MEMB-BRF1-MUT1/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/MEMB-BRF1-MUT1/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/MEMB-BRF1-CONT1/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/MEMB-BRF1-CONT1/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX11179257/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX11179257/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/e145wls_ctrl/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/e145wls_ctrl/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/sox2egfp_14m/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/sox2egfp_14m/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX11179258/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX11179258/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/sox2egfp_p3/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/sox2egfp_p3/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX11179259/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX11179259/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX11179260/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX11179260/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX11179261/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX11179261/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX11179263/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX11179263/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX9637224/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX9637224/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX7377164/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX7377164/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX7377166/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX7377166/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/e145wls_mut/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/e145wls_mut/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/sox2egfp_p56/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/sox2egfp_p56/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/sox2egfp_p15/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/sox2egfp_p15/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21636477/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21636477/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21636476/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX21636476/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX22219346/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX22219346/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX24151439/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX24151439/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX24151438/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX24151438/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX24151437/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX24151437/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX24151436/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX24151436/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX24151435/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX24151435/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX24151434/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX24151434/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX24151433/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX24151433/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX24151432/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX24151432/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX20679984/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX20679984/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX25532498/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX25532498/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX25532499/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX25532499/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX22219345/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX22219345/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX25839821/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX25839821/output.unfiltered.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX25839820/output.bus\n",
            "Compressing /content/ddrive/MyDrive/pituitary_atlas/processed/nac/SRX25839820/output.unfiltered.bus\n"
          ]
        }
      ],
      "source": [
        "import os\n",
        "import subprocess\n",
        "\n",
        "# Directory to scan\n",
        "directory = '/content/drive/MyDrive/pituitary_atlas/processed'\n",
        "\n",
        "# Walk through the directory and find all .bus files\n",
        "for root, dirs, files in os.walk(directory):\n",
        "    for file in files:\n",
        "        if file.endswith('.bus'):\n",
        "            file_path = os.path.join(root, file)\n",
        "            print(f'Compressing {file_path}')\n",
        "            command = f'bustools compress {file_path} -o {file_path}'\n",
        "            subprocess.run(command, shell=True)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "F5q4_SkaA78h",
        "outputId": "e54a8953-3cbc-48dc-e17f-4fac545b3957"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX17837195/output.bus\n",
            "Size: 364604241\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX17837196/output.bus\n",
            "Size: 309513041\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX17837197/output.bus\n",
            "Size: 456506801\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX17837198/output.bus\n",
            "Size: 436741393\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX17837199/output.bus\n",
            "Size: 347011153\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX17837200/output.bus\n",
            "Size: 429347697\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX17837201/output.bus\n",
            "Size: 450219633\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX17837202/output.bus\n",
            "Size: 399632849\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX10689726/output.bus\n",
            "Size: 1046971985\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX10689727/output.bus\n",
            "Size: 926407089\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX10689729/output.bus\n",
            "Size: 1106093905\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX10689725/output.bus\n",
            "Size: 1192768721\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX10689728/output.bus\n",
            "Size: 891519089\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/ERX10689730/output.bus\n",
            "Size: 275984529\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX25982300/output.bus\n",
            "Size: 1936088913\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX25982300/output.unfiltered.bus\n",
            "Size: 833712017\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX25982301/output.bus\n",
            "Size: 2185232721\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX25982301/output.unfiltered.bus\n",
            "Size: 1287633297\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX25982298/output.bus\n",
            "Size: 2029475281\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX25982298/output.unfiltered.bus\n",
            "Size: 977549969\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX25982297/output.bus\n",
            "Size: 2065613489\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX25982297/output.unfiltered.bus\n",
            "Size: 1017302417\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX25982296/output.bus\n",
            "Size: 1698094449\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX25982296/output.unfiltered.bus\n",
            "Size: 1039612529\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX25982295/output.bus\n",
            "Size: 2053459441\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX25982295/output.unfiltered.bus\n",
            "Size: 1145946065\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX25982294/output.bus\n",
            "Size: 2854962993\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX25982294/output.unfiltered.bus\n",
            "Size: 1585274769\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX26708357/output.bus\n",
            "Size: 12782142481\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX26708357/output.unfiltered.bus\n",
            "Size: 3235025937\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX25839821/output.bus\n",
            "Size: 7262957585\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX25839821/output.unfiltered.bus\n",
            "Size: 3042747057\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX25839820/output.bus\n",
            "Size: 12426360753\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX25839820/output.unfiltered.bus\n",
            "Size: 5238844817\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX22219345/output.bus\n",
            "Size: 4027779505\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX22219345/output.unfiltered.bus\n",
            "Size: 1530023409\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX25532499/output.bus\n",
            "Size: 5979531441\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX25532499/output.unfiltered.bus\n",
            "Size: 2353548849\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX25532498/output.bus\n",
            "Size: 7280570609\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX25532498/output.unfiltered.bus\n",
            "Size: 2758570673\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX21170825/output.bus\n",
            "Size: 12218980145\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX21170825/output.unfiltered.bus\n",
            "Size: 1994103985\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX25982299/output.bus\n",
            "Size: 2085041905\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/tcc/SRX25982299/output.unfiltered.bus\n",
            "Size: 918697841\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX25982294/output.bus\n",
            "Size: 3391856881\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX25982294/output.unfiltered.bus\n",
            "Size: 1955498001\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX25982295/output.bus\n",
            "Size: 2406111121\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX25982295/output.unfiltered.bus\n",
            "Size: 1396300817\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX25982296/output.bus\n",
            "Size: 2142949617\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX25982296/output.unfiltered.bus\n",
            "Size: 1351515217\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX25982297/output.bus\n",
            "Size: 2450018897\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX25982297/output.unfiltered.bus\n",
            "Size: 1271227793\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX25982298/output.bus\n",
            "Size: 2387372145\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX25982298/output.unfiltered.bus\n",
            "Size: 1204143313\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX25982301/output.bus\n",
            "Size: 2534589841\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX25982301/output.unfiltered.bus\n",
            "Size: 1551575761\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX25982300/output.bus\n",
            "Size: 2376738225\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX25982300/output.unfiltered.bus\n",
            "Size: 1059514257\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX25982299/output.bus\n",
            "Size: 2592613489\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX25982299/output.unfiltered.bus\n",
            "Size: 1177691985\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX17837195/output.bus\n",
            "Size: 442286641\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21170825/output.bus\n",
            "Size: 13285896913\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX21170825/output.unfiltered.bus\n",
            "Size: 2585008657\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX17837196/output.bus\n",
            "Size: 384865521\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX17837197/output.bus\n",
            "Size: 552581393\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX17837198/output.bus\n",
            "Size: 517479313\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX17837199/output.bus\n",
            "Size: 428737361\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX17837200/output.bus\n",
            "Size: 528847185\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX17837201/output.bus\n",
            "Size: 541318097\n",
            "Deleting /content/drive/MyDrive/pituitary_atlas/processed/nac/SRX17837202/output.bus\n",
            "Size: 503348785\n"
          ]
        }
      ],
      "source": [
        "# Directory to scan\n",
        "import os\n",
        "directory = '/content/drive/MyDrive/pituitary_atlas/processed'\n",
        "\n",
        "# Walk through the directory and find all .bus files\n",
        "for root, dirs, files in os.walk(directory):\n",
        "    for file in files:\n",
        "        if file.endswith('.bus'):\n",
        "          #delete file\n",
        "          print(f'Deleting {os.path.join(root, file)}')\n",
        "          #print size\n",
        "          print(f'Size: {os.path.getsize(os.path.join(root, file))}')\n",
        "          os.remove(os.path.join(root, file))"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "dShmv6oWCgMk"
      },
      "outputs": [],
      "source": [
        "import os\n",
        "\n",
        "# Directory to scan\n",
        "directory = '/content/drive/MyDrive/pituitary_atlas/raw'\n",
        "\n",
        "# Walk through the directory and find all files that don't end with .fastq or .fastq.gz\n",
        "for root, dirs, files in os.walk(directory):\n",
        "    for file in files:\n",
        "        # Check if the file does not end with .fastq or .fastq.gz\n",
        "        if not (file.endswith('.fastq') or file.endswith('.fastq.gz')):\n",
        "\n",
        "            #if it contains lite\n",
        "            if 'lite' in file:\n",
        "              print(f'File does not end with .fastq or .fastq.gz: {os.path.join(root, file)}')\n",
        "              #print size\n",
        "              print(f'Size: {os.path.getsize(os.path.join(root, file))}')\n",
        "              os.remove(os.path.join(root, file))\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "eCCH7RL_HiWp"
      },
      "outputs": [],
      "source": []
    }
  ],
  "metadata": {
    "colab": {
      "machine_shape": "hm",
      "provenance": [],
      "include_colab_link": true
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}